---
title: "Burden"
output: html_document
fontsize: 2pt
---
Motifs haven't been updated in this yet
ref = "total_ratio" changed to "self" when using show_burden_for_selected_features
Use a newly calculated set of motif change scores to define motif SNVs, but only focusing on motifs that more relevant
Here motifs are defined as relevant if they are 1) in top 50% brain expressed genes, and 2) are enriched in at least one gain-enriched brain module
Partition enhancers based on distance to TSS based on Yanyu's pipeline. 
Basically, her pipeline takes intersections between enhancers and distance ranges, and remove bases of enhancers that are not overlap. 

### NOTICE

normalized using the number of all SNVs, so it i not normalized by the number of all SNVs in regulatory regions. But the trend should be consistent. 

non ASD genes are the bottom 20%
```{r}
getwd()
knitr::opts_chunk$set(warning=FALSE, message=FALSE)
```
### read in coding tada data

```{r,echo=FALSE}
getwd()
# need to set this direcoty to full direcoty of ASD/analysis
coding_tada = read.delim("../other_annotation/gene_list/TADA_SNV_CNV_combined_Feb7.txt", header = TRUE, sep = "\t", stringsAsFactors = FALSE)
coding_tada_denovo = data.frame(coding_tada$RefSeqName, coding_tada$TadaName, coding_tada$mut.rate, coding_tada$dn.LoF, coding_tada$dn.mis3, coding_tada$BF.SNV.dn, coding_tada$BF.combined, coding_tada$qvalue.combined)
colnames(coding_tada_denovo) = c("RefSeqName", "TadaName", "mut.rate", "dn.LoF", "dn.mis3", "BF.SNV.dn", "BF.combined", "qvalue.combined")

#read in workspace that has unique mutation information
mutation = new.env()
load("../data/debug_region_list_073116_data_matrixtransformed_for_old_code.Rdata", envir = mutation)
write.table(mutation$mutation[,1:4],"160229_data_for_analysis.bed",col.names = FALSE, row.names = FALSE, sep = "\t", quote = FALSE)
# load releant functions
source("../lib/160930_screen_burden_functions_test_more_genelist_focused.R")

```




Noonan enhancers + GERP + NP_genes
```{r}
#knitr::kable(burden_trend_for_each_epi_data("160229_data_for_analysis.bed", "../other_annotation/epigenomic_annotation/", "Noonan_brain_union", "GERP", "NP_genes")$table)
burden_trend_for_each_epi_data("160229_data_for_analysis.bed", "../other_annotation/epigenomic_annotation/", "Noonan_brain_union", "GERP", "NP_genes")$graph
```

Encode DHS + GERP + NP_genes
```{r}
#knitr::kable(burden_trend_for_each_epi_data("160229_data_for_analysis.bed", "../other_annotation/epigenomic_annotation/", "Encode_DHS_union", "GERP", "NP_genes")$table)
burden_trend_for_each_epi_data("160229_data_for_analysis.bed", "../other_annotation/epigenomic_annotation/", "Encode_DHS_union", "GERP", "NP_genes")$graph
```

Noonan enhancers + CADD + NP_genes
```{r}
#knitr::kable(burden_trend_for_each_epi_data("160229_data_for_analysis.bed", "../other_annotation/epigenomic_annotation/", "Noonan_brain_union", "CADD", "NP_genes")$table)
burden_trend_for_each_epi_data("160229_data_for_analysis.bed", "../other_annotation/epigenomic_annotation/", "Noonan_brain_union", "CADD", "NP_genes")$graph

```

Encode DHS + CADD + NP_genes
```{r}
#knitr::kable(burden_trend_for_each_epi_data("160229_data_for_analysis.bed", "../other_annotation/epigenomic_annotation/", "Encode_DHS_union", "CADD", "NP_genes")$table)
burden_trend_for_each_epi_data("160229_data_for_analysis.bed", "../other_annotation/epigenomic_annotation/", "Encode_DHS_union", "CADD", "NP_genes")$graph
```

Noonan enhancers + GERP + all genes
```{r}
#knitr::kable(burden_trend_for_each_epi_data("160229_data_for_analysis.bed", "../other_annotation/epigenomic_annotation/", "Noonan_brain_union", "GERP", "all_genes")$table)
burden_trend_for_each_epi_data("160229_data_for_analysis.bed", "../other_annotation/epigenomic_annotation/", "Noonan_brain_union", "GERP", "all_genes")$graph
```

Encode DHS + GERP + all genes
```{r}
#knitr::kable(burden_trend_for_each_epi_data("160229_data_for_analysis.bed", "../other_annotation/epigenomic_annotation/", "Encode_DHS_union", "GERP", "all_genes")$table)
burden_trend_for_each_epi_data("160229_data_for_analysis.bed", "../other_annotation/epigenomic_annotation/", "Encode_DHS_union", "GERP", "all_genes")$graph
```

Noonan enhancers + CADD + all genes
```{r}
#knitr::kable(burden_trend_for_each_epi_data("160229_data_for_analysis.bed", "../other_annotation/epigenomic_annotation/", "Noonan_brain_union", "CADD", "all_genes")$table)
burden_trend_for_each_epi_data("160229_data_for_analysis.bed", "../other_annotation/epigenomic_annotation/", "Noonan_brain_union", "CADD", "all_genes")$graph

```

Encode DHS + CADD + all genes
```{r}
#knitr::kable(burden_trend_for_each_epi_data("160229_data_for_analysis.bed", "../other_annotation/epigenomic_annotation/", "Encode_DHS_union", "CADD", "all_genes")$table)
burden_trend_for_each_epi_data("160229_data_for_analysis.bed", "../other_annotation/epigenomic_annotation/", "Encode_DHS_union", "CADD", "all_genes")$graph
```