---
title: "Burden"
output: html_document
fontsize: 2pt
---

Use a newly calculated set of motif change scores to define motif SNVs, but only focusing on motifs that more relevant
Here motifs are defined as relevant if they are 1) in top 50% brain expressed genes, and 2) are enriched in at least one gain-enriched brain module
Partition enhancers based on distance to TSS based on Yanyu's pipeline. 
Basically, her pipeline takes intersections between enhancers and distance ranges, and remove bases of enhancers that are not overlap. 
```{r}
getwd()
knitr::opts_chunk$set(warning=FALSE, message=FALSE)
```
### read in coding tada data

```{r,echo=FALSE}
getwd()
# need to set this direcoty to full direcoty of ASD/analysis
coding_tada = read.delim("../other_annotation/gene_list/TADA_SNV_CNV_combined_Feb7.txt", header = TRUE, sep = "\t", stringsAsFactors = FALSE)
coding_tada_denovo = data.frame(coding_tada$RefSeqName, coding_tada$TadaName, coding_tada$mut.rate, coding_tada$dn.LoF, coding_tada$dn.mis3, coding_tada$BF.SNV.dn, coding_tada$BF.combined, coding_tada$qvalue.combined)
colnames(coding_tada_denovo) = c("RefSeqName", "TadaName", "mut.rate", "dn.LoF", "dn.mis3", "BF.SNV.dn", "BF.combined", "qvalue.combined")

#read in workspace that has unique mutation information
mutation = new.env()
load("../data/161026_Stephen_Scherer_Gemomic_Medicine_ASD_de_novo_SNVs_germline_with_NC_control_data_matrixtransformed_for_old_code.Rdata", envir = mutation)
write.table(mutation$mutation[,1:4],"161026_stephen_scherer_data.bed",col.names = FALSE, row.names = FALSE, sep = "\t", quote = FALSE)

# load releant functions
source("../lib/160930_screen_burden_functions_test_more_genelist_focused.R")

```


Whole genome

```{r}
temp = generate_mut_enhancer_pair_for_burden_analysis("161026_stephen_scherer_data.bed","../other_annotation/epigenomic_annotation/Whole_genome.promoter_1kb_genename.bed")
head(temp)
knitr::kable(show_burden_for_selected_features(temp,input_type = "table",ref = "self"),digit = c(rep(2,5),5,5,5), col.names = table_colname)

```

```{r}
temp = generate_mut_enhancer_pair_for_burden_analysis("161026_stephen_scherer_data.bed","../other_annotation/epigenomic_annotation/Whole_genome.5utr_genename.bed")
head(temp)
knitr::kable(show_burden_for_selected_features(temp,input_type = "table",ref = "self"),digit = c(rep(2,5),5,5,5), col.names = table_colname)

```

```{r}
temp = generate_mut_enhancer_pair_for_burden_analysis("161026_stephen_scherer_data.bed","../other_annotation/epigenomic_annotation/Whole_genome.3utr_genename.bed")
head(temp)
knitr::kable(show_burden_for_selected_features(temp,input_type = "table",ref = "self"),digit = c(rep(2,5),5,5,5), col.names = table_colname)

```

```{r}
# combine utr5 and utr3, the code here wouldn't have double counted mutations because of overlaps in utr5 and utr3

temp = generate_mut_enhancer_pair_for_burden_analysis("161026_stephen_scherer_data.bed","../other_annotation/epigenomic_annotation/Whole_genome.3utr_genename.bed")
temp2 = generate_mut_enhancer_pair_for_burden_analysis("161026_stephen_scherer_data.bed","../other_annotation/epigenomic_annotation/Whole_genome.5utr_genename.bed")
temp = rbind(temp,temp2)
temp = temp[!duplicated(temp),]
knitr::kable(show_burden_for_selected_features(temp,input_type = "table",ref = "self"),digit = c(rep(2,5),5,5,5), col.names = table_colname)

```




```{r}
temp = generate_mut_enhancer_pair_for_burden_analysis("161026_stephen_scherer_data.bed","../other_annotation/epigenomic_annotation/Whole_genome.yanyu_pipeline_enhancers.10000.bp_within_TSS.bed")
head(temp)
knitr::kable(show_burden_for_selected_features(temp,input_type = "table",ref = "self"),digit = c(rep(2,5),5,5,5), col.names = table_colname)

```

```{r}
temp = generate_mut_enhancer_pair_for_burden_analysis("161026_stephen_scherer_data.bed","../other_annotation/epigenomic_annotation/Whole_genome.yanyu_pipeline_enhancers.10000_to_20000.bp_within_TSS.bed")
head(temp)
knitr::kable(show_burden_for_selected_features(temp,input_type = "table",ref = "self"),digit = c(rep(2,5),5,5,5), col.names = table_colname)

```


Noonan enhancers
```{r}
temp = generate_mut_enhancer_pair_for_burden_analysis("161026_stephen_scherer_data.bed","../other_annotation/epigenomic_annotation/Noonan_brain_union.yanyu_pipeline_enhancers.10000.bp_within_TSS.bed")
head(temp)
knitr::kable(show_burden_for_selected_features(temp,input_type = "table",ref = "self"),digit = c(rep(2,5),5,5,5), col.names = table_colname)

```

```{r}
temp = generate_mut_enhancer_pair_for_burden_analysis("161026_stephen_scherer_data.bed","../other_annotation/epigenomic_annotation/Noonan_brain_union.yanyu_pipeline_enhancers.10000_to_20000.bp_within_TSS.bed")
head(temp)
knitr::kable(show_burden_for_selected_features(temp,input_type = "table",ref = "self"),digit = c(rep(2,5),5,5,5), col.names = table_colname)

```

```{r}
temp = generate_mut_enhancer_pair_for_burden_analysis("161026_stephen_scherer_data.bed","../other_annotation/epigenomic_annotation/Noonan_brain_union.promoter_1kb_genename.bed")
head(temp)
knitr::kable(show_burden_for_selected_features(temp,input_type = "table",ref = "self"),digit = c(rep(2,5),5,5,5), col.names = table_colname)

```

```{r}
temp = generate_mut_enhancer_pair_for_burden_analysis("161026_stephen_scherer_data.bed","../other_annotation/epigenomic_annotation/Noonan_brain_union.5utr_genename.bed")
head(temp)
knitr::kable(show_burden_for_selected_features(temp,input_type = "table",ref = "self"),digit = c(rep(2,5),5,5,5), col.names = table_colname)

```

```{r}
temp = generate_mut_enhancer_pair_for_burden_analysis("161026_stephen_scherer_data.bed","../other_annotation/epigenomic_annotation/Noonan_brain_union.3utr_genename.bed")
head(temp)
knitr::kable(show_burden_for_selected_features(temp,input_type = "table",ref = "self"),digit = c(rep(2,5),5,5,5), col.names = table_colname)

```


```{r}
# combine utr5 and utr3, the code here wouldn't have double counted mutations because of overlaps in utr5 and utr3

temp = generate_mut_enhancer_pair_for_burden_analysis("161026_stephen_scherer_data.bed","../other_annotation/epigenomic_annotation/Noonan_brain_union.3utr_genename.bed")
temp2 = generate_mut_enhancer_pair_for_burden_analysis("161026_stephen_scherer_data.bed","../other_annotation/epigenomic_annotation/Noonan_brain_union.5utr_genename.bed")
temp = rbind(temp,temp2)
temp = temp[!duplicated(temp),]
knitr::kable(show_burden_for_selected_features(temp,input_type = "table",ref = "self"),digit = c(rep(2,5),5,5,5), col.names = table_colname)

```


DHS
```{r}
temp = generate_mut_enhancer_pair_for_burden_analysis("161026_stephen_scherer_data.bed","../other_annotation/epigenomic_annotation/Encode_DHS_union.yanyu_pipeline_enhancers.10000.bp_within_TSS.bed")
head(temp)
knitr::kable(show_burden_for_selected_features(temp,input_type = "table",ref = "self"),digit = c(rep(2,5),5,5,5), col.names = table_colname)

```

```{r}
temp = generate_mut_enhancer_pair_for_burden_analysis("161026_stephen_scherer_data.bed","../other_annotation/epigenomic_annotation/Encode_DHS_union.yanyu_pipeline_enhancers.10000_to_20000.bp_within_TSS.bed")
head(temp)
knitr::kable(show_burden_for_selected_features(temp,input_type = "table",ref = "self"),digit = c(rep(2,5),5,5,5), col.names = table_colname)

```

```{r}
temp = generate_mut_enhancer_pair_for_burden_analysis("161026_stephen_scherer_data.bed","../other_annotation/epigenomic_annotation/Encode_DHS_union.promoter_1kb_genename.bed")
head(temp)
knitr::kable(show_burden_for_selected_features(temp,input_type = "table",ref = "self"),digit = c(rep(2,5),5,5,5), col.names = table_colname)

```

```{r}
temp = generate_mut_enhancer_pair_for_burden_analysis("161026_stephen_scherer_data.bed","../other_annotation/epigenomic_annotation/Encode_DHS_union.5utr_genename.bed")
head(temp)
knitr::kable(show_burden_for_selected_features(temp,input_type = "table",ref = "self"),digit = c(rep(2,5),5,5,5), col.names = table_colname)

```

```{r}
temp = generate_mut_enhancer_pair_for_burden_analysis("161026_stephen_scherer_data.bed","../other_annotation/epigenomic_annotation/Encode_DHS_union.3utr_genename.bed")
head(temp)
knitr::kable(show_burden_for_selected_features(temp,input_type = "table",ref = "self"),digit = c(rep(2,5),5,5,5), col.names = table_colname)

```

```{r}
# combine utr5 and utr3, the code here wouldn't have double counted mutations because of overlaps in utr5 and utr3

temp = generate_mut_enhancer_pair_for_burden_analysis("161026_stephen_scherer_data.bed","../other_annotation/epigenomic_annotation/Encode_DHS_union.3utr_genename.bed")
temp2 = generate_mut_enhancer_pair_for_burden_analysis("161026_stephen_scherer_data.bed","../other_annotation/epigenomic_annotation/Encode_DHS_union.5utr_genename.bed")
temp = rbind(temp,temp2)
temp = temp[!duplicated(temp),]
knitr::kable(show_burden_for_selected_features(temp,input_type = "table",ref = "self"),digit = c(rep(2,5),5,5,5), col.names = table_colname)

```


Brain roadmap
```{r}
temp = generate_mut_enhancer_pair_for_burden_analysis("161026_stephen_scherer_data.bed","../other_annotation/epigenomic_annotation/Brain_roadmap_union.yanyu_pipeline_enhancers.10000.bp_within_TSS.bed")
head(temp)
knitr::kable(show_burden_for_selected_features(temp,input_type = "table",ref = "self"),digit = c(rep(2,5),5,5,5), col.names = table_colname)

```

```{r}
temp = generate_mut_enhancer_pair_for_burden_analysis("161026_stephen_scherer_data.bed","../other_annotation/epigenomic_annotation/Brain_roadmap_union.yanyu_pipeline_enhancers.10000_to_20000.bp_within_TSS.bed")
head(temp)
knitr::kable(show_burden_for_selected_features(temp,input_type = "table",ref = "self"),digit = c(rep(2,5),5,5,5), col.names = table_colname)

```

```{r}
temp = generate_mut_enhancer_pair_for_burden_analysis("161026_stephen_scherer_data.bed","../other_annotation/epigenomic_annotation/Brain_roadmap_union.promoter_1kb_genename.bed")
head(temp)
knitr::kable(show_burden_for_selected_features(temp,input_type = "table",ref = "self"),digit = c(rep(2,5),5,5,5), col.names = table_colname)

```

```{r}
temp = generate_mut_enhancer_pair_for_burden_analysis("161026_stephen_scherer_data.bed","../other_annotation/epigenomic_annotation/Brain_roadmap_union.5utr_genename.bed")
head(temp)
knitr::kable(show_burden_for_selected_features(temp,input_type = "table",ref = "self"),digit = c(rep(2,5),5,5,5), col.names = table_colname)

```

```{r}
temp = generate_mut_enhancer_pair_for_burden_analysis("161026_stephen_scherer_data.bed","../other_annotation/epigenomic_annotation/Brain_roadmap_union.3utr_genename.bed")
head(temp)
knitr::kable(show_burden_for_selected_features(temp,input_type = "table",ref = "self"),digit = c(rep(2,5),5,5,5), col.names = table_colname)

```

```{r}
# combine utr5 and utr3, the code here wouldn't have double counted mutations because of overlaps in utr5 and utr3

temp = generate_mut_enhancer_pair_for_burden_analysis("161026_stephen_scherer_data.bed","../other_annotation/epigenomic_annotation/Brain_roadmap_union.3utr_genename.bed")
temp2 = generate_mut_enhancer_pair_for_burden_analysis("161026_stephen_scherer_data.bed","../other_annotation/epigenomic_annotation/Brain_roadmap_union.5utr_genename.bed")
temp = rbind(temp,temp2)
temp = temp[!duplicated(temp),]
knitr::kable(show_burden_for_selected_features(temp,input_type = "table",ref = "self"),digit = c(rep(2,5),5,5,5), col.names = table_colname)

```


Fantom fetal brain
```{r}
temp = generate_mut_enhancer_pair_for_burden_analysis("161026_stephen_scherer_data.bed","../other_annotation/epigenomic_annotation/Fantom_fetal_brain_500bp_union.yanyu_pipeline_enhancers.10000.bp_within_TSS.bed")
head(temp)
knitr::kable(show_burden_for_selected_features(temp,input_type = "table",ref = "self"),digit = c(rep(2,5),5,5,5), col.names = table_colname)

```

```{r}
temp = generate_mut_enhancer_pair_for_burden_analysis("161026_stephen_scherer_data.bed","../other_annotation/epigenomic_annotation/Fantom_fetal_brain_500bp_union.yanyu_pipeline_enhancers.10000_to_20000.bp_within_TSS.bed")
head(temp)
knitr::kable(show_burden_for_selected_features(temp,input_type = "table",ref = "self"),digit = c(rep(2,5),5,5,5), col.names = table_colname)

```

```{r}
temp = generate_mut_enhancer_pair_for_burden_analysis("161026_stephen_scherer_data.bed","../other_annotation/epigenomic_annotation/Fantom_fetal_brain_500bp_union.promoter_1kb_genename.bed")
temp
knitr::kable(show_burden_for_selected_features(temp,input_type = "table",ref = "self"),digit = c(rep(2,5),5,5,5), col.names = table_colname)

```

```{r}
temp = generate_mut_enhancer_pair_for_burden_analysis("161026_stephen_scherer_data.bed","../other_annotation/epigenomic_annotation/Fantom_fetal_brain_500bp_union.5utr_genename.bed")
head(temp)
knitr::kable(show_burden_for_selected_features(temp,input_type = "table",ref = "self"),digit = c(rep(2,5),5,5,5), col.names = table_colname)

```

```{r}
temp = generate_mut_enhancer_pair_for_burden_analysis("161026_stephen_scherer_data.bed","../other_annotation/epigenomic_annotation/Fantom_fetal_brain_500bp_union.3utr_genename.bed")
head(temp)
knitr::kable(show_burden_for_selected_features(temp,input_type = "table",ref = "self"),digit = c(rep(2,5),5,5,5), col.names = table_colname)

```

```{r}
# combine utr5 and utr3, the code here wouldn't have double counted mutations because of overlaps in utr5 and utr3

temp = generate_mut_enhancer_pair_for_burden_analysis("161026_stephen_scherer_data.bed","../other_annotation/epigenomic_annotation/Fantom_fetal_brain_500bp_union.3utr_genename.bed")
temp2 = generate_mut_enhancer_pair_for_burden_analysis("161026_stephen_scherer_data.bed","../other_annotation/epigenomic_annotation/Fantom_fetal_brain_500bp_union.5utr_genename.bed")
temp = rbind(temp,temp2)
temp = temp[!duplicated(temp),]
knitr::kable(show_burden_for_selected_features(temp,input_type = "table",ref = "self"),digit = c(rep(2,5),5,5,5), col.names = table_colname)

```