---
title: "Burden"
output: html_document
fontsize: 2pt
---
Motifs haven't been updated in this yet
ref = "total_ratio" changed to "self" when using show_burden_for_selected_features
Use a newly calculated set of motif change scores to define motif SNVs, but only focusing on motifs that more relevant
Here motifs are defined as relevant if they are 1) in top 50% brain expressed genes, and 2) are enriched in at least one gain-enriched brain module
Partition enhancers based on distance to TSS based on Yanyu's pipeline. 
Basically, her pipeline takes intersections between enhancers and distance ranges, and remove bases of enhancers that are not overlap. 

non ASD genes are the bottom 20%
not plotting mutation is 5'UTR or 3'UTR
added one row for showing the union of enhancers and promoters
Only have Fantom, Noonan and Whole genome
Look at more genesets. 
```{r}
getwd()
knitr::opts_chunk$set(warning=FALSE, message=FALSE)
```
### read in coding tada data

```{r,echo=FALSE}
getwd()
# need to set this direcoty to full direcoty of ASD/analysis
coding_tada = read.delim("../other_annotation/gene_list/TADA_SNV_CNV_combined_Feb7.txt", header = TRUE, sep = "\t", stringsAsFactors = FALSE)
coding_tada_denovo = data.frame(coding_tada$RefSeqName, coding_tada$TadaName, coding_tada$mut.rate, coding_tada$dn.LoF, coding_tada$dn.mis3, coding_tada$BF.SNV.dn, coding_tada$BF.combined, coding_tada$qvalue.combined)
colnames(coding_tada_denovo) = c("RefSeqName", "TadaName", "mut.rate", "dn.LoF", "dn.mis3", "BF.SNV.dn", "BF.combined", "qvalue.combined")

#read in workspace that has unique mutation information
mutation = new.env()
load("../data/debug_region_list_073116_data_matrixtransformed_for_old_code.Rdata", envir = mutation)
write.table(mutation$mutation[,1:4],"160229_data_for_analysis.bed",col.names = FALSE, row.names = FALSE, sep = "\t", quote = FALSE)

# load releant functions
source("../lib/160930_screen_burden_functions_test_more_genelist_focused.R")

```

```{r,echo=FALSE}
plot_summary <- data.frame(gene_list = NA, score = NA, cutoff = NA, burden = NA, CI_low = NA, CI_up = NA, pvalue = NA, asd_sub = NA, control_sub = NA, bed = NA, region_tag = NA, baseline_ASD = NA, baseline_control = NA)
```

Whole genome

```{r,echo=FALSE}
temp = generate_mut_enhancer_pair_for_burden_analysis("160229_data_for_analysis.bed","../other_annotation/epigenomic_annotation/Whole_genome.promoter_1kb_genename.bed")

temp1 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = stringent_ASD)
temp1 = data.frame(lapply(translate_to_plotting_v2(temp1, "Whole_genome", "Promoter","stringent_ASD"), function(y) if(is.numeric(y)) round(y, 3) else y)) 

temp2 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = ID_genes)
temp2 = data.frame(lapply(translate_to_plotting_v2(temp2, "Whole_genome", "Promoter","ID_genes"), function(y) if(is.numeric(y)) round(y, 3) else y)) 

temp3 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = SCZ_composite)
temp3 = data.frame(lapply(translate_to_plotting_v2(temp3, "Whole_genome", "Promoter","SCZ_composite"), function(y) if(is.numeric(y)) round(y, 3) else y)) 

temp4 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = relaxed_ASD)
temp4 = data.frame(lapply(translate_to_plotting_v2(temp4, "Whole_genome", "Promoter","NP_genes"), function(y) if(is.numeric(y)) round(y, 3) else y)) 

temp5 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = nonASD_genes)
temp5 = data.frame(lapply(translate_to_plotting_v2(temp5, "Whole_genome", "Promoter","nonASD_genes"), function(y) if(is.numeric(y)) round(y, 3) else y))

plot_summary <- do.call("rbind",list(plot_summary, temp1,temp2,temp3,temp4,temp5))

```


```{r,echo=FALSE}
# combine promoter and enhancers, the code here wouldn't have double counted mutations because of overlaps in utr5 and utr3

temp = generate_mut_enhancer_pair_for_burden_analysis("160229_data_for_analysis.bed","../other_annotation/epigenomic_annotation/Whole_genome.yanyu_pipeline_enhancers.10000.bp_within_TSS.bed")
temp2 = generate_mut_enhancer_pair_for_burden_analysis("160229_data_for_analysis.bed","../other_annotation/epigenomic_annotation/Whole_genome.yanyu_pipeline_enhancers.10000_to_20000.bp_within_TSS.bed")
temp3 = generate_mut_enhancer_pair_for_burden_analysis("160229_data_for_analysis.bed","../other_annotation/epigenomic_annotation/Whole_genome.promoter_1kb_genename.bed")

temp = do.call("rbind",list(temp,temp2,temp3))
temp = temp[!duplicated(temp),]

temp1 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = stringent_ASD)
temp1 = data.frame(lapply(translate_to_plotting_v2(temp1, "Whole_genome", "Promoter+enhancer","stringent_ASD"), function(y) if(is.numeric(y)) round(y, 3) else y)) 

temp2 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = ID_genes)
temp2 = data.frame(lapply(translate_to_plotting_v2(temp2, "Whole_genome", "Promoter+enhancer","ID_genes"), function(y) if(is.numeric(y)) round(y, 3) else y)) 

temp3 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = SCZ_composite)
temp3 = data.frame(lapply(translate_to_plotting_v2(temp3, "Whole_genome", "Promoter+enhancer","SCZ_composite"), function(y) if(is.numeric(y)) round(y, 3) else y)) 

temp4 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = relaxed_ASD)
temp4 = data.frame(lapply(translate_to_plotting_v2(temp4, "Whole_genome", "Promoter+enhancer","NP_genes"), function(y) if(is.numeric(y)) round(y, 3) else y)) 

temp5 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = nonASD_genes)
temp5 = data.frame(lapply(translate_to_plotting_v2(temp5, "Whole_genome", "Promoter+enhancer","nonASD_genes"), function(y) if(is.numeric(y)) round(y, 3) else y))

plot_summary <- do.call("rbind",list(plot_summary, temp1,temp2,temp3,temp4,temp5))


```

```{r,echo=FALSE}
# combine promoter and 10kb enhancers, the code here wouldn't have double counted mutations because of overlaps in utr5 and utr3

temp = generate_mut_enhancer_pair_for_burden_analysis("160229_data_for_analysis.bed","../other_annotation/epigenomic_annotation/Whole_genome.yanyu_pipeline_enhancers.10000.bp_within_TSS.bed")

temp3 = generate_mut_enhancer_pair_for_burden_analysis("160229_data_for_analysis.bed","../other_annotation/epigenomic_annotation/Whole_genome.promoter_1kb_genename.bed")

temp = do.call("rbind",list(temp,temp3))
temp = temp[!duplicated(temp),]

temp1 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = stringent_ASD)
temp1 = data.frame(lapply(translate_to_plotting_v2(temp1, "Whole_genome", "Promoter+10kb_enhancer","stringent_ASD"), function(y) if(is.numeric(y)) round(y, 3) else y)) 

temp2 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = ID_genes)
temp2 = data.frame(lapply(translate_to_plotting_v2(temp2, "Whole_genome", "Promoter+10kb_enhancer","ID_genes"), function(y) if(is.numeric(y)) round(y, 3) else y)) 

temp3 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = SCZ_composite)
temp3 = data.frame(lapply(translate_to_plotting_v2(temp3, "Whole_genome", "Promoter+10kb_enhancer","SCZ_composite"), function(y) if(is.numeric(y)) round(y, 3) else y)) 

temp4 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = relaxed_ASD)
temp4 = data.frame(lapply(translate_to_plotting_v2(temp4, "Whole_genome", "Promoter+10kb_enhancer","NP_genes"), function(y) if(is.numeric(y)) round(y, 3) else y)) 

temp5 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = nonASD_genes)
temp5 = data.frame(lapply(translate_to_plotting_v2(temp5, "Whole_genome", "Promoter+10kb_enhancer","nonASD_genes"), function(y) if(is.numeric(y)) round(y, 3) else y)) 


plot_summary <- do.call("rbind",list(plot_summary, temp1,temp2,temp3,temp4,temp5))


```


```{r,echo=FALSE}
temp = generate_mut_enhancer_pair_for_burden_analysis("160229_data_for_analysis.bed","../other_annotation/epigenomic_annotation/Whole_genome.yanyu_pipeline_enhancers.10000.bp_within_TSS.bed")

temp1 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = stringent_ASD)
temp1 = data.frame(lapply(translate_to_plotting_v2(temp1, "Whole_genome", "1-10Kb","stringent_ASD"), function(y) if(is.numeric(y)) round(y, 3) else y)) 

temp2 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = ID_genes)
temp2 = data.frame(lapply(translate_to_plotting_v2(temp2, "Whole_genome", "1-10Kb","ID_genes"), function(y) if(is.numeric(y)) round(y, 3) else y)) 

temp3 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = SCZ_composite)
temp3 = data.frame(lapply(translate_to_plotting_v2(temp3, "Whole_genome", "1-10Kb","SCZ_composite"), function(y) if(is.numeric(y)) round(y, 3) else y)) 

temp4 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = relaxed_ASD)
temp4 = data.frame(lapply(translate_to_plotting_v2(temp4, "Whole_genome", "1-10Kb","NP_genes"), function(y) if(is.numeric(y)) round(y, 3) else y)) 

temp5 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = nonASD_genes)
temp5 = data.frame(lapply(translate_to_plotting_v2(temp5, "Whole_genome", "1-10Kb","nonASD_genes"), function(y) if(is.numeric(y)) round(y, 3) else y))

plot_summary <- do.call("rbind",list(plot_summary, temp1,temp2,temp3,temp4,temp5))
```

```{r,echo=FALSE}
temp = generate_mut_enhancer_pair_for_burden_analysis("160229_data_for_analysis.bed","../other_annotation/epigenomic_annotation/Whole_genome.yanyu_pipeline_enhancers.10000_to_20000.bp_within_TSS.bed")

temp1 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = stringent_ASD)
temp1 = data.frame(lapply(translate_to_plotting_v2(temp1, "Whole_genome", "10-20Kb","stringent_ASD"), function(y) if(is.numeric(y)) round(y, 3) else y)) 

temp2 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = ID_genes)
temp2 = data.frame(lapply(translate_to_plotting_v2(temp2, "Whole_genome", "10-20Kb","ID_genes"), function(y) if(is.numeric(y)) round(y, 3) else y)) 

temp3 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = SCZ_composite)
temp3 = data.frame(lapply(translate_to_plotting_v2(temp3, "Whole_genome", "10-20Kb","SCZ_composite"), function(y) if(is.numeric(y)) round(y, 3) else y)) 

temp4 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = relaxed_ASD)
temp4 = data.frame(lapply(translate_to_plotting_v2(temp4, "Whole_genome", "10-20Kb","NP_genes"), function(y) if(is.numeric(y)) round(y, 3) else y)) 

temp5 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = nonASD_genes)
temp5 = data.frame(lapply(translate_to_plotting_v2(temp5, "Whole_genome", "10-20kb","nonASD_genes"), function(y) if(is.numeric(y)) round(y, 3) else y))

plot_summary <- do.call("rbind",list(plot_summary, temp1,temp2,temp3,temp4,temp5))

```


Noonan enhancers
```{r,echo=FALSE}
temp = generate_mut_enhancer_pair_for_burden_analysis("160229_data_for_analysis.bed","../other_annotation/epigenomic_annotation/Noonan_brain_union.promoter_1kb_genename.bed")

temp1 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = stringent_ASD)
temp1 = data.frame(lapply(translate_to_plotting_v2(temp1, "Noonan_brain_union", "Promoter","stringent_ASD"), function(y) if(is.numeric(y)) round(y, 3) else y)) 

temp2 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = ID_genes)
temp2 = data.frame(lapply(translate_to_plotting_v2(temp2, "Noonan_brain_union", "Promoter","ID_genes"), function(y) if(is.numeric(y)) round(y, 3) else y)) 

temp3 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = SCZ_composite)
temp3 = data.frame(lapply(translate_to_plotting_v2(temp3, "Noonan_brain_union", "Promoter","SCZ_composite"), function(y) if(is.numeric(y)) round(y, 3) else y)) 

temp4 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = relaxed_ASD)
temp4 = data.frame(lapply(translate_to_plotting_v2(temp4, "Noonan_brain_union", "Promoter","NP_genes"), function(y) if(is.numeric(y)) round(y, 3) else y)) 

temp5 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = nonASD_genes)
temp5 = data.frame(lapply(translate_to_plotting_v2(temp5, "Noonan_brain_union", "Promoter","nonASD_genes"), function(y) if(is.numeric(y)) round(y, 3) else y))

plot_summary <- do.call("rbind",list(plot_summary, temp1,temp2,temp3,temp4,temp5))

```


```{r,echo=FALSE}
# combine promoter and enhancers, the code here wouldn't have double counted mutations because of overlaps in utr5 and utr3

temp = generate_mut_enhancer_pair_for_burden_analysis("160229_data_for_analysis.bed","../other_annotation/epigenomic_annotation/Noonan_brain_union.yanyu_pipeline_enhancers.10000.bp_within_TSS.bed")
temp2 = generate_mut_enhancer_pair_for_burden_analysis("160229_data_for_analysis.bed","../other_annotation/epigenomic_annotation/Noonan_brain_union.yanyu_pipeline_enhancers.10000_to_20000.bp_within_TSS.bed")
temp3 = generate_mut_enhancer_pair_for_burden_analysis("160229_data_for_analysis.bed","../other_annotation/epigenomic_annotation/Noonan_brain_union.promoter_1kb_genename.bed")

temp = do.call("rbind",list(temp,temp2,temp3))
temp = temp[!duplicated(temp),]

temp1 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = stringent_ASD)
temp1 = data.frame(lapply(translate_to_plotting_v2(temp1, "Noonan_brain_union", "Promoter+enhancer","stringent_ASD"), function(y) if(is.numeric(y)) round(y, 3) else y)) 

temp2 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = ID_genes)
temp2 = data.frame(lapply(translate_to_plotting_v2(temp2, "Noonan_brain_union", "Promoter+enhancer","ID_genes"), function(y) if(is.numeric(y)) round(y, 3) else y)) 

temp3 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = SCZ_composite)
temp3 = data.frame(lapply(translate_to_plotting_v2(temp3, "Noonan_brain_union", "Promoter+enhancer","SCZ_composite"), function(y) if(is.numeric(y)) round(y, 3) else y)) 

temp4 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = relaxed_ASD)
temp4 = data.frame(lapply(translate_to_plotting_v2(temp4, "Noonan_brain_union", "Promoter+enhancer","NP_genes"), function(y) if(is.numeric(y)) round(y, 3) else y)) 

temp5 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = nonASD_genes)
temp5 = data.frame(lapply(translate_to_plotting_v2(temp5, "Noonan_brain_union", "Promoter+enhancer","nonASD_genes"), function(y) if(is.numeric(y)) round(y, 3) else y))

temp = do.call("rbind",list(temp1,temp2,temp3,temp4,temp5))

plot_summary <- rbind(plot_summary, temp)

```

```{r,echo=FALSE}
# combine promoter and 10kb enhancers, the code here wouldn't have double counted mutations because of overlaps in utr5 and utr3

temp = generate_mut_enhancer_pair_for_burden_analysis("160229_data_for_analysis.bed","../other_annotation/epigenomic_annotation/Noonan_brain_union.yanyu_pipeline_enhancers.10000.bp_within_TSS.bed")

temp3 = generate_mut_enhancer_pair_for_burden_analysis("160229_data_for_analysis.bed","../other_annotation/epigenomic_annotation/Noonan_brain_union.promoter_1kb_genename.bed")

temp = do.call("rbind",list(temp,temp3))
temp = temp[!duplicated(temp),]

temp1 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = stringent_ASD)
temp1 = data.frame(lapply(translate_to_plotting_v2(temp1, "Noonan_brain_union", "Promoter+10kb_enhancer","stringent_ASD"), function(y) if(is.numeric(y)) round(y, 3) else y)) 

temp2 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = ID_genes)
temp2 = data.frame(lapply(translate_to_plotting_v2(temp2, "Noonan_brain_union", "Promoter+10kb_enhancer","ID_genes"), function(y) if(is.numeric(y)) round(y, 3) else y)) 

temp3 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = SCZ_composite)
temp3 = data.frame(lapply(translate_to_plotting_v2(temp3, "Noonan_brain_union", "Promoter+10kb_enhancer","SCZ_composite"), function(y) if(is.numeric(y)) round(y, 3) else y)) 

temp4 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = relaxed_ASD)
temp4 = data.frame(lapply(translate_to_plotting_v2(temp4, "Noonan_brain_union", "Promoter+10kb_enhancer","NP_genes"), function(y) if(is.numeric(y)) round(y, 3) else y)) 

temp5 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = nonASD_genes)
temp5 = data.frame(lapply(translate_to_plotting_v2(temp5, "Noonan_brain_union", "Promoter+10kb_enhancer","nonASD_genes"), function(y) if(is.numeric(y)) round(y, 3) else y))

plot_summary <- do.call("rbind",list(plot_summary, temp1,temp2,temp3,temp4,temp5))


```

```{r,echo=FALSE}
temp = generate_mut_enhancer_pair_for_burden_analysis("160229_data_for_analysis.bed","../other_annotation/epigenomic_annotation/Noonan_brain_union.yanyu_pipeline_enhancers.10000.bp_within_TSS.bed")

temp1 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = stringent_ASD)
temp1 = data.frame(lapply(translate_to_plotting_v2(temp1, "Noonan_brain_union", "1-10Kb","stringent_ASD"), function(y) if(is.numeric(y)) round(y, 3) else y)) 

temp2 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = ID_genes)
temp2 = data.frame(lapply(translate_to_plotting_v2(temp2, "Noonan_brain_union", "1-10Kb","ID_genes"), function(y) if(is.numeric(y)) round(y, 3) else y)) 

temp3 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = SCZ_composite)
temp3 = data.frame(lapply(translate_to_plotting_v2(temp3, "Noonan_brain_union", "1-10Kb","SCZ_composite"), function(y) if(is.numeric(y)) round(y, 3) else y)) 

temp4 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = relaxed_ASD)
temp4 = data.frame(lapply(translate_to_plotting_v2(temp4, "Noonan_brain_union", "1-10Kb","NP_genes"), function(y) if(is.numeric(y)) round(y, 3) else y)) 

temp5 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = nonASD_genes)
temp5 = data.frame(lapply(translate_to_plotting_v2(temp5, "Noonan_brain_union", "1-10Kb","nonASD_genes"), function(y) if(is.numeric(y)) round(y, 3) else y))

temp = do.call("rbind",list(temp1,temp2,temp3,temp4,temp5))


plot_summary <- rbind(plot_summary, temp)
```

```{r,echo=FALSE}
temp = generate_mut_enhancer_pair_for_burden_analysis("160229_data_for_analysis.bed","../other_annotation/epigenomic_annotation/Noonan_brain_union.yanyu_pipeline_enhancers.10000_to_20000.bp_within_TSS.bed")

temp1 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = stringent_ASD)
temp1 = data.frame(lapply(translate_to_plotting_v2(temp1, "Noonan_brain_union", "10-20Kb","stringent_ASD"), function(y) if(is.numeric(y)) round(y, 3) else y)) 

temp2 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = ID_genes)
temp2 = data.frame(lapply(translate_to_plotting_v2(temp2, "Noonan_brain_union", "10-20Kb","ID_genes"), function(y) if(is.numeric(y)) round(y, 3) else y)) 

temp3 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = SCZ_composite)
temp3 = data.frame(lapply(translate_to_plotting_v2(temp3, "Noonan_brain_union", "10-20Kb","SCZ_composite"), function(y) if(is.numeric(y)) round(y, 3) else y)) 

temp4 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = relaxed_ASD)
temp4 = data.frame(lapply(translate_to_plotting_v2(temp4, "Noonan_brain_union", "10-20Kb","NP_genes"), function(y) if(is.numeric(y)) round(y, 3) else y)) 

temp5 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = nonASD_genes)
temp5 = data.frame(lapply(translate_to_plotting_v2(temp5, "Noonan_brain_union", "10-20Kb","nonASD_genes"), function(y) if(is.numeric(y)) round(y, 3) else y))

temp = do.call("rbind",list(temp1,temp2,temp3,temp4,temp5))


plot_summary <- rbind(plot_summary, temp)

```







Brain roadmap adult brain
```{r,echo=FALSE}
temp = generate_mut_enhancer_pair_for_burden_analysis("160229_data_for_analysis.bed","../other_annotation/epigenomic_annotation/Brain_roadmap_union.promoter_1kb_genename.bed")

temp1 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = stringent_ASD)
temp1 = data.frame(lapply(translate_to_plotting_v2(temp1, "Brain_roadmap", "Promoter","stringent_ASD"), function(y) if(is.numeric(y)) round(y, 3) else y)) 

temp2 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = ID_genes)
temp2 = data.frame(lapply(translate_to_plotting_v2(temp2, "Brain_roadmap", "Promoter","ID_genes"), function(y) if(is.numeric(y)) round(y, 3) else y)) 

temp3 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = SCZ_composite)
temp3 = data.frame(lapply(translate_to_plotting_v2(temp3, "Brain_roadmap", "Promoter","SCZ_composite"), function(y) if(is.numeric(y)) round(y, 3) else y)) 

temp4 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = relaxed_ASD)
temp4 = data.frame(lapply(translate_to_plotting_v2(temp4, "Brain_roadmap", "Promoter","NP_genes"), function(y) if(is.numeric(y)) round(y, 3) else y)) 

temp5 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = nonASD_genes)
temp5 = data.frame(lapply(translate_to_plotting_v2(temp5, "Brain_roadmap", "Promoter","nonASD_genes"), function(y) if(is.numeric(y)) round(y, 3) else y))

plot_summary <- do.call("rbind",list(plot_summary, temp1,temp2,temp3,temp4,temp5))

```


```{r,echo=FALSE}
# combine promoter and enhancers, the code here wouldn't have double counted mutations because of overlaps in utr5 and utr3

temp = generate_mut_enhancer_pair_for_burden_analysis("160229_data_for_analysis.bed","../other_annotation/epigenomic_annotation/Brain_roadmap_union.yanyu_pipeline_enhancers.10000.bp_within_TSS.bed")
temp2 = generate_mut_enhancer_pair_for_burden_analysis("160229_data_for_analysis.bed","../other_annotation/epigenomic_annotation/Brain_roadmap_union.yanyu_pipeline_enhancers.10000_to_20000.bp_within_TSS.bed")
temp3 = generate_mut_enhancer_pair_for_burden_analysis("160229_data_for_analysis.bed","../other_annotation/epigenomic_annotation/Brain_roadmap_union.promoter_1kb_genename.bed")

temp = do.call("rbind",list(temp,temp2,temp3))
temp = temp[!duplicated(temp),]

temp1 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = stringent_ASD)
temp1 = data.frame(lapply(translate_to_plotting_v2(temp1, "Brain_roadmap", "Promoter+enhancer","stringent_ASD"), function(y) if(is.numeric(y)) round(y, 3) else y)) 

temp2 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = ID_genes)
temp2 = data.frame(lapply(translate_to_plotting_v2(temp2, "Brain_roadmap", "Promoter+enhancer","ID_genes"), function(y) if(is.numeric(y)) round(y, 3) else y)) 

temp3 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = SCZ_composite)
temp3 = data.frame(lapply(translate_to_plotting_v2(temp3, "Brain_roadmap", "Promoter+enhancer","SCZ_composite"), function(y) if(is.numeric(y)) round(y, 3) else y)) 

temp4 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = relaxed_ASD)
temp4 = data.frame(lapply(translate_to_plotting_v2(temp4, "Brain_roadmap", "Promoter+enhancer","NP_genes"), function(y) if(is.numeric(y)) round(y, 3) else y)) 

temp5 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = nonASD_genes)
temp5 = data.frame(lapply(translate_to_plotting_v2(temp5, "Brain_roadmap", "Promoter+enhancer","nonASD_genes"), function(y) if(is.numeric(y)) round(y, 3) else y))

plot_summary <- do.call("rbind",list(plot_summary, temp1,temp2,temp3,temp4,temp5))

```

```{r,echo=FALSE}
# combine promoter and 10kb enhancers, the code here wouldn't have double counted mutations because of overlaps in utr5 and utr3

temp = generate_mut_enhancer_pair_for_burden_analysis("160229_data_for_analysis.bed","../other_annotation/epigenomic_annotation/Brain_roadmap_union.yanyu_pipeline_enhancers.10000.bp_within_TSS.bed")

temp3 = generate_mut_enhancer_pair_for_burden_analysis("160229_data_for_analysis.bed","../other_annotation/epigenomic_annotation/Brain_roadmap_union.promoter_1kb_genename.bed")

temp = do.call("rbind",list(temp,temp3))
temp = temp[!duplicated(temp),]

temp1 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = stringent_ASD)
temp1 = data.frame(lapply(translate_to_plotting_v2(temp1, "Brain_roadmap", "Promoter+10kb_enhancer","stringent_ASD"), function(y) if(is.numeric(y)) round(y, 3) else y)) 

temp2 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = ID_genes)
temp2 = data.frame(lapply(translate_to_plotting_v2(temp2, "Brain_roadmap", "Promoter+10kb_enhancer","ID_genes"), function(y) if(is.numeric(y)) round(y, 3) else y)) 

temp3 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = SCZ_composite)
temp3 = data.frame(lapply(translate_to_plotting_v2(temp3, "Brain_roadmap", "Promoter+10kb_enhancer","SCZ_composite"), function(y) if(is.numeric(y)) round(y, 3) else y)) 

temp4 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = relaxed_ASD)
temp4 = data.frame(lapply(translate_to_plotting_v2(temp4, "Brain_roadmap", "Promoter+10kb_enhancer","NP_genes"), function(y) if(is.numeric(y)) round(y, 3) else y)) 

temp5 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = nonASD_genes)
temp5 = data.frame(lapply(translate_to_plotting_v2(temp5, "Brain_roadmap", "Promoter+10kb_enhancer","nonASD_genes"), function(y) if(is.numeric(y)) round(y, 3) else y))


plot_summary <- do.call("rbind",list(plot_summary, temp1,temp2,temp3,temp4,temp5))


```


```{r,echo=FALSE}
temp = generate_mut_enhancer_pair_for_burden_analysis("160229_data_for_analysis.bed","../other_annotation/epigenomic_annotation/Brain_roadmap_union.yanyu_pipeline_enhancers.10000.bp_within_TSS.bed")

temp1 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = stringent_ASD)
temp1 = data.frame(lapply(translate_to_plotting_v2(temp1, "Brain_roadmap", "1-10Kb","stringent_ASD"), function(y) if(is.numeric(y)) round(y, 3) else y)) 

temp2 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = ID_genes)
temp2 = data.frame(lapply(translate_to_plotting_v2(temp2, "Brain_roadmap", "1-10Kb","ID_genes"), function(y) if(is.numeric(y)) round(y, 3) else y)) 

temp3 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = SCZ_composite)
temp3 = data.frame(lapply(translate_to_plotting_v2(temp3, "Brain_roadmap", "1-10Kb","SCZ_composite"), function(y) if(is.numeric(y)) round(y, 3) else y)) 

temp4 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = relaxed_ASD)
temp4 = data.frame(lapply(translate_to_plotting_v2(temp4, "Brain_roadmap", "1-10Kb","NP_genes"), function(y) if(is.numeric(y)) round(y, 3) else y)) 

temp5 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = nonASD_genes)
temp5 = data.frame(lapply(translate_to_plotting_v2(temp5, "Brain_roadmap", "1-10Kb","nonASD_genes"), function(y) if(is.numeric(y)) round(y, 3) else y))


plot_summary <- do.call("rbind",list(plot_summary, temp1,temp2,temp3,temp4,temp5))
```

```{r,echo=FALSE}
temp = generate_mut_enhancer_pair_for_burden_analysis("160229_data_for_analysis.bed","../other_annotation/epigenomic_annotation/Brain_roadmap_union.yanyu_pipeline_enhancers.10000_to_20000.bp_within_TSS.bed")

temp1 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = stringent_ASD)
temp1 = data.frame(lapply(translate_to_plotting_v2(temp1, "Brain_roadmap", "10-20Kb","stringent_ASD"), function(y) if(is.numeric(y)) round(y, 3) else y)) 

temp2 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = ID_genes)
temp2 = data.frame(lapply(translate_to_plotting_v2(temp2, "Brain_roadmap", "10-20Kb","ID_genes"), function(y) if(is.numeric(y)) round(y, 3) else y)) 

temp3 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = SCZ_composite)
temp3 = data.frame(lapply(translate_to_plotting_v2(temp3, "Brain_roadmap", "10-20Kb","SCZ_composite"), function(y) if(is.numeric(y)) round(y, 3) else y)) 

temp4 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = relaxed_ASD)
temp4 = data.frame(lapply(translate_to_plotting_v2(temp4, "Brain_roadmap", "10-20Kb","NP_genes"), function(y) if(is.numeric(y)) round(y, 3) else y)) 

temp5 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = nonASD_genes)
temp5 = data.frame(lapply(translate_to_plotting_v2(temp5, "Brain_roadmap", "10-20Kb","nonASD_genes"), function(y) if(is.numeric(y)) round(y, 3) else y))

plot_summary <- do.call("rbind",list(plot_summary, temp1,temp2,temp3,temp4,temp5))

```

Noonan + Brain roadmap union
```{r,echo=FALSE}
temp = generate_mut_enhancer_pair_for_burden_analysis("160229_data_for_analysis.bed","../other_annotation/epigenomic_annotation/Noonan_brain_roadmap_union.promoter_1kb_genename.bed")

temp1 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = stringent_ASD)
temp1 = data.frame(lapply(translate_to_plotting_v2(temp1, "Noonan_roadmap_union", "Promoter","stringent_ASD"), function(y) if(is.numeric(y)) round(y, 3) else y)) 

temp2 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = ID_genes)
temp2 = data.frame(lapply(translate_to_plotting_v2(temp2, "Noonan_roadmap_union", "Promoter","ID_genes"), function(y) if(is.numeric(y)) round(y, 3) else y)) 

temp3 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = SCZ_composite)
temp3 = data.frame(lapply(translate_to_plotting_v2(temp3, "Noonan_roadmap_union", "Promoter","SCZ_composite"), function(y) if(is.numeric(y)) round(y, 3) else y)) 

temp4 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = relaxed_ASD)
temp4 = data.frame(lapply(translate_to_plotting_v2(temp4, "Noonan_roadmap_union", "Promoter","NP_genes"), function(y) if(is.numeric(y)) round(y, 3) else y)) 

temp5 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = nonASD_genes)
temp5 = data.frame(lapply(translate_to_plotting_v2(temp5, "Noonan_roadmap_union", "Promoter","nonASD_genes"), function(y) if(is.numeric(y)) round(y, 3) else y))

plot_summary <- do.call("rbind",list(plot_summary, temp1,temp2,temp3,temp4,temp5))

```


```{r,echo=FALSE}
# combine promoter and enhancers, the code here wouldn't have double counted mutations because of overlaps in utr5 and utr3

temp = generate_mut_enhancer_pair_for_burden_analysis("160229_data_for_analysis.bed","../other_annotation/epigenomic_annotation/Noonan_brain_roadmap_union.yanyu_pipeline_enhancers.10000.bp_within_TSS.bed")
temp2 = generate_mut_enhancer_pair_for_burden_analysis("160229_data_for_analysis.bed","../other_annotation/epigenomic_annotation/Noonan_brain_roadmap_union.yanyu_pipeline_enhancers.10000_to_20000.bp_within_TSS.bed")
temp3 = generate_mut_enhancer_pair_for_burden_analysis("160229_data_for_analysis.bed","../other_annotation/epigenomic_annotation/Noonan_brain_roadmap_union.promoter_1kb_genename.bed")

temp = do.call("rbind",list(temp,temp2,temp3))
temp = temp[!duplicated(temp),]

temp1 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = stringent_ASD)
temp1 = data.frame(lapply(translate_to_plotting_v2(temp1, "Noonan_roadmap_union", "Promoter+enhancer","stringent_ASD"), function(y) if(is.numeric(y)) round(y, 3) else y)) 

temp2 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = ID_genes)
temp2 = data.frame(lapply(translate_to_plotting_v2(temp2, "Noonan_roadmap_union", "Promoter+enhancer","ID_genes"), function(y) if(is.numeric(y)) round(y, 3) else y)) 

temp3 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = SCZ_composite)
temp3 = data.frame(lapply(translate_to_plotting_v2(temp3, "Noonan_roadmap_union", "Promoter+enhancer","SCZ_composite"), function(y) if(is.numeric(y)) round(y, 3) else y)) 

temp4 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = relaxed_ASD)
temp4 = data.frame(lapply(translate_to_plotting_v2(temp4, "Noonan_roadmap_union", "Promoter+enhancer","NP_genes"), function(y) if(is.numeric(y)) round(y, 3) else y)) 

temp5 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = nonASD_genes)
temp5 = data.frame(lapply(translate_to_plotting_v2(temp5, "Noonan_roadmap_union", "Promoter+enhancer","nonASD_genes"), function(y) if(is.numeric(y)) round(y, 3) else y))

plot_summary <- do.call("rbind",list(plot_summary, temp1,temp2,temp3,temp4,temp5))

```

```{r,echo=FALSE}
# combine promoter and 10kb enhancers, the code here wouldn't have double counted mutations because of overlaps in utr5 and utr3

temp = generate_mut_enhancer_pair_for_burden_analysis("160229_data_for_analysis.bed","../other_annotation/epigenomic_annotation/Noonan_brain_roadmap_union.yanyu_pipeline_enhancers.10000.bp_within_TSS.bed")

temp3 = generate_mut_enhancer_pair_for_burden_analysis("160229_data_for_analysis.bed","../other_annotation/epigenomic_annotation/Noonan_brain_roadmap_union.promoter_1kb_genename.bed")

temp = do.call("rbind",list(temp,temp3))
temp = temp[!duplicated(temp),]

temp1 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = stringent_ASD)
temp1 = data.frame(lapply(translate_to_plotting_v2(temp1, "Noonan_roadmap_union", "Promoter+10kb_enhancer","stringent_ASD"), function(y) if(is.numeric(y)) round(y, 3) else y)) 

temp2 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = ID_genes)
temp2 = data.frame(lapply(translate_to_plotting_v2(temp2, "Noonan_roadmap_union", "Promoter+10kb_enhancer","ID_genes"), function(y) if(is.numeric(y)) round(y, 3) else y)) 

temp3 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = SCZ_composite)
temp3 = data.frame(lapply(translate_to_plotting_v2(temp3, "Noonan_roadmap_union", "Promoter+10kb_enhancer","SCZ_composite"), function(y) if(is.numeric(y)) round(y, 3) else y)) 

temp4 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = relaxed_ASD)
temp4 = data.frame(lapply(translate_to_plotting_v2(temp4, "Noonan_roadmap_union", "Promoter+10kb_enhancer","NP_genes"), function(y) if(is.numeric(y)) round(y, 3) else y)) 

temp5 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = nonASD_genes)
temp5 = data.frame(lapply(translate_to_plotting_v2(temp5, "Noonan_roadmap_union", "Promoter+10kb_enhancer","nonASD_genes"), function(y) if(is.numeric(y)) round(y, 3) else y))


plot_summary <- do.call("rbind",list(plot_summary, temp1,temp2,temp3,temp4,temp5))


```


```{r,echo=FALSE}
temp = generate_mut_enhancer_pair_for_burden_analysis("160229_data_for_analysis.bed","../other_annotation/epigenomic_annotation/Noonan_brain_roadmap_union.yanyu_pipeline_enhancers.10000.bp_within_TSS.bed")

temp1 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = stringent_ASD)
temp1 = data.frame(lapply(translate_to_plotting_v2(temp1, "Noonan_roadmap_union", "1-10Kb","stringent_ASD"), function(y) if(is.numeric(y)) round(y, 3) else y)) 

temp2 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = ID_genes)
temp2 = data.frame(lapply(translate_to_plotting_v2(temp2, "Noonan_roadmap_union", "1-10Kb","ID_genes"), function(y) if(is.numeric(y)) round(y, 3) else y)) 

temp3 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = SCZ_composite)
temp3 = data.frame(lapply(translate_to_plotting_v2(temp3, "Noonan_roadmap_union", "1-10Kb","SCZ_composite"), function(y) if(is.numeric(y)) round(y, 3) else y)) 

temp4 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = relaxed_ASD)
temp4 = data.frame(lapply(translate_to_plotting_v2(temp4, "Noonan_roadmap_union", "1-10Kb","NP_genes"), function(y) if(is.numeric(y)) round(y, 3) else y)) 

temp5 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = nonASD_genes)
temp5 = data.frame(lapply(translate_to_plotting_v2(temp5, "Noonan_roadmap_union", "1-10Kb","nonASD_genes"), function(y) if(is.numeric(y)) round(y, 3) else y))


plot_summary <- do.call("rbind",list(plot_summary, temp1,temp2,temp3,temp4,temp5))
```

```{r,echo=FALSE}
temp = generate_mut_enhancer_pair_for_burden_analysis("160229_data_for_analysis.bed","../other_annotation/epigenomic_annotation/Noonan_brain_roadmap_union.yanyu_pipeline_enhancers.10000_to_20000.bp_within_TSS.bed")

temp1 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = stringent_ASD)
temp1 = data.frame(lapply(translate_to_plotting_v2(temp1, "Noonan_roadmap_union", "10-20Kb","stringent_ASD"), function(y) if(is.numeric(y)) round(y, 3) else y)) 

temp2 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = ID_genes)
temp2 = data.frame(lapply(translate_to_plotting_v2(temp2, "Noonan_roadmap_union", "10-20Kb","ID_genes"), function(y) if(is.numeric(y)) round(y, 3) else y)) 

temp3 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = SCZ_composite)
temp3 = data.frame(lapply(translate_to_plotting_v2(temp3, "Noonan_roadmap_union", "10-20Kb","SCZ_composite"), function(y) if(is.numeric(y)) round(y, 3) else y)) 

temp4 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = relaxed_ASD)
temp4 = data.frame(lapply(translate_to_plotting_v2(temp4, "Noonan_roadmap_union", "10-20Kb","NP_genes"), function(y) if(is.numeric(y)) round(y, 3) else y)) 

temp5 = show_burden_for_selected_features_with_geneset(temp,input_type = "table",ref = "self", geneset = nonASD_genes)
temp5 = data.frame(lapply(translate_to_plotting_v2(temp5, "Noonan_roadmap_union", "10-20Kb","nonASD_genes"), function(y) if(is.numeric(y)) round(y, 3) else y))

plot_summary <- do.call("rbind",list(plot_summary, temp1,temp2,temp3,temp4,temp5))

```



```{r}
plot_summary = plot_summary[!is.na(plot_summary$gene_list),]
plot_summary = plot_summary[!duplicated(plot_summary),]
plot_summary = plot_summary[plot_summary$gene_list == "NP_genes" | plot_summary$gene_list == "nonASD_genes",]
plot_summary = plot_summary[plot_summary$region_tag == "Promoter" | plot_summary$region_tag == "Promoter+10kb_enhancer" | plot_summary$region_tag == "1-10Kb",]
plot_summary$gene_list = factor(plot_summary$gene_list, levels = c("NP_genes","nonASD_genes"))


```


```{r}
#scaling factor for plotting, which is the burden of baseline. Used for 160229 data only
scaling = 0.73
```

```{r,echo=FALSE,fig.width=9, fig.height=5}
score_map <- list()
score_map[['GERP']] <- 'GERP'
score_map[['Eigen']] <- 'Eigen'
score_map[['PhyloP']] <- 'PhyloP'
score_map[['CADD13_PHRED']] <- 'CADD'
score_map[['Motif']] <- 'Motif_noonan_50'

library(ggplot2)
for(sc in unique(plot_summary$score)){
        sub_sum <- plot_summary[plot_summary$score == sc, ]
        for(i in unique(sub_sum$cutoff)){
                sub_sum_tmp <- sub_sum[sub_sum$cutoff == i, ]
                
                p <- ggplot(sub_sum_tmp) + geom_tile(aes(x = as.factor(bed), y = as.factor(region_tag), fill = (asd_sub+1)/(control_sub+baseline_control/baseline_ASD)/(baseline_ASD/baseline_control))) + scale_fill_gradient2(midpoint = 1, na.value = 'black', limits = c(0, 2))
                p <- p + facet_grid(. ~ gene_list)
                newdat <- ggplot_build(p)$data[[1]] 
                x <- (newdat$xmin + newdat$xmax) / 2
                y <- (newdat$ymin + newdat$ymax) / 2
                deltax <- (newdat$xmax[1] - newdat$xmin[1]) / 2
                deltay <- (newdat$ymax[1] - newdat$ymin[1]) / 2
                p <- p + geom_text(aes(x, y, label=paste(asd_sub, control_sub, round(as.numeric(as.character(burden)), digits = 2), pvalue, sep = '\n')), col="red", size = 2.5)
                tmp <- sub_sum_tmp
                t <- order(tmp$bed)
                tmp <- tmp[t, ]
                t <- order(tmp$region_tag)
                tmp <- tmp[t, ]
                t <- order(tmp$gene_list)
                tmp <- tmp[t, ]
                ps <- as.numeric(as.character(tmp$pvalue))
                t1 <- ps <= 0.05
                t3 <- ps > 0.05 & ps <= 0.1
                t2 <- ps > 0.1
                ps[t1] <- 1.5
                ps[t2] <- 0
                ps[t3] <- 0.5
                #p <- p + geom_rect(aes(xmin = x - deltax, xmax = x + deltax, ymin = y - deltay, ymax = y + deltay),
                                  # fill = "transparent", color = "red", size = ps)
                if(sub_sum_tmp[1, 'score'] == 'all'){
                        strings <- 'No Cutoff'
                }else{
                        strings <- paste(score_map[[as.character(sub_sum_tmp[1, 'score'])]], as.character(sub_sum_tmp[1, 'cutoff']), sep = ' > ')
                }
                #                 for(i in 1 : length(x)){
                #                         xi <- x[i]
                #                         yi <- y[i]
                #                         pi <- ps[i]
                #                         
                #                         if(pi <= 0.05){
                #                                 print(c(pi,xi,yi))
                #                                 p <- p + geom_rect(aes(xmin = as.factor(xi - deltax), xmax = as.factor(xi + deltax), ymin = as.factor(yi - deltay), ymax = as.factor(yi + deltay)),
                #                fill = "transparent", color = "red", size = 1.5)
                #                         }
                #                 }
                p <- p + ggtitle(strings) + 
                        labs(x="Experimental Data",y="Regulatory Region", fill = 'Burden Trend') + 
                        theme(plot.title = element_text(face="bold", size=20)) +
                        theme(axis.title = element_text(face="bold", size=15)) +
                        theme(axis.text.x = element_text(angle = 45, hjust = 1))
                print(p)
                # line <- readline()
        }
}
```

```{r plot_dhs, fig.width=10, fig.height=8, echo=FALSE, eval=FALSE}
### extract and plot NP_gene result in GERP > 2 and CADD > 10
strings <- 'Neuropsychiatric Mutations'
sub <- plot_summary[((plot_summary$score == 'GERP' & plot_summary$cutoff == 2) | plot_summary$score == 'all' | (plot_summary$score == 'CADD13_PHRED' & plot_summary$cutoff == 10)) & plot_summary$gene_list == 'NP_gene', ]
sub$score <- as.character(sub$score)
sub$score[sub$score == 'GERP'] <- 'GERP > 2'
sub$score[sub$score == 'CADD13_PHRED'] <- 'CADD > 10'
p <- ggplot(sub) + geom_tile(aes(x = as.factor(bed), y = as.factor(region_tag), fill = ((asd_sub+1)/(control_sub+1))/baseline)) + scale_fill_gradient2(midpoint = 1, na.value = 'black', limits = c(0, 2))
p <- p + facet_grid(. ~ score)
p <- p + ggtitle(strings) + 
        labs(x="Experimental Data",y="Regulatory Region", fill = 'Burden Trend') + 
        theme(plot.title = element_text(face="bold", size=20)) +
        theme(axis.title = element_text(face="bold", size=15)) +
        theme(axis.text.x = element_text(angle = 45, hjust = 1))
newdat <- ggplot_build(p)$data[[1]] 
x <- (newdat$xmin + newdat$xmax) / 2
y <- (newdat$ymin + newdat$ymax) / 2
deltax <- (newdat$xmax[1] - newdat$xmin[1]) / 2
deltay <- (newdat$ymax[1] - newdat$ymin[1]) / 2
p <- p + geom_text(aes(x, y, label=paste(asd_sub, control_sub, round(as.numeric(as.character(burden)), digits = 2), pvalue, sep = '\n')), col="red", size = 2)
tmp <- sub
t <- order(tmp$bed)
tmp <- tmp[t, ]
t <- order(tmp$region_tag)
tmp <- tmp[t, ]
t <- order(tmp$score)
tmp <- tmp[t, ]
ps <- as.numeric(as.character(tmp$pvalue))
t1 <- ps <= 0.05
t3 <- ps > 0.05 & ps <= 0.1
t2 <- ps > 0.1
ps[t1] <- 1.5
ps[t2] <- 0
ps[t3] <- 0.5
#p <- p + geom_rect(aes(xmin = x - deltax, xmax = x + deltax, ymin = y - deltay, ymax = y + deltay), fill = "transparent", color = "red", size = ps)
p
### 
```

