---
title: "Untitled"
author: "Yuwen"
date: "November 20, 2016"
output: pdf_document
---

Study the effect of promoter, coding, and cg content on mutation rate.
For promoters and coding windows, remove those that have "n/a" as gene names, for coding also remove utrs
ALso have Noonan Roadmap brain union as a categorical variable
```{r,cache=TRUE}
source("../lib/161117_glm_for_mutation_count.R")
# remember only use SNV data
# what is different from 161120_mutation_rate_calibration_1.R is that for coding regions, I removed all utrs and exons without a gene name (or
# the genename is n/a)

# also used Noonan brain roadmap H3K27ac as a dummy variable. 

new_control = new.env()
load("../data/debug_region_list_073116_using_258_control_data_matrix.Rdata", new_control)
write.table(new_control$data_matrix[new_control$data_matrix$Prediction == "dnv_sibling" & new_control$data_matrix$Type == "SNV",c(1,2,3,7)], "temp.bed", col.names = FALSE, row.names = FALSE, sep = "\t", quote = FALSE)
control_model_258_estimate = estimate_effect_size("./temp.bed","../other_annotation/epigenomic_annotation/Whole_genome.promoter_yanyu_10kb_exons_no_utr_no_na_genes.50bp_window.bed","../other_annotation/epigenomic_annotation/Whole_genome.promoter_yanyu_10kb_exons_no_utr_no_na_genes.50bp_window.bed.fasta.cg","../other_annotation/epigenomic_annotation/Whole_genome.promoter_yanyu_10kb_exons_no_utr_no_na_genes.50bp_window.bed.mutrate",258, "../other_annotation/epigenomic_annotation/Noonan_brain_roadmap_union.bed",1e-9,"../other_annotation/epigenomic_annotation/Whole_genome.promoter_yanyu_10kb_exons_no_utr_no_na_genes.50bp_window.bed_genename.bed","../other_annotation/gene_list/TADA_SNV_CNV_combined_Feb7_tadaname_combined_qvalue.txt")
```

```{r}
control_model_258_estimate
```