
##### combine promoter and enhancer together for estimating relative risk
##### epi features and annotation features were not used in mutation rate adjustment
##### estimate the relative risk of Noonan Brain H3K27ac and Roadmap Brain H3K27ac separately.

```{r}
source("../lib/161117_glm_for_mutation_count.R")
```

##### top 6% TADA genes for fitting the 50bp-window model: estimate relative risk

```{r, cache=TRUE, dependson=verified_effect_size_estimate_noncoding_mutations_v3}
# remember only use SNV data
# what is different from 161120_mutation_rate_calibration_1.R is that for coding regions, I removed all utrs and exons without a gene name (or
# the genename is n/a)

# also used Noonan brain roadmap H3K27ac as a dummy variable. 
# when predict the effect of coding mutation rate for ASD cases, only use synonymous mutations. 

# Three epigenomic annotation (Noonan and Roadmap Brain H3k27ac, ENCODE DHS, and Roadmap DHS) and one conservation score(phastcons100way) were used to correct mutation rate and to estimate relative risks. 


# only use top 10% TADA genes for estimating relative risks
new_control = new.env()
load("../data/debug_region_list_073116_using_258_control_data_matrix.Rdata", new_control)

prefix <-system("date +%s", intern = TRUE) # prefix for temporary files that will be deleted at the end of the pipeline

write.table(new_control$data_matrix[new_control$data_matrix$Prediction == "dnv_proband" & new_control$data_matrix$Type == "SNV",c(1,2,3,7)], paste(prefix,"temp.bed",sep = ""), col.names = FALSE, row.names = FALSE, sep = "\t", quote = FALSE)
write.table(new_control$data_matrix[new_control$data_matrix$Prediction == "dnv_proband" ,c(1:5,8,7)], paste(prefix,"annovar_input_temp.bed",sep = ""), col.names = FALSE, row.names = FALSE, sep = "\t", quote = FALSE)

output_report <- verified_effect_size_estimate_noncoding_mutations_v3(mut_file = paste(prefix,"temp.bed",sep = ""), 
                                                                            window_file = "../other_annotation/epigenomic_annotation/Whole_genome.promoter_yanyu_10kb_exons_no_utr_no_na_genes.50bp_window.bed",
                                                                            cg_file = "../other_annotation/epigenomic_annotation/Whole_genome.promoter_yanyu_10kb_exons_no_utr_no_na_genes.50bp_window.bed.fasta.cg",
                                                                            mutrate_file = "../other_annotation/epigenomic_annotation/Whole_genome.promoter_yanyu_10kb_exons_no_utr_no_na_genes.50bp_window.bed.mutrate",
                                                                            sample_size = 314,
                                                                            epigenomic_marks_list = c("../other_annotation/epigenomic_annotation/Noonan_union.bed","../other_annotation/epigenomic_annotation/brain_roadmap_union.bed","../other_annotation/epigenomic_annotation/Epigenome_E081_E082_intersection.bed","../other_annotation/epigenomic_annotation/encode_DHS_union.bed"),
                                                                            overlap = 1e-9,
                                                                            gene_assign_file = "../other_annotation/epigenomic_annotation/Whole_genome.promoter_yanyu_10kb_exons_no_utr_no_na_genes.50bp_window.bed_genename.bed",
                                                                            rm_nonsyn = TRUE,
                                                                            annovar_folder = "/media/yuwen/Elements/ANNOVAR/annovar/",
                                                                            annovar_input = paste(prefix,"annovar_input_temp.bed",sep = ""),
                                                                            gene_prior_file = "../other_annotation/gene_list/TADA_SNV_CNV_combined_Feb7_tadaname_null_PPA_p0_0.94.txt",
                                                                            sequence_annotation_list = c("../other_annotation/epigenomic_annotation/Whole_genome.promoter_yanyu_10kb_exons_no_utr_no_na_genes.50bp_window.bed.phastcons100way"),
                                                                            optimization_prop = 0.06)



rm(new_control)
system(paste("rm ",prefix,"annovar_input_temp.bed",sep = ""))
system(paste("rm ",prefix, "temp.bed",sep = ""))
```


##### top 6% TADA genes for fitting the 50bp-window model: estimate relative risk
```{r,echo=FALSE}
fisher_info<-solve(-output_report$mle$hessian)
prop_sigma<-sqrt(diag(fisher_info))
upper<-output_report$mle$par+1.96*prop_sigma
lower<-output_report$mle$par-1.96*prop_sigma
a <- data.frame(relative_risk = output_report$mle$par, lower_bound = lower, upper_bournd = upper)
rownames(a) <- c("Noonan H3K27ac","Roadmap Brain H3K27ac", "Roadmap brain DHS", "ENCODE brain DHS","Phastcons100way")
knitr::kable(a)
```
##### top 6% TADA genes for fitting the 50bp-window model: parameters for adjusting mutation rates

```{r,echo=FALSE}
knitr::kable(output_report$regression_coef)
```


##### top all genes for fitting the 50bp-window model: estimate relative risk

```{r, cache=TRUE, dependson=verified_effect_size_estimate_noncoding_mutations_v3}
# remember only use SNV data
# what is different from 161120_mutation_rate_calibration_1.R is that for coding regions, I removed all utrs and exons without a gene name (or
# the genename is n/a)

# also used Noonan brain roadmap H3K27ac as a dummy variable. 
# when predict the effect of coding mutation rate for ASD cases, only use synonymous mutations. 

# Three epigenomic annotation (Noonan and Roadmap Brain H3k27ac, ENCODE DHS, and Roadmap DHS) and one conservation score(phastcons100way) were used to correct mutation rate and to estimate relative risks. 


# only use top 10% TADA genes for estimating relative risks
new_control = new.env()
load("../data/debug_region_list_073116_using_258_control_data_matrix.Rdata", new_control)

prefix <-system("date +%s", intern = TRUE) # prefix for temporary files that will be deleted at the end of the pipeline

write.table(new_control$data_matrix[new_control$data_matrix$Prediction == "dnv_proband" & new_control$data_matrix$Type == "SNV",c(1,2,3,7)], paste(prefix,"temp.bed",sep = ""), col.names = FALSE, row.names = FALSE, sep = "\t", quote = FALSE)
write.table(new_control$data_matrix[new_control$data_matrix$Prediction == "dnv_proband" ,c(1:5,8,7)], paste(prefix,"annovar_input_temp.bed",sep = ""), col.names = FALSE, row.names = FALSE, sep = "\t", quote = FALSE)

output_report <- verified_effect_size_estimate_noncoding_mutations_v3(mut_file = paste(prefix,"temp.bed",sep = ""), 
                                                                            window_file = "../other_annotation/epigenomic_annotation/Whole_genome.promoter_yanyu_10kb_exons_no_utr_no_na_genes.50bp_window.bed",
                                                                            cg_file = "../other_annotation/epigenomic_annotation/Whole_genome.promoter_yanyu_10kb_exons_no_utr_no_na_genes.50bp_window.bed.fasta.cg",
                                                                            mutrate_file = "../other_annotation/epigenomic_annotation/Whole_genome.promoter_yanyu_10kb_exons_no_utr_no_na_genes.50bp_window.bed.mutrate",
                                                                            sample_size = 314,
                                                                            epigenomic_marks_list = c("../other_annotation/epigenomic_annotation/Noonan_union.bed","../other_annotation/epigenomic_annotation/brain_roadmap_union.bed","../other_annotation/epigenomic_annotation/Epigenome_E081_E082_intersection.bed","../other_annotation/epigenomic_annotation/encode_DHS_union.bed"),
                                                                            overlap = 1e-9,
                                                                            gene_assign_file = "../other_annotation/epigenomic_annotation/Whole_genome.promoter_yanyu_10kb_exons_no_utr_no_na_genes.50bp_window.bed_genename.bed",
                                                                            rm_nonsyn = TRUE,
                                                                            annovar_folder = "/media/yuwen/Elements/ANNOVAR/annovar/",
                                                                            annovar_input = paste(prefix,"annovar_input_temp.bed",sep = ""),
                                                                            gene_prior_file = "../other_annotation/gene_list/TADA_SNV_CNV_combined_Feb7_tadaname_null_PPA_p0_0.94.txt",
                                                                            sequence_annotation_list = c("../other_annotation/epigenomic_annotation/Whole_genome.promoter_yanyu_10kb_exons_no_utr_no_na_genes.50bp_window.bed.phastcons100way"),
                                                                            optimization_prop = 1)



rm(new_control)
system(paste("rm ",prefix,"annovar_input_temp.bed",sep = ""))
system(paste("rm ",prefix, "temp.bed",sep = ""))
```


##### top 1% TADA genes for fitting the 50bp-window model: estimate relative risk
```{r,echo=FALSE}
fisher_info<-solve(-output_report$mle$hessian)
prop_sigma<-sqrt(diag(fisher_info))
upper<-output_report$mle$par+1.96*prop_sigma
lower<-output_report$mle$par-1.96*prop_sigma
a <- data.frame(relative_risk = output_report$mle$par, lower_bound = lower, upper_bournd = upper)
rownames(a) <- c("Noonan H3K27ac","Roadmap Brain H3K27ac", "Roadmap brain DHS", "ENCODE brain DHS","Phastcons100way")
knitr::kable(a)
```
##### top 1% TADA genes for fitting the 50bp-window model: parameters for adjusting mutation rates

```{r,echo=FALSE}
knitr::kable(output_report$regression_coef)
```