

```{r}
set.seed(100)
library(data.table)
library(parallel)
source("../lib/170122_base_level_simulation_functions.R")

```

***
**NOTE**

1. First adjust for mutation rate for each 50bp window, then apply the the adjusting factor for each base so we could get the base-level adjusted mutation rate. When adjusting for mutation rates, only consider features including promoter, coding, within-10kb non-coding, and 50bp cg contents.
2. Only consider the relative risk from active promoters and enhancers (union), marked by Noonan and Roadmap brain H3k27ac. 
3. Only use bases from top 6% TADA genes for now.
4. The model itself could be easily adapted to cases where more epigenomic features or functional annotations are included. 


***

##### First try a toy model generating a dataset with 10000base/gene, 500 genes, effect size 2
##### 
```{r, cache=TRUE, cache.extra = estimate_effect_size_for_simulation_data_mixture_with_categorization_v2, cache.extra = mutation_simulator}
# mutation_simulator to only have 1 epigegnomic feature added, the prior is 0.3 for each gene
mutation_count <- mutation_simulator(gene_number = 500,window_number=10000,feature_prob = 0.2,base_rate = 2e-6,sample_size = 314, p_risk = 0.3, effect_size = 2)
data <- mutation_count$data
sample_size <- mutation_count$sample_size
# transform so that the data could be applied to estimate_effect_size_for_simulation_data_mixture_with_categorization_v2
data <- mutation_count$data[,c("gene","window", "mut_count","base_mut_rate", "feature_indicator")]
colnames(data) <- c("genename","base_ID", "mut_count","adjusted_base_mutrate", "epi")

# make a prior propability file with each gene has prior probability of 0.3 (1-0.3 = 0.7)
prefix=system("date +%s", intern = TRUE) # prefix for temporary files that will be deleted at the end of the pipeline
prior <- data.frame(genename = seq(1,500), prior = 1-0.3)
write.table(prior, paste(prefix, "_temp_prior_for_simulation.txt", sep = ""), col.names = FALSE, row.names = FALSE, sep = "\t", quote = FALSE)
# in the data, feature is the 5th column, but it would become the 6th column after having prior probability column included. So it is 6
simulation_estimate <- estimate_effect_size_for_simulation_data_mixture_with_categorization_v2(data = data, feature_start = 6, feature_end = 6, feature_number = 1, gene_prior_file = paste(prefix, "_temp_prior_for_simulation.txt", sep = ""), sample_size = 314)
rm(data)
rm(mutation_count)
system(paste("rm ", paste(prefix, "_temp_prior_for_simulation.txt", sep = ""), sep = ""))
```

```{r, dependson = simulation_estimate}
simulation_estimate
```


##### estimate relative risk from top 6% TADA genes, afeter adjusting for mutation rates

```{r, cache=TRUE, cache.extra = adjusted_base_level_mutrate, cache.extra = estimate_effect_size_for_simulation_data_mixture_with_categorization_v2}
new_control = new.env()
load("../data/debug_region_list_073116_using_258_control_data_matrix.Rdata", new_control)

prefix <-system("date +%s", intern = TRUE) # prefix for temporary files that will be deleted at the end of the pipeline

write.table(new_control$data_matrix[new_control$data_matrix$Prediction == "dnv_proband" & new_control$data_matrix$Type == "SNV",c(1,2,3,7)], paste(prefix,"temp.bed",sep = ""), col.names = FALSE, row.names = FALSE, sep = "\t", quote = FALSE)
write.table(new_control$data_matrix[new_control$data_matrix$Prediction == "dnv_proband" ,c(1:5,8,7)], paste(prefix,"annovar_input_temp.bed",sep = ""), col.names = FALSE, row.names = FALSE, sep = "\t", quote = FALSE)

# first adjuste for mutation rates at 50-bp window level and then apply the scaling factor to each base of noncoding regions of selected top percentage of TADA genes.

preprocessing_time <- system.time(adjusted_base_level_mutrate <- adjust_mutation_rate_window_to_base(mut_file = paste(prefix,"temp.bed",sep = ""),
window_file = "../other_annotation/epigenomic_annotation/Whole_genome.promoter_yanyu_10kb_exons_no_utr_no_na_genes.50bp_window.bed",
cg_file = "../other_annotation/epigenomic_annotation/Whole_genome.promoter_yanyu_10kb_exons_no_utr_no_na_genes.50bp_window.bed.fasta.cg",
mutrate_file = "../other_annotation/epigenomic_annotation/Whole_genome.promoter_yanyu_10kb_exons_no_utr_no_na_genes.50bp_window.bed.mutrate",
sample_size = 314,
epigenomic_marks = "../other_annotation/epigenomic_annotation/Noonan_brain_roadmap_union.bed",
overlap = 0.5,
rm_nonsyn = FALSE,
annovar_folder = "/media/yuwen/Elements/ANNOVAR/annovar/",
annovar_input = "no.txt",
gene_assign_file = "../other_annotation/epigenomic_annotation/Whole_genome.promoter_yanyu_10kb_exons_no_utr_no_na_genes.50bp_window.bed_genename.bed",
gene_prior_file = "../other_annotation/gene_list/TADA_SNV_CNV_combined_Feb7_tadaname_null_PPA_p0_0.94.txt",
report_proportion = 0.06,
mutrate_ref_file = "/media/yuwen/Elements/mutation_rate_raw/MarkDaly/Daly_mutrate.bw",
node_n = 6))

# transfomr mutation table
adjusted_base_level_mutrate <- adjusted_base_level_mutrate[,c("genename","base_ID","mut_count","adjusted_base_mutrate","epi")]

simulation_estimate <- estimate_effect_size_for_simulation_data_mixture_with_categorization_v2(data = adjusted_base_level_mutrate, feature_start = 6, feature_end = 6, feature_number = 1, gene_prior_file = "../other_annotation/gene_list/TADA_SNV_CNV_combined_Feb7_tadaname_null_PPA_p0_0.94.txt", sample_size = 314)

rm(new_control)
system(paste("rm ",prefix,"annovar_input_temp.bed",sep = ""))
system(paste("rm ",prefix, "temp.bed",sep = ""))
rm(adjusted_base_level_mutrate)
```

```{r, dependson=simulation_estimate}
simulation_estimate
fisher_info<-solve(-simulation_estimate$mle$hessian)
prop_sigma<-sqrt(diag(fisher_info))
upper<-simulation_estimate$mle$par+1.96*prop_sigma
lower<-simulation_estimate$mle$par-1.96*prop_sigma
a = data.frame(relative_risk = simulation_estimate$mle$par, lower_bound = lower, upper_bournd = upper)
rownames(a) = c("Noonan+Roadmap Brain H3K27ac enhancer + promoter")
knitr::kable(a)
```