---
title: "Estimate the relative risks of splicing mutations along with other epigenomic features"
author: "Yuwen Liu"
date: "January 12, 2017"
output: html_document
---

***
**NOTE**

The coding mutation rate in the glm model is adjusted from synonymous mutations only from ASD cases. So after fitting the model, the 50bp coding window would represent the synonymous mutation rate. It is hard to get a splicing mutation rate for each window (it requires to get all the 50bp windows that are within certain distance of intron-exon junctions. These regions are where Xiong et al., gave splicing disruption score (SPIDEX) to each base). For now, **I will assume the splicing mutation rate for each gene is proportional to its synonymous mutation rate, and this proportion is the same for the genes**. 

I could use the ratio between spilicing mutations and synonymous mutations in cases to roughly estimate the splicing mutation rate for each gene. But this tends to be overestimating splicing mutation rate if splicing mutaitons increase the risk of ASD. 

Alternatively, I would assume that the proportion is insensitive to sequencing studies. i.e., I am able to use the proportion that is estimated from control studies. This assumption is very likely to hold as batch effects may mostly affect the proportion of observed mutations in coding regions. But within coding regions, the bias by batch effect may be very small for detecting splicing mutations.  

***


```{r}
source("../lib/161117_glm_for_mutation_count.R")
```

```{r, cache=TRUE, dependson=verified_effect_size_estimate_noncoding_mutations_v4}
# remember only use SNV data
# what is different from 161120_mutation_rate_calibration_1.R is that for coding regions, I removed all utrs and exons without a gene name (or
# the genename is n/a)

# also used Noonan brain roadmap H3K27ac as a dummy variable. 
# when predict the effect of coding mutation rate for ASD cases, only use synonymous mutations. 

# Three epigenomic annotation (Noonan and Roadmap Brain H3k27ac, ENCODE DHS, and Roadmap DHS) and one conservation score(phastcons100way) were used to correct mutation rate and to estimate relative risks. 



# only use top 6% TADA genes for estimating relative risks
new_control = new.env()
load("../data/debug_region_list_073116_data_matrixtransformed_for_old_code.Rdata", new_control)

prefix <-system("date +%s", intern = TRUE) # prefix for temporary files that will be deleted at the end of the pipeline

write.table(new_control$data_matrix[new_control$data_matrix$Prediction == "dnv_proband" & new_control$data_matrix$Type == "SNV",c(1,2,3,7)], paste(prefix,"temp.bed",sep = ""), col.names = FALSE, row.names = FALSE, sep = "\t", quote = FALSE)
write.table(new_control$data_matrix[new_control$data_matrix$Prediction == "dnv_proband" ,c(1:5,8,7)], paste(prefix,"annovar_input_temp.bed",sep = ""), col.names = FALSE, row.names = FALSE, sep = "\t", quote = FALSE)

proband_SNV_mut_spedix <- merge(new_control$data_matrix[new_control$data_matrix$Prediction == "dnv_proband" & new_control$data_matrix$Type == "SNV",c(1,2,3,7)], new_control$mut_spedix, by.x =  "ID", by.y = "index")

output_report <- verified_effect_size_estimate_noncoding_mutations_v4(mut_file = paste(prefix,"temp.bed",sep = ""), 
                                                                            window_file = "../other_annotation/epigenomic_annotation/Whole_genome.promoter_yanyu_10kb_exons_no_utr_no_na_genes.50bp_window.bed",
                                                                            cg_file = "../other_annotation/epigenomic_annotation/Whole_genome.promoter_yanyu_10kb_exons_no_utr_no_na_genes.50bp_window.bed.fasta.cg",
                                                                            mutrate_file = "../other_annotation/epigenomic_annotation/Whole_genome.promoter_yanyu_10kb_exons_no_utr_no_na_genes.50bp_window.bed.mutrate",
                                                                            sample_size = 314,
                                                                            epigenomic_marks_list = c("../other_annotation/epigenomic_annotation/Noonan_brain_roadmap_union.bed"),
                                                                            overlap = 1e-9,
                                                                            gene_assign_file = "../other_annotation/epigenomic_annotation/Whole_genome.promoter_yanyu_10kb_exons_no_utr_no_na_genes.50bp_window.bed_genename.bed",
                                                                            rm_nonsyn = TRUE,
                                                                            annovar_folder = "/media/yuwen/Elements/ANNOVAR/annovar/",
                                                                            annovar_input = paste(prefix,"annovar_input_temp.bed",sep = ""),
                                                                            gene_prior_file = "../other_annotation/gene_list/TADA_SNV_CNV_combined_Feb7_tadaname_null_PPA_p0_0.94.txt",
                                                                            sequence_annotation_list = c("../other_annotation/epigenomic_annotation/Whole_genome.promoter_yanyu_10kb_exons_no_utr_no_na_genes.50bp_window.bed.phastcons100way"),
                                                                            optimization_prop = 0.06,
                                                                      mut_spidex = proband_SNV_mut_spedix)



rm(new_control)
system(paste("rm ",prefix,"annovar_input_temp.bed",sep = ""))
system(paste("rm ",prefix, "temp.bed",sep = ""))
```


##### top 6% TADA genes for fitting the 50bp-window model: estimate relative risk (Noonan_brain_roadmap_union, phastcons100way, splicing)

```{r,echo=FALSE}
fisher_info<-solve(-output_report$mle$hessian)
prop_sigma<-sqrt(diag(fisher_info))
upper<-output_report$mle$par+1.96*prop_sigma
lower<-output_report$mle$par-1.96*prop_sigma
a = data.frame(relative_risk = output_report$mle$par, lower_bound = lower, upper_bournd = upper)
rownames(a) = c("Noonan+Roadmap_brain_H3K27ac","Phastcons100way","splicing")
knitr::kable(a)
```

##### top 6% TADA genes for fitting the 50bp-window model: parameters for adjusting mutation rates

```{r,echo=FALSE}
knitr::kable(output_report$regression_coef)
```

```{r}
output_report$mle
```

```{r,, eval=FALSE, echo=FALSE}
prefix=system("date +%s", intern = TRUE)
mut_file = paste(prefix,"temp.bed",sep = "")
window_file = "../other_annotation/epigenomic_annotation/Whole_genome.promoter_yanyu_10kb_exons_no_utr_no_na_genes.50bp_window.bed"
cg_file = "../other_annotation/epigenomic_annotation/Whole_genome.promoter_yanyu_10kb_exons_no_utr_no_na_genes.50bp_window.bed.fasta.cg"
mutrate_file = "../other_annotation/epigenomic_annotation/Whole_genome.promoter_yanyu_10kb_exons_no_utr_no_na_genes.50bp_window.bed.mutrate"
sample_size = 314
epigenomic_marks_list = c("../other_annotation/epigenomic_annotation/Noonan_brain_roadmap_union.bed")
overlap = 1e-9
gene_assign_file = "../other_annotation/epigenomic_annotation/Whole_genome.promoter_yanyu_10kb_exons_no_utr_no_na_genes.50bp_window.bed_genename.bed"
rm_nonsyn = TRUE
annovar_folder = "/media/yuwen/Elements/ANNOVAR/annovar/"
annovar_input = paste(prefix,"annovar_input_temp.bed",sep = "")
gene_prior_file = "../other_annotation/gene_list/TADA_SNV_CNV_combined_Feb7_tadaname_null_PPA_p0_0.94.txt"
sequence_annotation_list = c("../other_annotation/epigenomic_annotation/Whole_genome.promoter_yanyu_10kb_exons_no_utr_no_na_genes.50bp_window.bed.phastcons100way")
optimization_prop = 0.06
mut_spidex = proband_SNV_mut_spedix # merge from proband mutation table and 
```