### Workflow to incoporate splicing mutations to our model

***

1. Identify refseq intron regions that don't overlap existing 50-bp windows, and partition those regions into 50-bp windows and combine these windows with the previous windows. Then fit a glm model to adjust for mutatin rate for each window, recoreded the adjusting effect for each window. 

2. Identify the cutoff that is going to be used to define splicing mutations, then in `spidex_public_noncommercial_v1_0.tab`, only retain the lines that satisfy the cutoff setting. 

3. bedtools to link data from Step1 to data from Step2.

4. Get allele-specific mutation rate for each base that has spedix scores `spidex_public_noncommercial_v1_0_collapse_unique_base_merged_extended_for_tri_extract_changed_by_YL.bed.fasta.trinuc.mutrate`.

5. bedtools to link data from step3 to data from step4.

6. scale mutation rate get the splicing mutation rate at each base (with non-0 splicing mutation rate) sfor each gene.