<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title></title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/united.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-1.1/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-1.1/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">ASD Project</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="meeting.html">Meeting</a>
</li>
<li>
  <a href="Note.html">Note</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">




</div>


<pre class="r"><code>library(data.table)
library(parallel)
source(&quot;../lib/170122_base_level_simulation_functions.R&quot;)</code></pre>
<pre><code>## 
## Attaching package: &#39;reshape&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:data.table&#39;:
## 
##     melt</code></pre>
<hr />
<p><strong>NOTE</strong></p>
<ol style="list-style-type: decimal">
<li>First adjust for mutation rate for each 50bp window, then apply the the adjusting factor for each base so we could get the base-level adjusted mutation rate. When adjusting for mutation rates, only consider features including promoter, coding, within-10kb non-coding, and 50bp cg contents.</li>
<li>Only consider the relative risk from active promoters and enhancers (union), marked by Noonan and Roadmap brain H3k27ac.</li>
<li>Only use bases from top 6% TADA genes for now.</li>
<li>The model itself could be easily adapted to cases where more epigenomic features or functional annotations are included.</li>
<li>The output will be R object which store base-level information in a compact mode</li>
<li>100way phylop and gerp annotation is added, brain h3k27ac, fetal DHS and adult DHS.</li>
</ol>
<hr />
<div id="estimate-relative-risk-from-top-0.5-tada-genes-afeter-adjusting-for-mutation-rates" class="section level1">
<h1><span class="header-section-number">1</span> <strong>Estimate relative risk from top 0.5% TADA genes, afeter adjusting for mutation rates</strong></h1>
<pre class="r"><code>new_control = new.env()
load(&quot;../data/debug_region_list_073116_using_258_control_data_matrix.Rdata&quot;, new_control)

prefix &lt;-system(&quot;date +%s&quot;, intern = TRUE) # prefix for temporary files that will be deleted at the end of the pipeline

write.table(new_control$data_matrix[new_control$data_matrix$Prediction == &quot;dnv_proband&quot; &amp; new_control$data_matrix$Type == &quot;SNV&quot;,c(1,2,3,7)], paste(prefix,&quot;temp.bed&quot;,sep = &quot;&quot;), col.names = FALSE, row.names = FALSE, sep = &quot;\t&quot;, quote = FALSE)
write.table(new_control$data_matrix[new_control$data_matrix$Prediction == &quot;dnv_proband&quot; ,c(1:5,8,7)], paste(prefix,&quot;annovar_input_temp.bed&quot;,sep = &quot;&quot;), col.names = FALSE, row.names = FALSE, sep = &quot;\t&quot;, quote = FALSE)

# first adjuste for mutation rates at 50-bp window level and then apply the scaling factor to each base of noncoding regions of selected top percentage of TADA genes.

preprocessing_time &lt;- system.time(adjusted_base_level_rr &lt;- adjust_mutation_rate_window_to_base_compact_more_feature_v2(mut_file = paste(prefix,&quot;temp.bed&quot;,sep = &quot;&quot;),
window_file = &quot;../other_annotation/epigenomic_annotation/Whole_genome.promoter_yanyu_10kb_exons_no_utr_no_na_genes.50bp_window.bed&quot;,
cg_file = &quot;../other_annotation/epigenomic_annotation/Whole_genome.promoter_yanyu_10kb_exons_no_utr_no_na_genes.50bp_window.bed.fasta.cg&quot;,
mutrate_file = &quot;../other_annotation/epigenomic_annotation/Whole_genome.promoter_yanyu_10kb_exons_no_utr_no_na_genes.50bp_window.bed.mutrate&quot;,
sample_size = 314,
epigenomic_marks = c(&quot;../other_annotation/epigenomic_annotation/Noonan_brain_roadmap_union.bed&quot;,&quot;../other_annotation/epigenomic_annotation/Epigenome_E081_E082_intersection.bed&quot;,&quot;../other_annotation/epigenomic_annotation/encode_DHS_union.bed&quot;),
sequence_annotation = c(&quot;phylop_100way&quot;, &quot;gerp&quot;), 
sequence_annotation_cutoff = list(phylop_100way=2, gerp =2 ), 
sequence_annotation_ref = list(phylop_100way = &quot;../other_annotation/conservation_annotation/hg19.100way.phyloP100way.bw&quot;, gerp = &quot;../other_annotation/conservation_annotation/hg19_gerp_score.bw&quot;),
overlap = 0.5,
rm_nonsyn = FALSE,
annovar_folder = &quot;/media/yuwen/Elements/ANNOVAR/annovar/&quot;,
annovar_input = &quot;no.txt&quot;,
gene_assign_file = &quot;../other_annotation/epigenomic_annotation/Whole_genome.promoter_yanyu_10kb_exons_no_utr_no_na_genes.50bp_window.bed_genename.bed&quot;,
gene_prior_file = &quot;../other_annotation/gene_list/TADA_SNV_CNV_combined_Feb7_tadaname_null_PPA_p0_0.94.txt&quot;,
report_proportion = 1,
mutrate_ref_file = &quot;/media/yuwen/Elements/mutation_rate_raw/MarkDaly/Daly_mutrate.bw&quot;,
node_n = 6,
feature_start = 5,
feature_end = 9,
feature_number = 5, 
chunk_partition_num =20))</code></pre>
<pre><code>## 
Read 12.2% of 7061846 rows
Read 18.3% of 7061846 rows
Read 22.9% of 7061846 rows
Read 28.0% of 7061846 rows
Read 33.7% of 7061846 rows
Read 39.9% of 7061846 rows
Read 46.9% of 7061846 rows
Read 54.4% of 7061846 rows
Read 58.5% of 7061846 rows
Read 67.3% of 7061846 rows
Read 75.1% of 7061846 rows
Read 80.9% of 7061846 rows
Read 87.7% of 7061846 rows
Read 93.3% of 7061846 rows
Read 99.4% of 7061846 rows
Read 7061846 rows and 8 (of 8) columns from 0.337 GB file in 00:00:21</code></pre>
<pre><code>## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.

## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.</code></pre>
<pre><code>## Warning in `[.data.table`(coverage, , `:=`(c(&quot;coding&quot;, &quot;promoter&quot;, &quot;nf&quot;), :
## with=FALSE ignored, it isn&#39;t needed when using :=. See ?&#39;:=&#39; for examples.</code></pre>
<pre><code>## 
Read 38.2% of 7061846 rows
Read 62.7% of 7061846 rows
Read 84.3% of 7061846 rows
Read 7061846 rows and 6 (of 6) columns from 0.342 GB file in 00:00:05</code></pre>
<pre><code>## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.</code></pre>
<pre><code>## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.</code></pre>
<pre><code>## 
Read 45.3% of 15595717 rows
Read 73.0% of 15595717 rows
Read 15595717 rows and 2 (of 2) columns from 0.399 GB file in 00:00:04
## 
Read 24.7% of 15595717 rows
Read 43.8% of 15595717 rows
Read 63.0% of 15595717 rows
Read 82.5% of 15595717 rows
Read 15595717 rows and 8 (of 8) columns from 0.803 GB file in 00:00:06
## 
Read 2.0% of 15595717 rows
Read 24.6% of 15595717 rows
Read 47.7% of 15595717 rows
Read 70.8% of 15595717 rows
Read 88.4% of 15595717 rows
Read 15595717 rows and 8 (of 8) columns from 0.803 GB file in 00:00:07
## 
Read 0.0% of 15595717 rows
Read 22.6% of 15595717 rows
Read 44.8% of 15595717 rows
Read 66.0% of 15595717 rows
Read 87.6% of 15595717 rows
Read 15595717 rows and 8 (of 8) columns from 0.803 GB file in 00:00:07
## 
Read 30.0% of 15595717 rows
Read 51.7% of 15595717 rows
Read 73.4% of 15595717 rows
Read 95.2% of 15595717 rows
Read 15595717 rows and 8 (of 8) columns from 0.803 GB file in 00:00:06
## 
Read 47.1% of 15595717 rows
Read 76.6% of 15595717 rows
Read 15595717 rows and 2 (of 2) columns from 0.317 GB file in 00:00:04
## 
Read 63.5% of 15595717 rows
Read 99.4% of 15595717 rows
Read 15595717 rows and 2 (of 2) columns from 0.307 GB file in 00:00:04
## 
Read 53.1% of 15582498 rows
Read 83.6% of 15582498 rows
Read 15582498 rows and 2 (of 2) columns from 0.400 GB file in 00:00:04
## 
Read 26.0% of 15582498 rows
Read 47.7% of 15582498 rows
Read 68.6% of 15582498 rows
Read 89.9% of 15582498 rows
Read 15582498 rows and 8 (of 8) columns from 0.805 GB file in 00:00:06
## 
Read 26.4% of 15582498 rows
Read 47.9% of 15582498 rows
Read 70.0% of 15582498 rows
Read 93.1% of 15582498 rows
Read 15582498 rows and 8 (of 8) columns from 0.805 GB file in 00:00:06
## 
Read 27.5% of 15582498 rows
Read 51.1% of 15582498 rows
Read 74.7% of 15582498 rows
Read 98.1% of 15582498 rows
Read 15582498 rows and 8 (of 8) columns from 0.805 GB file in 00:00:06
## 
Read 18.9% of 15582498 rows
Read 40.8% of 15582498 rows
Read 63.0% of 15582498 rows
Read 85.5% of 15582498 rows
Read 15582498 rows and 8 (of 8) columns from 0.805 GB file in 00:00:06
## 
Read 3.0% of 15582498 rows
Read 38.0% of 15582498 rows
Read 75.0% of 15582498 rows
Read 15582498 rows and 2 (of 2) columns from 0.317 GB file in 00:00:05
## 
Read 57.8% of 15582498 rows
Read 94.5% of 15582498 rows
Read 15582498 rows and 2 (of 2) columns from 0.308 GB file in 00:00:04
## 
Read 40.0% of 15587094 rows
Read 68.2% of 15587094 rows
Read 97.1% of 15587094 rows
Read 15587094 rows and 2 (of 2) columns from 0.401 GB file in 00:00:05
## 
Read 15.4% of 15587094 rows
Read 38.1% of 15587094 rows
Read 61.1% of 15587094 rows
Read 84.4% of 15587094 rows
Read 15587094 rows and 8 (of 8) columns from 0.806 GB file in 00:00:06
## 
Read 20.5% of 15587094 rows
Read 43.0% of 15587094 rows
Read 64.8% of 15587094 rows
Read 87.1% of 15587094 rows
Read 15587094 rows and 8 (of 8) columns from 0.806 GB file in 00:00:06
## 
Read 33.5% of 15587094 rows
Read 57.9% of 15587094 rows
Read 80.8% of 15587094 rows
Read 15587094 rows and 8 (of 8) columns from 0.806 GB file in 00:00:05
## 
Read 30.7% of 15587094 rows
Read 55.0% of 15587094 rows
Read 79.0% of 15587094 rows
Read 15587094 rows and 8 (of 8) columns from 0.806 GB file in 00:00:05
## 
Read 63.8% of 15587094 rows
Read 15587094 rows and 2 (of 2) columns from 0.318 GB file in 00:00:03
## 
Read 71.8% of 15587094 rows
Read 15587094 rows and 2 (of 2) columns from 0.309 GB file in 00:00:03
## 
Read 48.1% of 15585976 rows
Read 75.6% of 15585976 rows
Read 15585976 rows and 2 (of 2) columns from 0.402 GB file in 00:00:04
## 
Read 31.8% of 15585976 rows
Read 54.5% of 15585976 rows
Read 78.3% of 15585976 rows
Read 15585976 rows and 8 (of 8) columns from 0.808 GB file in 00:00:05
## 
Read 0.0% of 15585976 rows
Read 24.7% of 15585976 rows
Read 48.8% of 15585976 rows
Read 73.8% of 15585976 rows
Read 15585976 rows and 8 (of 8) columns from 0.808 GB file in 00:00:07
## 
Read 31.0% of 15585976 rows
Read 55.2% of 15585976 rows
Read 79.6% of 15585976 rows
Read 15585976 rows and 8 (of 8) columns from 0.808 GB file in 00:00:05
## 
Read 33.4% of 15585976 rows
Read 57.4% of 15585976 rows
Read 81.4% of 15585976 rows
Read 15585976 rows and 8 (of 8) columns from 0.808 GB file in 00:00:05
## 
Read 57.4% of 15585976 rows
Read 91.9% of 15585976 rows
Read 15585976 rows and 2 (of 2) columns from 0.320 GB file in 00:00:04
## 
Read 65.4% of 15585976 rows
Read 15585976 rows and 2 (of 2) columns from 0.310 GB file in 00:00:03
## 
Read 43.4% of 15585264 rows
Read 72.1% of 15585264 rows
Read 15585264 rows and 2 (of 2) columns from 0.402 GB file in 00:00:04
## 
Read 0.0% of 15585264 rows
Read 26.9% of 15585264 rows
Read 49.0% of 15585264 rows
Read 71.4% of 15585264 rows
Read 91.9% of 15585264 rows
Read 15585264 rows and 8 (of 8) columns from 0.808 GB file in 00:00:07
## 
Read 24.8% of 15585264 rows
Read 47.8% of 15585264 rows
Read 71.3% of 15585264 rows
Read 93.0% of 15585264 rows
Read 15585264 rows and 8 (of 8) columns from 0.808 GB file in 00:00:06
## 
Read 27.7% of 15585264 rows
Read 51.8% of 15585264 rows
Read 75.9% of 15585264 rows
Read 100.0% of 15585264 rows
Read 15585264 rows and 8 (of 8) columns from 0.808 GB file in 00:00:06
## 
Read 25.5% of 15585264 rows
Read 49.6% of 15585264 rows
Read 73.8% of 15585264 rows
Read 98.2% of 15585264 rows
Read 15585264 rows and 8 (of 8) columns from 0.808 GB file in 00:00:06
## 
Read 61.1% of 15585264 rows
Read 99.1% of 15585264 rows
Read 15585264 rows and 2 (of 2) columns from 0.320 GB file in 00:00:04
## 
Read 64.1% of 15585264 rows
Read 15585264 rows and 2 (of 2) columns from 0.310 GB file in 00:00:03
## 
Read 49.1% of 15586076 rows
Read 78.6% of 15586076 rows
Read 15586076 rows and 2 (of 2) columns from 0.403 GB file in 00:00:04
## 
Read 0.0% of 15586076 rows
Read 24.8% of 15586076 rows
Read 49.9% of 15586076 rows
Read 74.5% of 15586076 rows
Read 97.4% of 15586076 rows
Read 15586076 rows and 8 (of 8) columns from 0.809 GB file in 00:00:07
## 
Read 42.2% of 15586076 rows
Read 66.0% of 15586076 rows
Read 91.7% of 15586076 rows
Read 15586076 rows and 8 (of 8) columns from 0.809 GB file in 00:00:05
## 
Read 0.0% of 15586076 rows
Read 24.2% of 15586076 rows
Read 50.2% of 15586076 rows
Read 73.7% of 15586076 rows
Read 98.0% of 15586076 rows
Read 15586076 rows and 8 (of 8) columns from 0.809 GB file in 00:00:07
## 
Read 33.6% of 15586076 rows
Read 56.5% of 15586076 rows
Read 79.9% of 15586076 rows
Read 15586076 rows and 8 (of 8) columns from 0.809 GB file in 00:00:05
## 
Read 60.5% of 15586076 rows
Read 97.5% of 15586076 rows
Read 15586076 rows and 2 (of 2) columns from 0.321 GB file in 00:00:04
## 
Read 73.7% of 15586076 rows
Read 15586076 rows and 2 (of 2) columns from 0.311 GB file in 00:00:03
## 
Read 49.8% of 15576312 rows
Read 78.8% of 15576312 rows
Read 15576312 rows and 2 (of 2) columns from 0.400 GB file in 00:00:04
## 
Read 0.0% of 15576312 rows
Read 24.4% of 15576312 rows
Read 48.7% of 15576312 rows
Read 73.8% of 15576312 rows
Read 98.8% of 15576312 rows
Read 15576312 rows and 8 (of 8) columns from 0.804 GB file in 00:00:07
## 
Read 42.2% of 15576312 rows
Read 68.9% of 15576312 rows
Read 95.7% of 15576312 rows
Read 15576312 rows and 8 (of 8) columns from 0.804 GB file in 00:00:05
## 
Read 38.9% of 15576312 rows
Read 63.1% of 15576312 rows
Read 86.9% of 15576312 rows
Read 15576312 rows and 8 (of 8) columns from 0.804 GB file in 00:00:05
## 
Read 25.0% of 15576312 rows
Read 48.4% of 15576312 rows
Read 71.8% of 15576312 rows
Read 95.5% of 15576312 rows
Read 15576312 rows and 8 (of 8) columns from 0.804 GB file in 00:00:06
## 
Read 67.3% of 15576312 rows
Read 15576312 rows and 2 (of 2) columns from 0.318 GB file in 00:00:03
## 
Read 14.5% of 15576312 rows
Read 55.2% of 15576312 rows
Read 96.4% of 15576312 rows
Read 15576312 rows and 2 (of 2) columns from 0.308 GB file in 00:00:05
## 
Read 51.8% of 15588644 rows
Read 82.9% of 15588644 rows
Read 15588644 rows and 2 (of 2) columns from 0.400 GB file in 00:00:04
## 
Read 0.0% of 15588644 rows
Read 23.9% of 15588644 rows
Read 45.9% of 15588644 rows
Read 68.3% of 15588644 rows
Read 90.5% of 15588644 rows
Read 15588644 rows and 8 (of 8) columns from 0.806 GB file in 00:00:07
## 
Read 26.4% of 15588644 rows
Read 48.3% of 15588644 rows
Read 70.4% of 15588644 rows
Read 92.8% of 15588644 rows
Read 15588644 rows and 8 (of 8) columns from 0.806 GB file in 00:00:06
## 
Read 27.4% of 15588644 rows
Read 50.0% of 15588644 rows
Read 72.8% of 15588644 rows
Read 96.0% of 15588644 rows
Read 15588644 rows and 8 (of 8) columns from 0.806 GB file in 00:00:06
## 
Read 26.1% of 15588644 rows
Read 47.5% of 15588644 rows
Read 69.0% of 15588644 rows
Read 90.5% of 15588644 rows
Read 15588644 rows and 8 (of 8) columns from 0.806 GB file in 00:00:06
## 
Read 64.1% of 15588644 rows
Read 15588644 rows and 2 (of 2) columns from 0.318 GB file in 00:00:03
## 
Read 69.5% of 15588644 rows
Read 15588644 rows and 2 (of 2) columns from 0.308 GB file in 00:00:03
## 
Read 49.8% of 15589549 rows
Read 79.1% of 15589549 rows
Read 15589549 rows and 2 (of 2) columns from 0.400 GB file in 00:00:04
## 
Read 24.3% of 15589549 rows
Read 47.4% of 15589549 rows
Read 69.8% of 15589549 rows
Read 91.7% of 15589549 rows
Read 15589549 rows and 8 (of 8) columns from 0.806 GB file in 00:00:06
## 
Read 0.0% of 15589549 rows
Read 23.0% of 15589549 rows
Read 46.3% of 15589549 rows
Read 69.7% of 15589549 rows
Read 93.4% of 15589549 rows
Read 15589549 rows and 8 (of 8) columns from 0.806 GB file in 00:00:08
## 
Read 31.2% of 15589549 rows
Read 52.9% of 15589549 rows
Read 76.5% of 15589549 rows
Read 99.9% of 15589549 rows
Read 15589549 rows and 8 (of 8) columns from 0.806 GB file in 00:00:06
## 
Read 29.2% of 15589549 rows
Read 52.9% of 15589549 rows
Read 76.7% of 15589549 rows
Read 15589549 rows and 8 (of 8) columns from 0.806 GB file in 00:00:05
## 
Read 64.9% of 15589549 rows
Read 15589549 rows and 2 (of 2) columns from 0.318 GB file in 00:00:03
## 
Read 63.7% of 15589549 rows
Read 15589549 rows and 2 (of 2) columns from 0.308 GB file in 00:00:03
## 
Read 42.4% of 15589059 rows
Read 67.4% of 15589059 rows
Read 91.5% of 15589059 rows
Read 15589059 rows and 2 (of 2) columns from 0.401 GB file in 00:00:05
## 
Read 32.7% of 15589059 rows
Read 49.6% of 15589059 rows
Read 62.6% of 15589059 rows
Read 80.1% of 15589059 rows
Read 15589059 rows and 8 (of 8) columns from 0.807 GB file in 00:00:06
## 
Read 20.1% of 15589059 rows
Read 39.8% of 15589059 rows
Read 60.6% of 15589059 rows
Read 82.0% of 15589059 rows
Read 15589059 rows and 8 (of 8) columns from 0.807 GB file in 00:00:06
## 
Read 21.2% of 15589059 rows
Read 41.9% of 15589059 rows
Read 63.1% of 15589059 rows
Read 84.0% of 15589059 rows
Read 15589059 rows and 8 (of 8) columns from 0.807 GB file in 00:00:06
## 
Read 27.2% of 15589059 rows
Read 50.4% of 15589059 rows
Read 74.2% of 15589059 rows
Read 98.4% of 15589059 rows
Read 15589059 rows and 8 (of 8) columns from 0.807 GB file in 00:00:06
## 
Read 70.6% of 15589059 rows
Read 15589059 rows and 2 (of 2) columns from 0.318 GB file in 00:00:03
## 
Read 58.1% of 15589059 rows
Read 93.8% of 15589059 rows
Read 15589059 rows and 2 (of 2) columns from 0.309 GB file in 00:00:04
## 
Read 56.1% of 15589361 rows
Read 87.6% of 15589361 rows
Read 15589361 rows and 2 (of 2) columns from 0.401 GB file in 00:00:04
## 
Read 0.0% of 15589361 rows
Read 23.5% of 15589361 rows
Read 48.2% of 15589361 rows
Read 71.3% of 15589361 rows
Read 97.1% of 15589361 rows
Read 15589361 rows and 8 (of 8) columns from 0.807 GB file in 00:00:07
## 
Read 28.7% of 15589361 rows
Read 52.7% of 15589361 rows
Read 76.7% of 15589361 rows
Read 15589361 rows and 8 (of 8) columns from 0.807 GB file in 00:00:05
## 
Read 32.9% of 15589361 rows
Read 56.8% of 15589361 rows
Read 80.6% of 15589361 rows
Read 15589361 rows and 8 (of 8) columns from 0.807 GB file in 00:00:05
## 
Read 26.9% of 15589361 rows
Read 48.5% of 15589361 rows
Read 70.8% of 15589361 rows
Read 92.0% of 15589361 rows
Read 15589361 rows and 8 (of 8) columns from 0.807 GB file in 00:00:06
## 
Read 70.6% of 15589361 rows
Read 15589361 rows and 2 (of 2) columns from 0.319 GB file in 00:00:03
## 
Read 68.5% of 15589361 rows
Read 15589361 rows and 2 (of 2) columns from 0.309 GB file in 00:00:03
## 
Read 52.0% of 15585223 rows
Read 82.3% of 15585223 rows
Read 15585223 rows and 2 (of 2) columns from 0.401 GB file in 00:00:04
## 
Read 0.0% of 15585223 rows
Read 22.5% of 15585223 rows
Read 43.1% of 15585223 rows
Read 65.8% of 15585223 rows
Read 89.1% of 15585223 rows
Read 15585223 rows and 8 (of 8) columns from 0.807 GB file in 00:00:07
## 
Read 29.1% of 15585223 rows
Read 53.1% of 15585223 rows
Read 77.1% of 15585223 rows
Read 15585223 rows and 8 (of 8) columns from 0.807 GB file in 00:00:05
## 
Read 42.7% of 15585223 rows
Read 64.7% of 15585223 rows
Read 88.0% of 15585223 rows
Read 15585223 rows and 8 (of 8) columns from 0.807 GB file in 00:00:05
## 
Read 29.4% of 15585223 rows
Read 53.1% of 15585223 rows
Read 77.1% of 15585223 rows
Read 15585223 rows and 8 (of 8) columns from 0.807 GB file in 00:00:05
## 
Read 70.6% of 15585223 rows
Read 15585223 rows and 2 (of 2) columns from 0.319 GB file in 00:00:03
## 
Read 74.1% of 15585223 rows
Read 15585223 rows and 2 (of 2) columns from 0.309 GB file in 00:00:03
## 
Read 43.9% of 15588721 rows
Read 68.6% of 15588721 rows
Read 96.1% of 15588721 rows
Read 15588721 rows and 2 (of 2) columns from 0.400 GB file in 00:00:05
## 
Read 27.1% of 15588721 rows
Read 50.3% of 15588721 rows
Read 72.5% of 15588721 rows
Read 93.3% of 15588721 rows
Read 15588721 rows and 8 (of 8) columns from 0.804 GB file in 00:00:06
## 
Read 0.0% of 15588721 rows
Read 24.8% of 15588721 rows
Read 46.6% of 15588721 rows
Read 67.5% of 15588721 rows
Read 88.5% of 15588721 rows
Read 15588721 rows and 8 (of 8) columns from 0.804 GB file in 00:00:08
## 
Read 26.8% of 15588721 rows
Read 49.9% of 15588721 rows
Read 73.4% of 15588721 rows
Read 96.2% of 15588721 rows
Read 15588721 rows and 8 (of 8) columns from 0.804 GB file in 00:00:06
## 
Read 24.2% of 15588721 rows
Read 47.1% of 15588721 rows
Read 70.0% of 15588721 rows
Read 93.0% of 15588721 rows
Read 15588721 rows and 8 (of 8) columns from 0.804 GB file in 00:00:06
## 
Read 57.4% of 15588721 rows
Read 91.4% of 15588721 rows
Read 15588721 rows and 2 (of 2) columns from 0.317 GB file in 00:00:04
## 
Read 58.7% of 15588721 rows
Read 92.9% of 15588721 rows
Read 15588721 rows and 2 (of 2) columns from 0.307 GB file in 00:00:04
## 
Read 32.0% of 15586341 rows
Read 54.3% of 15586341 rows
Read 78.1% of 15586341 rows
Read 15586341 rows and 2 (of 2) columns from 0.398 GB file in 00:00:05
## 
Read 10.9% of 15586341 rows
Read 27.0% of 15586341 rows
Read 43.2% of 15586341 rows
Read 62.2% of 15586341 rows
Read 81.7% of 15586341 rows
Read 15586341 rows and 8 (of 8) columns from 0.804 GB file in 00:00:07
## 
Read 19.8% of 15586341 rows
Read 38.0% of 15586341 rows
Read 57.7% of 15586341 rows
Read 77.9% of 15586341 rows
Read 97.1% of 15586341 rows
Read 15586341 rows and 8 (of 8) columns from 0.804 GB file in 00:00:07
## 
Read 35.4% of 15586341 rows
Read 59.0% of 15586341 rows
Read 83.1% of 15586341 rows
Read 15586341 rows and 8 (of 8) columns from 0.804 GB file in 00:00:05
## 
Read 38.2% of 15586341 rows
Read 62.4% of 15586341 rows
Read 87.4% of 15586341 rows
Read 15586341 rows and 8 (of 8) columns from 0.804 GB file in 00:00:05
## 
Read 56.5% of 15586341 rows
Read 91.8% of 15586341 rows
Read 15586341 rows and 2 (of 2) columns from 0.316 GB file in 00:00:04
## 
Read 63.8% of 15586341 rows
Read 15586341 rows and 2 (of 2) columns from 0.306 GB file in 00:00:03
## 
Read 35.0% of 15587116 rows
Read 56.6% of 15587116 rows
Read 79.8% of 15587116 rows
Read 15587116 rows and 2 (of 2) columns from 0.400 GB file in 00:00:05
## 
Read 0.0% of 15587116 rows
Read 25.0% of 15587116 rows
Read 48.1% of 15587116 rows
Read 69.7% of 15587116 rows
Read 92.9% of 15587116 rows
Read 15587116 rows and 8 (of 8) columns from 0.806 GB file in 00:00:09
## 
Read 0.0% of 15587116 rows
Read 25.7% of 15587116 rows
Read 52.2% of 15587116 rows
Read 75.4% of 15587116 rows
Read 15587116 rows and 8 (of 8) columns from 0.806 GB file in 00:00:07
## 
Read 33.0% of 15587116 rows
Read 56.3% of 15587116 rows
Read 78.7% of 15587116 rows
Read 15587116 rows and 8 (of 8) columns from 0.806 GB file in 00:00:05
## 
Read 14.5% of 15587116 rows
Read 36.6% of 15587116 rows
Read 59.0% of 15587116 rows
Read 81.4% of 15587116 rows
Read 15587116 rows and 8 (of 8) columns from 0.806 GB file in 00:00:06
## 
Read 70.0% of 15587116 rows
Read 15587116 rows and 2 (of 2) columns from 0.318 GB file in 00:00:03
## 
Read 66.9% of 15587116 rows
Read 15587116 rows and 2 (of 2) columns from 0.308 GB file in 00:00:03
## 
Written 46.3% of 15592216 rows in 2 secs using 8 threads. anyBufferGrown=no; maxBuffUsed=47%. Finished in 2 secs.      
Written 61.3% of 15592216 rows in 3 secs using 8 threads. anyBufferGrown=no; maxBuffUsed=49%. Finished in 1 secs.      
                                                                                                                                     

Read 45.6% of 15592216 rows
Read 74.0% of 15592216 rows
Read 15592216 rows and 2 (of 2) columns from 0.400 GB file in 00:00:04
## 
Read 0.0% of 15592216 rows
Read 26.1% of 15592216 rows
Read 51.5% of 15592216 rows
Read 76.7% of 15592216 rows
Read 99.3% of 15592216 rows
Read 15592216 rows and 8 (of 8) columns from 0.806 GB file in 00:00:08
## 
Read 0.0% of 15592216 rows
Read 16.7% of 15592216 rows
Read 32.8% of 15592216 rows
Read 49.2% of 15592216 rows
Read 67.0% of 15592216 rows
Read 86.3% of 15592216 rows
Read 15592216 rows and 8 (of 8) columns from 0.806 GB file in 00:00:10
## 
Read 0.0% of 15592216 rows
Read 26.4% of 15592216 rows
Read 53.1% of 15592216 rows
Read 79.8% of 15592216 rows
Read 15592216 rows and 8 (of 8) columns from 0.806 GB file in 00:00:07
## 
Read 0.0% of 15592216 rows
Read 25.8% of 15592216 rows
Read 52.1% of 15592216 rows
Read 78.1% of 15592216 rows
Read 15592216 rows and 8 (of 8) columns from 0.806 GB file in 00:00:08
## 
Read 63.5% of 15592216 rows
Read 15592216 rows and 2 (of 2) columns from 0.318 GB file in 00:00:03
## 
Read 64.4% of 15592216 rows
Read 15592216 rows and 2 (of 2) columns from 0.309 GB file in 00:00:03
## 
Read 46.2% of 15592972 rows
Read 76.3% of 15592972 rows
Read 15592972 rows and 2 (of 2) columns from 0.401 GB file in 00:00:04
## 
Read 0.0% of 15592972 rows
Read 26.6% of 15592972 rows
Read 53.6% of 15592972 rows
Read 80.1% of 15592972 rows
Read 15592972 rows and 8 (of 8) columns from 0.806 GB file in 00:00:08
## 
Read 0.0% of 15592972 rows
Read 18.6% of 15592972 rows
Read 39.6% of 15592972 rows
Read 63.0% of 15592972 rows
Read 86.4% of 15592972 rows
Read 15592972 rows and 8 (of 8) columns from 0.806 GB file in 00:00:07
## 
Read 19.1% of 15592972 rows
Read 38.6% of 15592972 rows
Read 62.1% of 15592972 rows
Read 85.7% of 15592972 rows
Read 15592972 rows and 8 (of 8) columns from 0.806 GB file in 00:00:06
## 
Read 0.0% of 15592972 rows
Read 25.8% of 15592972 rows
Read 51.4% of 15592972 rows
Read 76.1% of 15592972 rows
Read 15592972 rows and 8 (of 8) columns from 0.806 GB file in 00:00:12
## 
Read 61.6% of 15592972 rows
Read 98.8% of 15592972 rows
Read 15592972 rows and 2 (of 2) columns from 0.318 GB file in 00:00:04
## 
Read 66.7% of 15592972 rows
Read 15592972 rows and 2 (of 2) columns from 0.309 GB file in 00:00:03
## 
Read 36.1% of 15586293 rows
Read 58.9% of 15586293 rows
Read 82.4% of 15586293 rows
Read 15586293 rows and 2 (of 2) columns from 0.400 GB file in 00:00:05
## 
Read 0.0% of 15586293 rows
Read 21.4% of 15586293 rows
Read 43.1% of 15586293 rows
Read 64.7% of 15586293 rows
Read 86.3% of 15586293 rows
Read 15586293 rows and 8 (of 8) columns from 0.805 GB file in 00:00:08
## 
Read 14.6% of 15586293 rows
Read 36.8% of 15586293 rows
Read 58.8% of 15586293 rows
Read 81.1% of 15586293 rows
Read 15586293 rows and 8 (of 8) columns from 0.805 GB file in 00:00:06
## 
Read 39.5% of 15586293 rows
Read 60.7% of 15586293 rows
Read 82.6% of 15586293 rows
Read 15586293 rows and 8 (of 8) columns from 0.805 GB file in 00:00:05
## 
Read 31.8% of 15586293 rows
Read 53.4% of 15586293 rows
Read 74.9% of 15586293 rows
Read 96.3% of 15586293 rows
Read 15586293 rows and 8 (of 8) columns from 0.805 GB file in 00:00:06
## 
Read 68.8% of 15586293 rows
Read 15586293 rows and 2 (of 2) columns from 0.318 GB file in 00:00:03
## 
Read 63.3% of 15586293 rows
Read 15586293 rows and 2 (of 2) columns from 0.309 GB file in 00:00:03
## 
Read 46.6% of 15586405 rows
Read 73.5% of 15586405 rows
Read 15586405 rows and 2 (of 2) columns from 0.400 GB file in 00:00:04
## 
Read 0.0% of 15586405 rows
Read 26.8% of 15586405 rows
Read 51.1% of 15586405 rows
Read 75.4% of 15586405 rows
Read 97.1% of 15586405 rows
Read 15586405 rows and 8 (of 8) columns from 0.805 GB file in 00:00:08
## 
Read 27.6% of 15586405 rows
Read 51.6% of 15586405 rows
Read 74.6% of 15586405 rows
Read 97.1% of 15586405 rows
Read 15586405 rows and 8 (of 8) columns from 0.805 GB file in 00:00:06
## 
Read 23.5% of 15586405 rows
Read 46.4% of 15586405 rows
Read 69.6% of 15586405 rows
Read 92.8% of 15586405 rows
Read 15586405 rows and 8 (of 8) columns from 0.805 GB file in 00:00:06
## 
Read 24.7% of 15586405 rows
Read 47.9% of 15586405 rows
Read 71.6% of 15586405 rows
Read 95.4% of 15586405 rows
Read 15586405 rows and 8 (of 8) columns from 0.805 GB file in 00:00:06
## 
Read 54.6% of 15586405 rows
Read 92.1% of 15586405 rows
Read 15586405 rows and 2 (of 2) columns from 0.317 GB file in 00:00:04
## 
Read 74.8% of 15586405 rows
Read 15586405 rows and 2 (of 2) columns from 0.308 GB file in 00:00:03
## 
Read 28.0% of 15582205 rows
Read 48.6% of 15582205 rows
Read 71.4% of 15582205 rows
Read 95.7% of 15582205 rows
Read 15582205 rows and 2 (of 2) columns from 0.398 GB file in 00:00:06
## 
Read 0.0% of 15582205 rows
Read 26.5% of 15582205 rows
Read 53.4% of 15582205 rows
Read 79.7% of 15582205 rows
Read 15582205 rows and 8 (of 8) columns from 0.804 GB file in 00:00:12
## 
Read 20.0% of 15582205 rows
Read 43.7% of 15582205 rows
Read 66.7% of 15582205 rows
Read 90.6% of 15582205 rows
Read 15582205 rows and 8 (of 8) columns from 0.804 GB file in 00:00:06
## 
Read 23.6% of 15582205 rows
Read 47.4% of 15582205 rows
Read 70.1% of 15582205 rows
Read 93.5% of 15582205 rows
Read 15582205 rows and 8 (of 8) columns from 0.804 GB file in 00:00:06
## 
Read 23.7% of 15582205 rows
Read 47.6% of 15582205 rows
Read 71.3% of 15582205 rows
Read 95.2% of 15582205 rows
Read 15582205 rows and 8 (of 8) columns from 0.804 GB file in 00:00:06
## 
Read 0.0% of 15582205 rows
Read 35.6% of 15582205 rows
Read 72.6% of 15582205 rows
Read 15582205 rows and 2 (of 2) columns from 0.316 GB file in 00:00:05
## 
Read 51.4% of 15582205 rows
Read 90.2% of 15582205 rows
Read 15582205 rows and 2 (of 2) columns from 0.307 GB file in 00:00:04</code></pre>
<pre class="r"><code>rm(new_control)
system(paste(&quot;rm &quot;,prefix,&quot;annovar_input_temp.bed&quot;,sep = &quot;&quot;))
system(paste(&quot;rm &quot;,prefix, &quot;temp.bed&quot;,sep = &quot;&quot;))</code></pre>
<pre class="r"><code># notice now feature start is 5, and feature end is 6, two features 
simulation_estimate &lt;- estimate_effect_size_for_simulation_data_mixture_with_categorization_for_compact_data(data = adjusted_base_level_rr$base_info, feature_start = 5, feature_end = 9, feature_number = 5, gene_prior_file = &quot;../other_annotation/gene_list/TADA_SNV_CNV_combined_Feb7_tadaname_null_PPA_p0_0.94.txt&quot;, sample_size = 314)</code></pre>
<pre><code>## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.</code></pre>
<pre class="r"><code>fisher_info&lt;-solve(-simulation_estimate$mle$hessian)
prop_sigma&lt;-sqrt(diag(fisher_info))
upper&lt;-simulation_estimate$mle$par+1.96*prop_sigma
lower&lt;-simulation_estimate$mle$par-1.96*prop_sigma
a = data.frame(relative_risk = simulation_estimate$mle$par, lower_bound = lower, upper_bournd = upper)
rownames(a) = c(&quot;Noonan+Roadmap Brain H3K27ac enhancer + promoter&quot;,&quot;Epigenome_DHS&quot;, &quot;ENCODE_DHS&quot;, &quot;Phylop&gt;2&quot;, &quot;Gerp&gt;2&quot;)
knitr::kable(a)</code></pre>
<table>
<thead>
<tr class="header">
<th></th>
<th align="right">relative_risk</th>
<th align="right">lower_bound</th>
<th align="right">upper_bournd</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Noonan+Roadmap Brain H3K27ac enhancer + promoter</td>
<td align="right">0.8959283</td>
<td align="right">0.1872562</td>
<td align="right">1.6046004</td>
</tr>
<tr class="even">
<td>Epigenome_DHS</td>
<td align="right">-0.6461505</td>
<td align="right">-2.3473682</td>
<td align="right">1.0550672</td>
</tr>
<tr class="odd">
<td>ENCODE_DHS</td>
<td align="right">-0.4597383</td>
<td align="right">-1.4500728</td>
<td align="right">0.5305963</td>
</tr>
<tr class="even">
<td>Phylop&gt;2</td>
<td align="right">-0.5451650</td>
<td align="right">-2.4313140</td>
<td align="right">1.3409840</td>
</tr>
<tr class="odd">
<td>Gerp&gt;2</td>
<td align="right">0.6537426</td>
<td align="right">-0.1432876</td>
<td align="right">1.4507727</td>
</tr>
</tbody>
</table>
<pre class="r"><code>adjusted_base_level_rr$partition_time1</code></pre>
<pre><code>##  [1] 1.153 1.216 1.287 1.432 1.341 1.432 1.357 1.301 1.417 1.549 1.533
## [12] 1.361 1.685 1.502 1.779 1.406 1.695 1.367 1.577 1.802</code></pre>
<pre class="r"><code>adjusted_base_level_rr$partition_time2</code></pre>
<pre><code>##  [1] 34.057 34.236 34.338 34.368 34.519 35.557 35.645 34.955 35.201 35.386
## [11] 35.013 35.632 36.437 36.273 37.030 35.960 35.997 36.090 35.663 36.480</code></pre>
</div>
<div id="now-will-run-foward-selection-to-select-features" class="section level1">
<h1><span class="header-section-number">2</span> <strong>now will run foward selection to select features</strong></h1>
<p>Will start from Brain H3K27ac, to Epigenomc DHS, Encode DHS, to phylop and to Gerp. For each additional feature, after parameter optimization, will calculate the deviance between two models and then use 1 degree of freedom chi square distribution to see if the improvement is significant.</p>
<pre class="r"><code># model with only Brain H3K27ac data
model_estimate_1 &lt;- estimate_effect_size_for_simulation_data_mixture_with_categorization_for_compact_data_model_selection(data = adjusted_base_level_rr$base_info, features = c(1), gene_prior_file = &quot;../other_annotation/gene_list/TADA_SNV_CNV_combined_Feb7_tadaname_null_PPA_p0_0.94.txt&quot;, sample_size = 314)</code></pre>
<pre><code>## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.</code></pre>
<pre class="r"><code>model_estimate_1$mle</code></pre>
<pre><code>## $par
## [1] 0.5687403
## 
## $value
## [1] -25794.11
## 
## $counts
## function gradient 
##       NA       NA 
## 
## $convergence
## [1] 0
## 
## $message
## NULL
## 
## $hessian
##           [,1]
## [1,] -10.57385</code></pre>
<pre class="r"><code>fisher_info&lt;-solve(-model_estimate_1$mle$hessian)
prop_sigma&lt;-sqrt(diag(fisher_info))
upper&lt;-model_estimate_1$mle$par+1.96*prop_sigma
lower&lt;-model_estimate_1$mle$par-1.96*prop_sigma
a = data.frame(relative_risk = model_estimate_1$mle$par, lower_bound = lower, upper_bournd = upper)
rownames(a) = c(&quot;Noonan+Roadmap Brain H3K27ac enhancer + promoter&quot;)
knitr::kable(a)</code></pre>
<table>
<thead>
<tr class="header">
<th></th>
<th align="right">relative_risk</th>
<th align="right">lower_bound</th>
<th align="right">upper_bournd</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Noonan+Roadmap Brain H3K27ac enhancer + promoter</td>
<td align="right">0.5687403</td>
<td align="right">-0.0340128</td>
<td align="right">1.171493</td>
</tr>
</tbody>
</table>
<pre class="r"><code># model with Brain H3K27ac data + Epigenome brain DHS
model_estimate_2 &lt;- estimate_effect_size_for_simulation_data_mixture_with_categorization_for_compact_data_model_selection(data = adjusted_base_level_rr$base_info, features = c(1,2), gene_prior_file = &quot;../other_annotation/gene_list/TADA_SNV_CNV_combined_Feb7_tadaname_null_PPA_p0_0.94.txt&quot;, sample_size = 314)</code></pre>
<pre><code>## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.</code></pre>
<pre><code>## Warning in if (features != 0) {: the condition has length &gt; 1 and only the
## first element will be used</code></pre>
<pre class="r"><code>model_estimate_2$mle</code></pre>
<pre><code>## $par
## [1]  0.7291273 -0.6301970
## 
## $value
## [1] -25793.74
## 
## $counts
## function gradient 
##       35       NA 
## 
## $convergence
## [1] 0
## 
## $message
## NULL
## 
## $hessian
##            [,1]      [,2]
## [1,] -12.034174 -1.857848
## [2,]  -1.857848 -1.723183</code></pre>
<pre class="r"><code>fisher_info&lt;-solve(-model_estimate_2$mle$hessian)
prop_sigma&lt;-sqrt(diag(fisher_info))
upper&lt;-model_estimate_2$mle$par+1.96*prop_sigma
lower&lt;-model_estimate_2$mle$par-1.96*prop_sigma
a = data.frame(relative_risk = model_estimate_2$mle$par, lower_bound = lower, upper_bournd = upper)
rownames(a) = c(&quot;Noonan+Roadmap Brain H3K27ac enhancer + promoter&quot;, &quot;Epigenome DHS&quot;)
knitr::kable(a)</code></pre>
<table>
<thead>
<tr class="header">
<th></th>
<th align="right">relative_risk</th>
<th align="right">lower_bound</th>
<th align="right">upper_bournd</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Noonan+Roadmap Brain H3K27ac enhancer + promoter</td>
<td align="right">0.7291273</td>
<td align="right">0.1102836</td>
<td align="right">1.347971</td>
</tr>
<tr class="even">
<td>Epigenome DHS</td>
<td align="right">-0.6301970</td>
<td align="right">-2.2655948</td>
<td align="right">1.005201</td>
</tr>
</tbody>
</table>
<pre class="r"><code># calculate chi-square p-value 
llr &lt;- model_estimate_1$mle$value - model_estimate_2$mle$value

llr</code></pre>
<pre><code>## [1] -0.3731553</code></pre>
<pre class="r"><code># calculate chi-squre pvalue
pchisq(-2 * llr, df = 1, lower.tail = FALSE)</code></pre>
<pre><code>## [1] 0.3876468</code></pre>
<pre class="r"><code># model with Brain H3K27ac data + ENCODE DHS
model_estimate_3 &lt;- estimate_effect_size_for_simulation_data_mixture_with_categorization_for_compact_data_model_selection(data = adjusted_base_level_rr$base_info, features = c(1,3), gene_prior_file = &quot;../other_annotation/gene_list/TADA_SNV_CNV_combined_Feb7_tadaname_null_PPA_p0_0.94.txt&quot;, sample_size = 314)</code></pre>
<pre><code>## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.</code></pre>
<pre><code>## Warning in if (features != 0) {: the condition has length &gt; 1 and only the
## first element will be used</code></pre>
<pre class="r"><code>model_estimate_3$mle</code></pre>
<pre><code>## $par
## [1]  0.8938155 -0.6030159
## 
## $value
## [1] -25793.41
## 
## $counts
## function gradient 
##       53       NA 
## 
## $convergence
## [1] 0
## 
## $message
## NULL
## 
## $hessian
##            [,1]      [,2]
## [1,] -10.448669 -4.162762
## [2,]  -4.162762 -5.279994</code></pre>
<pre class="r"><code>fisher_info&lt;-solve(-model_estimate_3$mle$hessian)
prop_sigma&lt;-sqrt(diag(fisher_info))
upper&lt;-model_estimate_3$mle$par+1.96*prop_sigma
lower&lt;-model_estimate_3$mle$par-1.96*prop_sigma
a = data.frame(relative_risk = model_estimate_3$mle$par, lower_bound = lower, upper_bournd = upper)
rownames(a) = c(&quot;Noonan+Roadmap Brain H3K27ac enhancer + promoter&quot;, &quot;Encode DHS&quot;)
knitr::kable(a)</code></pre>
<table>
<thead>
<tr class="header">
<th></th>
<th align="right">relative_risk</th>
<th align="right">lower_bound</th>
<th align="right">upper_bournd</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Noonan+Roadmap Brain H3K27ac enhancer + promoter</td>
<td align="right">0.8938155</td>
<td align="right">0.1616734</td>
<td align="right">1.6259575</td>
</tr>
<tr class="even">
<td>Encode DHS</td>
<td align="right">-0.6030159</td>
<td align="right">-1.6329493</td>
<td align="right">0.4269174</td>
</tr>
</tbody>
</table>
<pre class="r"><code># calculate chi-square p-value 
llr &lt;- model_estimate_1$mle$value - model_estimate_3$mle$value

llr</code></pre>
<pre><code>## [1] -0.707191</code></pre>
<pre class="r"><code># calculate chi-squre pvalue
pchisq(-2 * llr, df = 1, lower.tail = FALSE)</code></pre>
<pre><code>## [1] 0.2343303</code></pre>
<pre class="r"><code># model with Brain H3K27ac data + Phylop &gt; 2
model_estimate_4 &lt;- estimate_effect_size_for_simulation_data_mixture_with_categorization_for_compact_data_model_selection(data = adjusted_base_level_rr$base_info, features = c(1,4), gene_prior_file = &quot;../other_annotation/gene_list/TADA_SNV_CNV_combined_Feb7_tadaname_null_PPA_p0_0.94.txt&quot;, sample_size = 314)</code></pre>
<pre><code>## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.</code></pre>
<pre><code>## Warning in if (features != 0) {: the condition has length &gt; 1 and only the
## first element will be used</code></pre>
<pre class="r"><code>model_estimate_4$mle</code></pre>
<pre><code>## $par
## [1]  0.5978384 -0.4981799
## 
## $value
## [1] -25794.01
## 
## $counts
## function gradient 
##       31       NA 
## 
## $convergence
## [1] 0
## 
## $message
## NULL
## 
## $hessian
##             [,1]       [,2]
## [1,] -10.9044968 -0.4197054
## [2,]  -0.4197054 -0.8077131</code></pre>
<pre class="r"><code>fisher_info&lt;-solve(-model_estimate_4$mle$hessian)
prop_sigma&lt;-sqrt(diag(fisher_info))
upper&lt;-model_estimate_4$mle$par+1.96*prop_sigma
lower&lt;-model_estimate_4$mle$par-1.96*prop_sigma
a = data.frame(relative_risk = model_estimate_4$mle$par, lower_bound = lower, upper_bournd = upper)
rownames(a) = c(&quot;Noonan+Roadmap Brain H3K27ac enhancer + promoter&quot;, &quot;PhyloP &gt; 2&quot;)
knitr::kable(a)</code></pre>
<table>
<thead>
<tr class="header">
<th></th>
<th align="right">relative_risk</th>
<th align="right">lower_bound</th>
<th align="right">upper_bournd</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Noonan+Roadmap Brain H3K27ac enhancer + promoter</td>
<td align="right">0.5978384</td>
<td align="right">-0.001732</td>
<td align="right">1.197409</td>
</tr>
<tr class="even">
<td>PhyloP &gt; 2</td>
<td align="right">-0.4981799</td>
<td align="right">-2.701179</td>
<td align="right">1.704820</td>
</tr>
</tbody>
</table>
<pre class="r"><code># calculate chi-square p-value 
llr &lt;- model_estimate_1$mle$value - model_estimate_4$mle$value

llr</code></pre>
<pre><code>## [1] -0.1073395</code></pre>
<pre class="r"><code># calculate chi-squre pvalue
pchisq(-2 * llr, df = 1, lower.tail = FALSE)</code></pre>
<pre><code>## [1] 0.6431245</code></pre>
<pre class="r"><code># model with Brain H3K27ac data + GERP &gt; 2
model_estimate_5 &lt;- estimate_effect_size_for_simulation_data_mixture_with_categorization_for_compact_data_model_selection(data = adjusted_base_level_rr$base_info, features = c(1,5), gene_prior_file = &quot;../other_annotation/gene_list/TADA_SNV_CNV_combined_Feb7_tadaname_null_PPA_p0_0.94.txt&quot;, sample_size = 314)</code></pre>
<pre><code>## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.</code></pre>
<pre><code>## Warning in if (features != 0) {: the condition has length &gt; 1 and only the
## first element will be used</code></pre>
<pre class="r"><code>model_estimate_5$mle</code></pre>
<pre><code>## $par
## [1] 0.475683 0.433631
## 
## $value
## [1] -25793.67
## 
## $counts
## function gradient 
##       43       NA 
## 
## $convergence
## [1] 0
## 
## $message
## NULL
## 
## $hessian
##            [,1]      [,2]
## [1,] -10.690934 -3.135604
## [2,]  -3.135604 -6.505456</code></pre>
<pre class="r"><code>fisher_info&lt;-solve(-model_estimate_5$mle$hessian)
prop_sigma&lt;-sqrt(diag(fisher_info))
upper&lt;-model_estimate_5$mle$par+1.96*prop_sigma
lower&lt;-model_estimate_5$mle$par-1.96*prop_sigma
a = data.frame(relative_risk = model_estimate_5$mle$par, lower_bound = lower, upper_bournd = upper)
rownames(a) = c(&quot;Noonan+Roadmap Brain H3K27ac enhancer + promoter&quot;, &quot;GERP &gt; 2&quot;)
knitr::kable(a)</code></pre>
<table>
<thead>
<tr class="header">
<th></th>
<th align="right">relative_risk</th>
<th align="right">lower_bound</th>
<th align="right">upper_bournd</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Noonan+Roadmap Brain H3K27ac enhancer + promoter</td>
<td align="right">0.475683</td>
<td align="right">-0.1712280</td>
<td align="right">1.122594</td>
</tr>
<tr class="even">
<td>GERP &gt; 2</td>
<td align="right">0.433631</td>
<td align="right">-0.3956725</td>
<td align="right">1.262935</td>
</tr>
</tbody>
</table>
<pre class="r"><code># calculate chi-square p-value 
llr &lt;- model_estimate_1$mle$value - model_estimate_5$mle$value

llr</code></pre>
<pre><code>## [1] -0.4429764</code></pre>
<pre class="r"><code># calculate chi-squre pvalue
pchisq(-2 * llr, df = 1, lower.tail = FALSE)</code></pre>
<pre><code>## [1] 0.3465764</code></pre>
<pre class="r"><code># model with only Brain H3K27ac data
model_estimate_1 &lt;- estimate_effect_size_for_simulation_data_mixture_with_categorization_for_compact_data_model_selection(data = adjusted_base_level_rr$base_info, features = c(1), gene_prior_file = &quot;../other_annotation/gene_list/TADA_SNV_CNV_combined_Feb7_tadaname_null_PPA_p0_0.94.txt&quot;, sample_size = 314)</code></pre>
<pre><code>## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.</code></pre>
<pre class="r"><code>model_estimate_1$mle</code></pre>
<pre><code>## $par
## [1] 0.5687403
## 
## $value
## [1] -25794.11
## 
## $counts
## function gradient 
##       NA       NA 
## 
## $convergence
## [1] 0
## 
## $message
## NULL
## 
## $hessian
##           [,1]
## [1,] -10.57385</code></pre>
<pre class="r"><code>fisher_info&lt;-solve(-model_estimate_1$mle$hessian)
prop_sigma&lt;-sqrt(diag(fisher_info))
upper&lt;-model_estimate_1$mle$par+1.96*prop_sigma
lower&lt;-model_estimate_1$mle$par-1.96*prop_sigma
a = data.frame(relative_risk = model_estimate_1$mle$par, lower_bound = lower, upper_bournd = upper)
rownames(a) = c(&quot;Noonan+Roadmap Brain H3K27ac enhancer + promoter&quot;)
knitr::kable(a)</code></pre>
<table>
<thead>
<tr class="header">
<th></th>
<th align="right">relative_risk</th>
<th align="right">lower_bound</th>
<th align="right">upper_bournd</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Noonan+Roadmap Brain H3K27ac enhancer + promoter</td>
<td align="right">0.5687403</td>
<td align="right">-0.0340128</td>
<td align="right">1.171493</td>
</tr>
</tbody>
</table>
<pre class="r"><code># model without any parameters
model_estimate_0 &lt;- estimate_effect_size_for_simulation_data_mixture_with_categorization_for_compact_data_model_selection(data = adjusted_base_level_rr$base_info, features = 0, gene_prior_file = &quot;../other_annotation/gene_list/TADA_SNV_CNV_combined_Feb7_tadaname_null_PPA_p0_0.94.txt&quot;, sample_size = 314)</code></pre>
<pre><code>## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.</code></pre>
<pre class="r"><code>model_estimate_0$likelihood</code></pre>
<pre><code>## [1] -25795.14</code></pre>
<pre class="r"><code># calculate chi-square p-value 
llr &lt;- model_estimate_0$likelihood - model_estimate_1$mle$value

llr</code></pre>
<pre><code>## [1] -1.025471</code></pre>
<pre class="r"><code># calculate chi-squre pvalue
pchisq(-2 * llr, df = 1, lower.tail = FALSE)</code></pre>
<pre><code>## [1] 0.152112</code></pre>
</div>
<div id="now-use-estimated-relative-risks-to-identify-new-asd-genes" class="section level1">
<h1><span class="header-section-number">3</span> <strong>Now use estimated relative risks to identify new ASD genes</strong></h1>
<pre class="r"><code>source(&quot;../lib/TADA_denovo.R&quot;)
source(&quot;../lib/170117_gene_validation_functions.R&quot;)</code></pre>
<div id="sart-without-using-splicing-mutations" class="section level2">
<h2><span class="header-section-number">3.1</span> <em>Sart without using splicing mutations</em></h2>
<pre class="r"><code>TADA_table&lt;-bayes_factor_for_each_gene_base_level_from_collapsed_data(adjusted_base_level_rr$base_info, 
                                                                      features = c(1), 
                                                                      rr = 0.6575217,
                                                                      coding_bayes_table_file = &quot;../other_annotation/gene_list/TADA_SNV_CNV_combined_Feb7_tadaname_BF_for_de_novo_mutations.txt&quot;,
                                                                      TADA_p0 = 0.94)</code></pre>
<div id="number-of-gained-and-lost-genes-after-having-non-coding-mutations" class="section level3">
<h3><span class="header-section-number">3.1.1</span> Number of gained and lost genes after having non-coding mutations</h3>
<pre class="r"><code>TADA_table_filtered = TADA_table$gene_BF_table[!is.element(TADA_table$gene_BF_table$genename, TADA_table$genes_with_no_epi),]
knitr::kable(performance_table(TADA_table_filtered))</code></pre>
<table>
<thead>
<tr class="header">
<th></th>
<th align="right">novel</th>
<th align="right">consistent</th>
<th align="right">lost</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>q0.1</td>
<td align="right">3</td>
<td align="right">55</td>
<td align="right">2</td>
</tr>
<tr class="even">
<td>q0.2</td>
<td align="right">8</td>
<td align="right">90</td>
<td align="right">3</td>
</tr>
<tr class="odd">
<td>q0.3</td>
<td align="right">9</td>
<td align="right">127</td>
<td align="right">5</td>
</tr>
</tbody>
</table>
</div>
<div id="gene-set-enrichment-analysis-fdr-0.1" class="section level3">
<h3><span class="header-section-number">3.1.2</span> Gene set enrichment analysis FDR 0.1</h3>
<pre class="r"><code>knitr::kable(foe_table(TADA_table_filtered[TADA_table_filtered$FDR_coding &gt;=0.1 &amp; TADA_table_filtered$FDR_all &lt; 0.1, ]$genename, TADA_table_filtered$genename))</code></pre>
<table>
<thead>
<tr class="header">
<th></th>
<th align="right">HI</th>
<th align="right">FMRP</th>
<th align="right">RVIS top 25%</th>
<th align="right">Brainspan top 25%</th>
<th align="right">Brain GO</th>
<th align="right">presynaptic</th>
<th align="right">postsynaptic</th>
<th align="right">ExAC top 25%</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Fold of enrichment</td>
<td align="right">9.6533090</td>
<td align="right">6.6578727</td>
<td align="right">4.007815</td>
<td align="right">3.5050705</td>
<td align="right">2.1072233</td>
<td align="right">16.5098650</td>
<td align="right">3.1211229</td>
<td align="right">2.5158636</td>
</tr>
<tr class="even">
<td>pvalue</td>
<td align="right">0.0011098</td>
<td align="right">0.1428123</td>
<td align="right">0.015525</td>
<td align="right">0.0232116</td>
<td align="right">0.4034691</td>
<td align="right">0.0593588</td>
<td align="right">0.2874136</td>
<td align="right">0.1734212</td>
</tr>
</tbody>
</table>
</div>
<div id="gene-set-enrichment-analysis-fdr-0.2" class="section level3">
<h3><span class="header-section-number">3.1.3</span> Gene set enrichment analysis FDR 0.2</h3>
<pre class="r"><code>knitr::kable(foe_table(TADA_table_filtered[TADA_table_filtered$FDR_coding &gt;=0.2 &amp; TADA_table_filtered$FDR_all &lt; 0.2, ]$genename, TADA_table_filtered$genename))</code></pre>
<table>
<thead>
<tr class="header">
<th></th>
<th align="right">HI</th>
<th align="right">FMRP</th>
<th align="right">RVIS top 25%</th>
<th align="right">Brainspan top 25%</th>
<th align="right">Brain GO</th>
<th align="right">presynaptic</th>
<th align="right">postsynaptic</th>
<th align="right">ExAC top 25%</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Fold of enrichment</td>
<td align="right">3.6199909</td>
<td align="right">2.496702</td>
<td align="right">2.0039072</td>
<td align="right">1.3144015</td>
<td align="right">0.7902087</td>
<td align="right">6.1911994</td>
<td align="right">2.3408422</td>
<td align="right">0.9434488</td>
</tr>
<tr class="even">
<td>pvalue</td>
<td align="right">0.0417051</td>
<td align="right">0.337010</td>
<td align="right">0.1130872</td>
<td align="right">0.4107859</td>
<td align="right">0.7478928</td>
<td align="right">0.1505861</td>
<td align="right">0.2073205</td>
<td align="right">0.6691887</td>
</tr>
</tbody>
</table>
</div>
<div id="gene-set-enrichment-analysis-fdr-0.3" class="section level3">
<h3><span class="header-section-number">3.1.4</span> Gene set enrichment analysis FDR 0.3</h3>
<pre class="r"><code>knitr::kable(foe_table(TADA_table_filtered[TADA_table_filtered$FDR_coding &gt;=0.3 &amp; TADA_table_filtered$FDR_all &lt; 0.3, ]$genename, TADA_table_filtered$genename))</code></pre>
<table>
<thead>
<tr class="header">
<th></th>
<th align="right">HI</th>
<th align="right">FMRP</th>
<th align="right">RVIS top 25%</th>
<th align="right">Brainspan top 25%</th>
<th align="right">Brain GO</th>
<th align="right">presynaptic</th>
<th align="right">postsynaptic</th>
<th align="right">ExAC top 25%</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Fold of enrichment</td>
<td align="right">3.217770</td>
<td align="right">4.438582</td>
<td align="right">1.7812509</td>
<td align="right">1.9472614</td>
<td align="right">1.4048156</td>
<td align="right">5.5032883</td>
<td align="right">1.0403743</td>
<td align="right">0.8386212</td>
</tr>
<tr class="even">
<td>pvalue</td>
<td align="right">0.057845</td>
<td align="right">0.071330</td>
<td align="right">0.1647571</td>
<td align="right">0.0817439</td>
<td align="right">0.4286935</td>
<td align="right">0.1677444</td>
<td align="right">0.6382368</td>
<td align="right">0.7343074</td>
</tr>
</tbody>
</table>
</div>
<div id="fdr-0.3-new-genes" class="section level3">
<h3><span class="header-section-number">3.1.5</span> FDR 0.3 new genes</h3>
<pre class="r"><code>knitr::kable(TADA_table_filtered[TADA_table_filtered$FDR_coding &gt;= 0.3 &amp; TADA_table_filtered$FDR_all &lt; 0.3,])</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">genename</th>
<th align="right">logBF_noncoding</th>
<th align="right">logBF_coding</th>
<th align="right">logBF_all</th>
<th align="right">BF_noncoding</th>
<th align="right">BF_coding</th>
<th align="right">BF_all</th>
<th align="right">FDR_coding</th>
<th align="right">FDR_all</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">JUP</td>
<td align="right">2.5872789</td>
<td align="right">2.131726</td>
<td align="right">4.719005</td>
<td align="right">13.293549</td>
<td align="right">8.429401</td>
<td align="right">112.05665</td>
<td align="right">0.4713748</td>
<td align="right">0.0441116</td>
</tr>
<tr class="even">
<td align="left">APBB1</td>
<td align="right">1.8901033</td>
<td align="right">2.325413</td>
<td align="right">4.215517</td>
<td align="right">6.620052</td>
<td align="right">10.230909</td>
<td align="right">67.72915</td>
<td align="right">0.4502329</td>
<td align="right">0.0659638</td>
</tr>
<tr class="odd">
<td align="left">MSL2</td>
<td align="right">0.6235923</td>
<td align="right">2.536128</td>
<td align="right">3.159721</td>
<td align="right">1.865618</td>
<td align="right">12.630673</td>
<td align="right">23.56401</td>
<td align="right">0.3972018</td>
<td align="right">0.1777773</td>
</tr>
<tr class="even">
<td align="left">PPM1D</td>
<td align="right">0.6246737</td>
<td align="right">2.526200</td>
<td align="right">3.150873</td>
<td align="right">1.867636</td>
<td align="right">12.505890</td>
<td align="right">23.35646</td>
<td align="right">0.4001637</td>
<td align="right">0.1801570</td>
</tr>
<tr class="odd">
<td align="left">SLC38A3</td>
<td align="right">0.5877401</td>
<td align="right">2.510331</td>
<td align="right">3.098071</td>
<td align="right">1.799916</td>
<td align="right">12.309005</td>
<td align="right">22.15518</td>
<td align="right">0.4074689</td>
<td align="right">0.1873184</td>
</tr>
<tr class="even">
<td align="left">PLXNB1</td>
<td align="right">0.6101821</td>
<td align="right">2.352516</td>
<td align="right">2.962698</td>
<td align="right">1.840766</td>
<td align="right">10.511988</td>
<td align="right">19.35012</td>
<td align="right">0.4468929</td>
<td align="right">0.2015386</td>
</tr>
<tr class="odd">
<td align="left">LRRC4</td>
<td align="right">0.6103642</td>
<td align="right">2.344952</td>
<td align="right">2.955317</td>
<td align="right">1.841102</td>
<td align="right">10.432776</td>
<td align="right">19.20780</td>
<td align="right">0.4477285</td>
<td align="right">0.2039202</td>
</tr>
<tr class="even">
<td align="left">CBX4</td>
<td align="right">0.5351793</td>
<td align="right">2.237874</td>
<td align="right">2.773053</td>
<td align="right">1.707754</td>
<td align="right">9.373381</td>
<td align="right">16.00743</td>
<td align="right">0.4613862</td>
<td align="right">0.2801414</td>
</tr>
<tr class="odd">
<td align="left">SIRPB2</td>
<td align="right">-0.0047030</td>
<td align="right">2.736432</td>
<td align="right">2.731729</td>
<td align="right">0.995308</td>
<td align="right">15.431826</td>
<td align="right">15.35942</td>
<td align="right">0.3009029</td>
<td align="right">0.2957133</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="cacluate-the-bayes-factor-table-for-splicing-mutations" class="section level2">
<h2><span class="header-section-number">3.2</span> <em>cacluate the bayes factor table for splicing mutations</em></h2>
<ul>
<li>The mutation rate needs to be adjusted after taking into consideration of windows from introns far from TSSs</li>
</ul>
<pre class="r"><code>new_control = new.env()
load(&quot;../data/debug_region_list_073116_data_matrixtransformed_for_old_code.Rdata&quot;, new_control)

prefix &lt;-system(&quot;date +%s&quot;, intern = TRUE) # prefix for temporary files that will be deleted at the end of the pipeline

write.table(new_control$data_matrix[new_control$data_matrix$Prediction == &quot;dnv_proband&quot; &amp; new_control$data_matrix$Type == &quot;SNV&quot;,c(1,2,3,7)], paste(prefix,&quot;temp.bed&quot;,sep = &quot;&quot;), col.names = FALSE, row.names = FALSE, sep = &quot;\t&quot;, quote = FALSE)
#write.table(new_control$data_matrix[new_control$data_matrix$Prediction == &quot;dnv_proband&quot; ,c(1:5,8,7)], paste(prefix,&quot;annovar_input_temp.bed&quot;,sep = &quot;&quot;), col.names = FALSE, row.names = FALSE, sep = &quot;\t&quot;, quote = FALSE)

# SNV_ID for cases or controls. 
SNV_ID = new_control$data_matrix[new_control$data_matrix$Prediction == &quot;dnv_proband&quot; &amp; new_control$data_matrix$Type == &quot;SNV&quot;,c(1,2,3,7)]$ID
proband_SNV_mut_spidex &lt;- new_control$mut_spedix[is.element(new_control$mut_spedix$index, SNV_ID),] # this will be paseed as an argument to the function below

### generate Bayes factor table for splicing mutations
splicing_BF_table &lt;- generate_BF_for_splicing_mutations(mut_file = paste(prefix,&quot;temp.bed&quot;,sep = &quot;&quot;), 
                                               window_file = &quot;../other_annotation/epigenomic_annotation/Whole_genome.promoter_yanyu_10kb_exons_introns_with_spidex_no_na_genes.50bp_window.bed&quot;, 
                                               cg_file = &quot;../other_annotation/epigenomic_annotation/Whole_genome.promoter_yanyu_10kb_exons_introns_with_spidex_no_na_genes.50bp_window.bed.fasta.cg&quot;, 
                                               mutrate_file = &quot;../other_annotation/epigenomic_annotation/Whole_genome.promoter_yanyu_10kb_exons_introns_with_spidex_no_na_genes.50bp_window.bed.mutrate&quot;, 
                                               sample_size = 314,
                                               rm_nonsyn = FALSE,
                                               annovar_folder = &quot;/media/yuwen/Elements/ANNOVAR/annovar/&quot;, 
                                               annovar_input = &quot;no.txt&quot;, 
                                               spidex_mutrate_prefix = &quot;../other_annotation/spidex_database/mutrate_spedix_score_by_chr/lower_10pct_-1.416/spidex_public_noncommercial_v1_0_collapse_unique_base.extended_3bp.bed.fasta.trinuc.mutrate_with_spedix_gene.lower_10pct&quot;,
                                               splicing_cutoff = -1.416,
                                               log_splicing_rr = log(2.1),
                                               proband_SNV_mut_spidex = proband_SNV_mut_spidex)</code></pre>
<pre><code>## 
Read 0.0% of 8286737 rows
Read 20.9% of 8286737 rows
Read 28.8% of 8286737 rows
Read 47.8% of 8286737 rows
Read 77.1% of 8286737 rows
Read 78.4% of 8286737 rows
Read 85.3% of 8286737 rows
Read 92.2% of 8286737 rows
Read 99.3% of 8286737 rows
Read 8286737 rows and 8 (of 8) columns from 0.405 GB file in 00:00:13</code></pre>
<pre><code>## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.</code></pre>
<pre><code>## 
Read 34.3% of 8286737 rows
Read 98.6% of 8286737 rows
Read 8286737 rows and 4 (of 4) columns from 0.273 GB file in 00:00:04</code></pre>
<pre><code>## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.</code></pre>
<pre><code>## Warning in `[.data.table`(coverage, , `:=`(c(&quot;coding&quot;, &quot;promoter&quot;, &quot;nf&quot;, :
## with=FALSE ignored, it isn&#39;t needed when using :=. See ?&#39;:=&#39; for examples.</code></pre>
<pre><code>## 
Read 3.9% of 8286737 rows
Read 24.4% of 8286737 rows
Read 44.5% of 8286737 rows
Read 65.9% of 8286737 rows
Read 87.7% of 8286737 rows
Read 8286737 rows and 6 (of 6) columns from 0.410 GB file in 00:00:07</code></pre>
<pre><code>## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.</code></pre>
<pre><code>## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.</code></pre>
<pre><code>## 
Read 81.3% of 3197446 rows
Read 3197446 rows and 10 (of 10) columns from 0.298 GB file in 00:00:03</code></pre>
<pre><code>## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.

## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.

## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.

## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.

## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.

## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.

## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.

## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.

## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.

## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.

## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.

## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.

## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.

## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.

## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.

## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.

## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.

## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.

## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.

## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.

## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.

## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.

## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.

## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.

## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.

## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.

## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.

## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.

## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.

## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.

## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.

## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.

## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.

## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.

## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.

## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.

## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.

## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.

## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.

## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.

## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.

## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.

## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.

## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.

## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.

## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.

## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.

## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.

## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.

## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.

## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.

## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.

## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.

## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.

## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.

## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.

## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.

## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.

## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.

## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.

## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.

## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.

## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.

## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.

## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.

## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.

## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.

## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.

## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.

## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.

## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.

## Warning in `names&lt;-.data.table`(`*tmp*`, value = value): This is R&lt;3.1.0
## where colnames(x)&lt;-value deep copies the entire table (several times).
## Please upgrade to R&gt;=3.1.0 and see ?setnames which allows you to change
## names by name with built-in checks and warnings.</code></pre>
<pre class="r"><code>rm(new_control)
system(paste(&quot;rm &quot;,prefix,&quot;annovar_input_temp.bed&quot;,sep = &quot;&quot;))
system(paste(&quot;rm &quot;,prefix, &quot;temp.bed&quot;,sep = &quot;&quot;))</code></pre>
<pre class="r"><code>### currently the table below has duplicated entries, because there are a few genes that have appeared both on X chromosome and Y chromosome. 
### Right now, I first remove genes that appear on both chromosomes
splicing_BF_table_dup_removed &lt;- splicing_BF_table$splicing_BF_table[!duplicated(splicing_BF_table$splicing_BF_table$genename)]</code></pre>
<pre class="r"><code>TADA_table$gene_BF_table</code></pre>
<pre><code>##        genename logBF_noncoding logBF_coding logBF_all BF_noncoding
##     1:    SCN2A    -0.006766934    23.561577 23.554810    0.9932559
##     2:     CHD8    -0.014154934    22.713553 22.699398    0.9859448
##     3:  SYNGAP1    -0.053829740    15.412931 15.359101    0.9475934
##     4:   ARID1B    -0.032391010    14.856753 14.824362    0.9681280
##     5:   DYRK1A    -0.042500155    14.746632 14.704131    0.9583903
##    ---                                                             
## 18661:     PKD1    -0.059707314    -4.269452 -4.329160    0.9420402
## 18662:     PLEC    -0.157351908    -5.001259 -5.158611    0.8544033
## 18663:    FCGBP    -0.015437672    -5.176946 -5.192383    0.9846809
## 18664:      TTN     0.000000000    -5.905323 -5.905323    1.0000000
## 18665:    MUC16     0.000000000    -6.536660 -6.536660    1.0000000
##           BF_coding       BF_all   FDR_coding      FDR_all
##     1: 1.708689e+10 1.697165e+10 9.168826e-10 9.231080e-10
##     2: 7.317632e+09 7.214781e+09 1.528915e-09 1.547288e-09
##     3: 4.940272e+06 4.681369e+06 1.058088e-06 1.116561e-06
##     4: 2.832734e+06 2.742449e+06 2.176203e-06 2.265576e-06
##     5: 2.537352e+06 2.431774e+06 2.975838e-06 3.100950e-06
##    ---                                                    
## 18661: 1.398944e-02 1.317862e-02 9.412171e-01 9.412750e-01
## 18662: 6.729469e-03 5.749681e-03 9.412202e-01 9.412781e-01
## 18663: 5.645222e-03 5.558742e-03 9.412233e-01 9.412813e-01
## 18664: 2.724902e-03 2.724902e-03 9.412265e-01 9.412844e-01
## 18665: 1.449321e-03 1.449321e-03 9.412296e-01 9.412875e-01</code></pre>
</div>
<div id="now-combine-information-from-splicing-to-other-non_coding-mutations" class="section level2">
<h2><span class="header-section-number">3.3</span> <em>now combine information from splicing to other non_coding mutations</em></h2>
<pre class="r"><code>combined_BF_table &lt;- splicing_BF_table_dup_removed[TADA_table$gene_BF_table, on = &quot;genename&quot;]
combined_BF_table</code></pre>
<pre><code>##        genename logBF_splicing BF_splicing logBF_noncoding logBF_coding
##     1:    SCN2A   -0.015087809   0.9850254    -0.006766934    23.561577
##     2:     CHD8   -0.013165998   0.9869203    -0.014154934    22.713553
##     3:  SYNGAP1   -0.012624522   0.9874548    -0.053829740    15.412931
##     4:   ARID1B   -0.013266887   0.9868207    -0.032391010    14.856753
##     5:   DYRK1A   -0.008601051   0.9914358    -0.042500155    14.746632
##    ---                                                                 
## 18661:     PKD1   -0.021357969   0.9788685    -0.059707314    -4.269452
## 18662:     PLEC   -0.015255892   0.9848599    -0.157351908    -5.001259
## 18663:    FCGBP   -0.008248457   0.9917855    -0.015437672    -5.176946
## 18664:      TTN   -0.219350162   0.8030405     0.000000000    -5.905323
## 18665:    MUC16   -0.054426911   0.9470277     0.000000000    -6.536660
##        logBF_all BF_noncoding    BF_coding       BF_all   FDR_coding
##     1: 23.554810    0.9932559 1.708689e+10 1.697165e+10 9.168826e-10
##     2: 22.699398    0.9859448 7.317632e+09 7.214781e+09 1.528915e-09
##     3: 15.359101    0.9475934 4.940272e+06 4.681369e+06 1.058088e-06
##     4: 14.824362    0.9681280 2.832734e+06 2.742449e+06 2.176203e-06
##     5: 14.704131    0.9583903 2.537352e+06 2.431774e+06 2.975838e-06
##    ---                                                              
## 18661: -4.329160    0.9420402 1.398944e-02 1.317862e-02 9.412171e-01
## 18662: -5.158611    0.8544033 6.729469e-03 5.749681e-03 9.412202e-01
## 18663: -5.192383    0.9846809 5.645222e-03 5.558742e-03 9.412233e-01
## 18664: -5.905323    1.0000000 2.724902e-03 2.724902e-03 9.412265e-01
## 18665: -6.536660    1.0000000 1.449321e-03 1.449321e-03 9.412296e-01
##             FDR_all
##     1: 9.231080e-10
##     2: 1.547288e-09
##     3: 1.116561e-06
##     4: 2.265576e-06
##     5: 3.100950e-06
##    ---             
## 18661: 9.412750e-01
## 18662: 9.412781e-01
## 18663: 9.412813e-01
## 18664: 9.412844e-01
## 18665: 9.412875e-01</code></pre>
<pre class="r"><code># get a list of genes that don&#39;t have splicing info.
genes_with_no_info_from_splicing_BF_table &lt;- combined_BF_table[is.na(combined_BF_table$logBF_splicing),]$genename

# change the logBF_splicing of these genes to 0 and BF of these genes to 1
combined_BF_table[is.na(combined_BF_table$logBF_splicing),][,c(&quot;logBF_splicing&quot;)] = 0
combined_BF_table[is.na(combined_BF_table$BF_splicing),][,c(&quot;BF_splicing&quot;)] = 1

combined_BF_table[is.element(combined_BF_table$genename, genes_with_no_info_from_splicing_BF_table),]</code></pre>
<pre><code>##       genename logBF_splicing BF_splicing logBF_noncoding logBF_coding
##    1:     MLL3              0           1      0.00000000     5.368897
##    2:     MLL5              0           1      0.00000000     4.721863
##    3:     MFRP              0           1     -0.01754074     3.984587
##    4:     CCIN              0           1     -0.02681743     3.931569
##    5:  IRF2BPL              0           1     -0.08355408     3.470252
##   ---                                                                 
## 2943:    GPR98              0           1      0.00000000    -2.599270
## 2944:    EPPK1              0           1     -0.08886584    -2.573861
## 2945:  PRIC285              0           1      0.00000000    -2.703290
## 2946:   ZNF469              0           1     -0.01039400    -2.763213
## 2947:     MLL2              0           1      0.00000000    -2.906505
##       logBF_all BF_noncoding    BF_coding       BF_all FDR_coding
##    1:  5.368897    1.0000000 214.62602200 214.62602200 0.02174861
##    2:  4.721863    1.0000000 112.37739057 112.37739057 0.04279135
##    3:  3.967047    0.9826122  53.76310336  52.82828156 0.08571576
##    4:  3.904751    0.9735390  50.98689831  49.63773219 0.09362261
##    5:  3.386698    0.9198413  32.14482920  29.56814282 0.14422228
##   ---                                                            
## 2943: -2.599270    1.0000000   0.07432784   0.07432784 0.94111490
## 2944: -2.662726    0.9149683   0.07624065   0.06975778 0.94110618
## 2945: -2.703290    1.0000000   0.06698477   0.06698477 0.94112368
## 2946: -2.773607    0.9896598   0.06308872   0.06243637 0.94112956
## 2947: -2.906505    1.0000000   0.05466643   0.05466643 0.94115027
##          FDR_all
##    1: 0.02255157
##    2: 0.04192966
##    3: 0.09149403
##    4: 0.10120764
##    5: 0.15439946
##   ---           
## 2943: 0.94116712
## 2944: 0.94117878
## 2945: 0.94118463
## 2946: 0.94119051
## 2947: 0.94120826</code></pre>
<pre class="r"><code># add splicing info to BF_all
combined_BF_table$logBF_all &lt;- combined_BF_table$logBF_splicing + combined_BF_table$logBF_noncoding + combined_BF_table$logBF_coding
combined_BF_table$BF_all &lt;- exp(combined_BF_table$logBF_all)
combined_BF_table &lt;- combined_BF_table[order(combined_BF_table$BF_all, decreasing = TRUE),]
combined_BF_table$FDR_all = Bayesian.FDR(combined_BF_table$BF_all, 0.94)$FDR # here 0.94 was chosen to be consisten with the prior used for getting BF and FDR using other types of mutations. </code></pre>
<div id="number-of-gained-and-lost-genes-after-having-non-coding-mutations-and-splicing-mutations" class="section level3">
<h3><span class="header-section-number">3.3.1</span> Number of gained and lost genes after having non-coding mutations and splicing mutations</h3>
<pre class="r"><code>TADA_table_filtered =combined_BF_table[!is.element(combined_BF_table$genename, TADA_table$genes_with_no_epi),]
knitr::kable(performance_table(TADA_table_filtered))</code></pre>
<table>
<thead>
<tr class="header">
<th></th>
<th align="right">novel</th>
<th align="right">consistent</th>
<th align="right">lost</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>q0.1</td>
<td align="right">3</td>
<td align="right">55</td>
<td align="right">2</td>
</tr>
<tr class="even">
<td>q0.2</td>
<td align="right">11</td>
<td align="right">87</td>
<td align="right">6</td>
</tr>
<tr class="odd">
<td>q0.3</td>
<td align="right">11</td>
<td align="right">126</td>
<td align="right">6</td>
</tr>
</tbody>
</table>
</div>
<div id="gene-set-enrichment-analysis-fdr-0.1-1" class="section level3">
<h3><span class="header-section-number">3.3.2</span> Gene set enrichment analysis FDR 0.1</h3>
<pre class="r"><code>knitr::kable(foe_table(TADA_table_filtered[TADA_table_filtered$FDR_coding &gt;=0.1 &amp; TADA_table_filtered$FDR_all &lt; 0.1, ]$genename, TADA_table_filtered$genename))</code></pre>
<table>
<thead>
<tr class="header">
<th></th>
<th align="right">HI</th>
<th align="right">FMRP</th>
<th align="right">RVIS top 25%</th>
<th align="right">Brainspan top 25%</th>
<th align="right">Brain GO</th>
<th align="right">presynaptic</th>
<th align="right">postsynaptic</th>
<th align="right">ExAC top 25%</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Fold of enrichment</td>
<td align="right">9.6533090</td>
<td align="right">6.6578727</td>
<td align="right">4.007815</td>
<td align="right">3.5050705</td>
<td align="right">2.1072233</td>
<td align="right">16.5098650</td>
<td align="right">3.1211229</td>
<td align="right">2.5158636</td>
</tr>
<tr class="even">
<td>pvalue</td>
<td align="right">0.0011098</td>
<td align="right">0.1428123</td>
<td align="right">0.015525</td>
<td align="right">0.0232116</td>
<td align="right">0.4034691</td>
<td align="right">0.0593588</td>
<td align="right">0.2874136</td>
<td align="right">0.1734212</td>
</tr>
</tbody>
</table>
</div>
<div id="gene-set-enrichment-analysis-fdr-0.2-1" class="section level3">
<h3><span class="header-section-number">3.3.3</span> Gene set enrichment analysis FDR 0.2</h3>
<pre class="r"><code>knitr::kable(foe_table(TADA_table_filtered[TADA_table_filtered$FDR_coding &gt;=0.2 &amp; TADA_table_filtered$FDR_all &lt; 0.2, ]$genename, TADA_table_filtered$genename))</code></pre>
<table>
<thead>
<tr class="header">
<th></th>
<th align="right">HI</th>
<th align="right">FMRP</th>
<th align="right">RVIS top 25%</th>
<th align="right">Brainspan top 25%</th>
<th align="right">Brain GO</th>
<th align="right">presynaptic</th>
<th align="right">postsynaptic</th>
<th align="right">ExAC top 25%</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Fold of enrichment</td>
<td align="right">2.6327206</td>
<td align="right">1.8157835</td>
<td align="right">1.8217338</td>
<td align="right">1.2745711</td>
<td align="right">1.1493945</td>
<td align="right">4.5026905</td>
<td align="right">1.7024307</td>
<td align="right">1.3722892</td>
</tr>
<tr class="even">
<td>pvalue</td>
<td align="right">0.0972986</td>
<td align="right">0.4317384</td>
<td align="right">0.1137694</td>
<td align="right">0.3873558</td>
<td align="right">0.5386502</td>
<td align="right">0.2010312</td>
<td align="right">0.3316037</td>
<td align="right">0.3286614</td>
</tr>
</tbody>
</table>
</div>
<div id="gene-set-enrichment-analysis-fdr-0.3-1" class="section level3">
<h3><span class="header-section-number">3.3.4</span> Gene set enrichment analysis FDR 0.3</h3>
<pre class="r"><code>knitr::kable(foe_table(TADA_table_filtered[TADA_table_filtered$FDR_coding &gt;=0.3 &amp; TADA_table_filtered$FDR_all &lt; 0.3, ]$genename, TADA_table_filtered$genename))</code></pre>
<table>
<thead>
<tr class="header">
<th></th>
<th align="right">HI</th>
<th align="right">FMRP</th>
<th align="right">RVIS top 25%</th>
<th align="right">Brainspan top 25%</th>
<th align="right">Brain GO</th>
<th align="right">presynaptic</th>
<th align="right">postsynaptic</th>
<th align="right">ExAC top 25%</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Fold of enrichment</td>
<td align="right">2.6327206</td>
<td align="right">3.6315669</td>
<td align="right">1.8217338</td>
<td align="right">1.9118567</td>
<td align="right">1.7240918</td>
<td align="right">4.5026905</td>
<td align="right">0.8512153</td>
<td align="right">1.3722892</td>
</tr>
<tr class="even">
<td>pvalue</td>
<td align="right">0.0972986</td>
<td align="right">0.1020691</td>
<td align="right">0.1137694</td>
<td align="right">0.0626662</td>
<td align="right">0.2463658</td>
<td align="right">0.2010312</td>
<td align="right">0.7114238</td>
<td align="right">0.3286614</td>
</tr>
</tbody>
</table>
</div>
<div id="fdr-0.3-new-genes-1" class="section level3">
<h3><span class="header-section-number">3.3.5</span> FDR 0.3 new genes</h3>
<pre class="r"><code>knitr::kable(TADA_table_filtered[TADA_table_filtered$FDR_coding &gt;= 0.3 &amp; TADA_table_filtered$FDR_all &lt; 0.3,])</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">genename</th>
<th align="right">logBF_splicing</th>
<th align="right">BF_splicing</th>
<th align="right">logBF_noncoding</th>
<th align="right">logBF_coding</th>
<th align="right">logBF_all</th>
<th align="right">BF_noncoding</th>
<th align="right">BF_coding</th>
<th align="right">BF_all</th>
<th align="right">FDR_coding</th>
<th align="right">FDR_all</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">JUP</td>
<td align="right">-0.0064879</td>
<td align="right">0.9935331</td>
<td align="right">2.5872789</td>
<td align="right">2.131726</td>
<td align="right">4.712517</td>
<td align="right">13.2935488</td>
<td align="right">8.429401</td>
<td align="right">111.33199</td>
<td align="right">0.4713748</td>
<td align="right">0.0413829</td>
</tr>
<tr class="even">
<td align="left">APBB1</td>
<td align="right">-0.0052030</td>
<td align="right">0.9948105</td>
<td align="right">1.8901033</td>
<td align="right">2.325413</td>
<td align="right">4.210314</td>
<td align="right">6.6200523</td>
<td align="right">10.230909</td>
<td align="right">67.37767</td>
<td align="right">0.4502329</td>
<td align="right">0.0659406</td>
</tr>
<tr class="odd">
<td align="left">HGF</td>
<td align="right">0.7342728</td>
<td align="right">2.0839660</td>
<td align="right">-0.0049385</td>
<td align="right">2.447064</td>
<td align="right">3.176398</td>
<td align="right">0.9950737</td>
<td align="right">11.554370</td>
<td align="right">23.96029</td>
<td align="right">0.4257441</td>
<td align="right">0.1822509</td>
</tr>
<tr class="even">
<td align="left">MSL2</td>
<td align="right">0.0000000</td>
<td align="right">1.0000000</td>
<td align="right">0.6235923</td>
<td align="right">2.536128</td>
<td align="right">3.159721</td>
<td align="right">1.8656178</td>
<td align="right">12.630673</td>
<td align="right">23.56401</td>
<td align="right">0.3972018</td>
<td align="right">0.1866457</td>
</tr>
<tr class="odd">
<td align="left">PPM1D</td>
<td align="right">-0.0007843</td>
<td align="right">0.9992160</td>
<td align="right">0.6246737</td>
<td align="right">2.526200</td>
<td align="right">3.150089</td>
<td align="right">1.8676365</td>
<td align="right">12.505890</td>
<td align="right">23.33815</td>
<td align="right">0.4001637</td>
<td align="right">0.1888176</td>
</tr>
<tr class="even">
<td align="left">SLC38A3</td>
<td align="right">-0.0049565</td>
<td align="right">0.9950557</td>
<td align="right">0.5877401</td>
<td align="right">2.510331</td>
<td align="right">3.093115</td>
<td align="right">1.7999162</td>
<td align="right">12.309005</td>
<td align="right">22.04564</td>
<td align="right">0.4074689</td>
<td align="right">0.1932555</td>
</tr>
<tr class="odd">
<td align="left">ARHGAP44</td>
<td align="right">0.7308175</td>
<td align="right">2.0767777</td>
<td align="right">-0.0112135</td>
<td align="right">2.362564</td>
<td align="right">3.082168</td>
<td align="right">0.9888491</td>
<td align="right">10.618144</td>
<td align="right">21.80563</td>
<td align="right">0.4447959</td>
<td align="right">0.1976005</td>
</tr>
<tr class="even">
<td align="left">LRRC4</td>
<td align="right">0.0000000</td>
<td align="right">1.0000000</td>
<td align="right">0.6103642</td>
<td align="right">2.344952</td>
<td align="right">2.955317</td>
<td align="right">1.8411018</td>
<td align="right">10.432776</td>
<td align="right">19.20780</td>
<td align="right">0.4477285</td>
<td align="right">0.2105404</td>
</tr>
<tr class="odd">
<td align="left">PLXNB1</td>
<td align="right">-0.0173372</td>
<td align="right">0.9828122</td>
<td align="right">0.6101821</td>
<td align="right">2.352516</td>
<td align="right">2.945361</td>
<td align="right">1.8407665</td>
<td align="right">10.511988</td>
<td align="right">19.01753</td>
<td align="right">0.4468929</td>
<td align="right">0.2148734</td>
</tr>
<tr class="even">
<td align="left">CBX4</td>
<td align="right">-0.0031636</td>
<td align="right">0.9968414</td>
<td align="right">0.5351793</td>
<td align="right">2.237874</td>
<td align="right">2.769890</td>
<td align="right">1.7077544</td>
<td align="right">9.373381</td>
<td align="right">15.95687</td>
<td align="right">0.4613862</td>
<td align="right">0.2798614</td>
</tr>
<tr class="odd">
<td align="left">SIRPB2</td>
<td align="right">-0.0009817</td>
<td align="right">0.9990188</td>
<td align="right">-0.0047030</td>
<td align="right">2.736432</td>
<td align="right">2.730747</td>
<td align="right">0.9953080</td>
<td align="right">15.431826</td>
<td align="right">15.34435</td>
<td align="right">0.3009029</td>
<td align="right">0.2951882</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="cacluate-the-bayes-factor-table-for-splicing-mutations-summing-mutation-rate-from-chrx-and-chry-for-a-gene-if-it-is-on-both-chrx-and-chry" class="section level2">
<h2><span class="header-section-number">3.4</span> <em>cacluate the bayes factor table for splicing mutations, summing mutation rate from ChrX and ChrY for a gene if it is on both ChrX and ChrY</em></h2>
<pre class="r"><code>### currently the table below has duplicated entries, because there are a few genes that have appeared both on X chromosome and Y chromosome. 
### Now I sum up the matation rates from both chrx and chrY for such genes. 

head(splicing_BF_table$splicing_BF_table)</code></pre>
<pre><code>##    genename logBF_splicing BF_splicing
## 1:   SAMD11  -0.0105229458   0.9895322
## 2:    NOC2L  -0.0099645541   0.9900849
## 3:   KLHL17  -0.0060241772   0.9939939
## 4:  PLEKHN1  -0.0105621082   0.9894935
## 5:    PERM1  -0.0005756045   0.9994246
## 6:     HES4  -0.0015764676   0.9984248</code></pre>
<pre class="r"><code>splicing_BF_table_collapsed &lt;- by(splicing_BF_table$splicing_BF_table, splicing_BF_table$splicing_BF_table$genename, function(x) sum(x$logBF_splicing)) 
#splicing_BF_table_collapsed 
splicing_BF_table_collapsed &lt;- data.table(genename = names(splicing_BF_table_collapsed), logBF_splicing = as.vector(splicing_BF_table_collapsed), BF_splicing = exp(as.vector(splicing_BF_table_collapsed)))</code></pre>
</div>
<div id="now-combine-information-from-splicing-collapsed-version-to-other-non_coding-mutations" class="section level2">
<h2><span class="header-section-number">3.5</span> <em>now combine information from splicing (collapsed version) to other non_coding mutations</em></h2>
<pre class="r"><code>combined_BF_table &lt;- splicing_BF_table_collapsed[TADA_table$gene_BF_table, on = &quot;genename&quot;]
combined_BF_table</code></pre>
<pre><code>##        genename logBF_splicing BF_splicing logBF_noncoding logBF_coding
##     1:    SCN2A   -0.015087809   0.9850254    -0.006766934    23.561577
##     2:     CHD8   -0.013165998   0.9869203    -0.014154934    22.713553
##     3:  SYNGAP1   -0.012624522   0.9874548    -0.053829740    15.412931
##     4:   ARID1B   -0.013266887   0.9868207    -0.032391010    14.856753
##     5:   DYRK1A   -0.008601051   0.9914358    -0.042500155    14.746632
##    ---                                                                 
## 18661:     PKD1   -0.021357969   0.9788685    -0.059707314    -4.269452
## 18662:     PLEC   -0.015255892   0.9848599    -0.157351908    -5.001259
## 18663:    FCGBP   -0.008248457   0.9917855    -0.015437672    -5.176946
## 18664:      TTN   -0.219350162   0.8030405     0.000000000    -5.905323
## 18665:    MUC16   -0.054426911   0.9470277     0.000000000    -6.536660
##        logBF_all BF_noncoding    BF_coding       BF_all   FDR_coding
##     1: 23.554810    0.9932559 1.708689e+10 1.697165e+10 9.168826e-10
##     2: 22.699398    0.9859448 7.317632e+09 7.214781e+09 1.528915e-09
##     3: 15.359101    0.9475934 4.940272e+06 4.681369e+06 1.058088e-06
##     4: 14.824362    0.9681280 2.832734e+06 2.742449e+06 2.176203e-06
##     5: 14.704131    0.9583903 2.537352e+06 2.431774e+06 2.975838e-06
##    ---                                                              
## 18661: -4.329160    0.9420402 1.398944e-02 1.317862e-02 9.412171e-01
## 18662: -5.158611    0.8544033 6.729469e-03 5.749681e-03 9.412202e-01
## 18663: -5.192383    0.9846809 5.645222e-03 5.558742e-03 9.412233e-01
## 18664: -5.905323    1.0000000 2.724902e-03 2.724902e-03 9.412265e-01
## 18665: -6.536660    1.0000000 1.449321e-03 1.449321e-03 9.412296e-01
##             FDR_all
##     1: 9.231080e-10
##     2: 1.547288e-09
##     3: 1.116561e-06
##     4: 2.265576e-06
##     5: 3.100950e-06
##    ---             
## 18661: 9.412750e-01
## 18662: 9.412781e-01
## 18663: 9.412813e-01
## 18664: 9.412844e-01
## 18665: 9.412875e-01</code></pre>
<pre class="r"><code># get a list of genes that don&#39;t have splicing info.
genes_with_no_info_from_splicing_BF_table &lt;- combined_BF_table[is.na(combined_BF_table$logBF_splicing),]$genename

# change the logBF_splicing of these genes to 0 and BF of these genes to 1
combined_BF_table[is.na(combined_BF_table$logBF_splicing),][,c(&quot;logBF_splicing&quot;)] = 0
combined_BF_table[is.na(combined_BF_table$BF_splicing),][,c(&quot;BF_splicing&quot;)] = 1

combined_BF_table[is.element(combined_BF_table$genename, genes_with_no_info_from_splicing_BF_table),]</code></pre>
<pre><code>##       genename logBF_splicing BF_splicing logBF_noncoding logBF_coding
##    1:     MLL3              0           1      0.00000000     5.368897
##    2:     MLL5              0           1      0.00000000     4.721863
##    3:     MFRP              0           1     -0.01754074     3.984587
##    4:     CCIN              0           1     -0.02681743     3.931569
##    5:  IRF2BPL              0           1     -0.08355408     3.470252
##   ---                                                                 
## 2943:    GPR98              0           1      0.00000000    -2.599270
## 2944:    EPPK1              0           1     -0.08886584    -2.573861
## 2945:  PRIC285              0           1      0.00000000    -2.703290
## 2946:   ZNF469              0           1     -0.01039400    -2.763213
## 2947:     MLL2              0           1      0.00000000    -2.906505
##       logBF_all BF_noncoding    BF_coding       BF_all FDR_coding
##    1:  5.368897    1.0000000 214.62602200 214.62602200 0.02174861
##    2:  4.721863    1.0000000 112.37739057 112.37739057 0.04279135
##    3:  3.967047    0.9826122  53.76310336  52.82828156 0.08571576
##    4:  3.904751    0.9735390  50.98689831  49.63773219 0.09362261
##    5:  3.386698    0.9198413  32.14482920  29.56814282 0.14422228
##   ---                                                            
## 2943: -2.599270    1.0000000   0.07432784   0.07432784 0.94111490
## 2944: -2.662726    0.9149683   0.07624065   0.06975778 0.94110618
## 2945: -2.703290    1.0000000   0.06698477   0.06698477 0.94112368
## 2946: -2.773607    0.9896598   0.06308872   0.06243637 0.94112956
## 2947: -2.906505    1.0000000   0.05466643   0.05466643 0.94115027
##          FDR_all
##    1: 0.02255157
##    2: 0.04192966
##    3: 0.09149403
##    4: 0.10120764
##    5: 0.15439946
##   ---           
## 2943: 0.94116712
## 2944: 0.94117878
## 2945: 0.94118463
## 2946: 0.94119051
## 2947: 0.94120826</code></pre>
<pre class="r"><code># add splicing info to BF_all
combined_BF_table$logBF_all &lt;- combined_BF_table$logBF_splicing + combined_BF_table$logBF_noncoding + combined_BF_table$logBF_coding
combined_BF_table$BF_all &lt;- exp(combined_BF_table$logBF_all)
combined_BF_table &lt;- combined_BF_table[order(combined_BF_table$BF_all, decreasing = TRUE),]
combined_BF_table$FDR_all = Bayesian.FDR(combined_BF_table$BF_all, 0.94)$FDR # here 0.94 was chosen to be consisten with the prior used for getting BF and FDR using other types of mutations. 

head(combined_BF_table)</code></pre>
<pre><code>##    genename logBF_splicing BF_splicing logBF_noncoding logBF_coding
## 1:    SCN2A   -0.015087809   0.9850254    -0.006766934     23.56158
## 2:     CHD8   -0.013165998   0.9869203    -0.014154934     22.71355
## 3:  SYNGAP1   -0.012624522   0.9874548    -0.053829740     15.41293
## 4:   ARID1B   -0.013266887   0.9868207    -0.032391010     14.85675
## 5:   DYRK1A   -0.008601051   0.9914358    -0.042500155     14.74663
## 6:    KDM5B   -0.009239517   0.9908030    -0.053723033     12.62463
##    logBF_all BF_noncoding    BF_coding       BF_all   FDR_coding
## 1:  23.53972    0.9932559 1.708689e+10 1.671751e+10 9.168826e-10
## 2:  22.68623    0.9859448 7.317632e+09 7.120414e+09 1.528915e-09
## 3:  15.34648    0.9475934 4.940272e+06 4.622641e+06 1.058088e-06
## 4:  14.81110    0.9681280 2.832734e+06 2.706306e+06 2.176203e-06
## 5:  14.69553    0.9583903 2.537352e+06 2.410947e+06 2.975838e-06
## 6:  12.56167    0.9476946 3.039542e+05 2.854065e+05 1.106990e-05
##         FDR_all
## 1: 9.371414e-10
## 2: 1.568694e-09
## 3: 1.130747e-06
## 4: 2.295289e-06
## 5: 3.135850e-06
## 6: 1.176145e-05</code></pre>
<div id="number-of-gained-and-lost-genes-after-having-non-coding-mutations-and-splicing-mutations-1" class="section level3">
<h3><span class="header-section-number">3.5.1</span> Number of gained and lost genes after having non-coding mutations and splicing mutations</h3>
<pre class="r"><code>TADA_table_filtered =combined_BF_table[!is.element(combined_BF_table$genename, TADA_table$genes_with_no_epi),]
knitr::kable(performance_table(TADA_table_filtered))</code></pre>
<table>
<thead>
<tr class="header">
<th></th>
<th align="right">novel</th>
<th align="right">consistent</th>
<th align="right">lost</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>q0.1</td>
<td align="right">3</td>
<td align="right">55</td>
<td align="right">2</td>
</tr>
<tr class="even">
<td>q0.2</td>
<td align="right">11</td>
<td align="right">87</td>
<td align="right">6</td>
</tr>
<tr class="odd">
<td>q0.3</td>
<td align="right">11</td>
<td align="right">126</td>
<td align="right">6</td>
</tr>
</tbody>
</table>
</div>
<div id="fdr-0.1-new-genes" class="section level3">
<h3><span class="header-section-number">3.5.2</span> FDR 0.1 new genes</h3>
<pre class="r"><code>knitr::kable(TADA_table_filtered[TADA_table_filtered$FDR_coding &gt;= 0.1 &amp; TADA_table_filtered$FDR_all &lt; 0.1,])</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">genename</th>
<th align="right">logBF_splicing</th>
<th align="right">BF_splicing</th>
<th align="right">logBF_noncoding</th>
<th align="right">logBF_coding</th>
<th align="right">logBF_all</th>
<th align="right">BF_noncoding</th>
<th align="right">BF_coding</th>
<th align="right">BF_all</th>
<th align="right">FDR_coding</th>
<th align="right">FDR_all</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">JUP</td>
<td align="right">-0.0064879</td>
<td align="right">0.9935331</td>
<td align="right">2.5872789</td>
<td align="right">2.131726</td>
<td align="right">4.712517</td>
<td align="right">13.293549</td>
<td align="right">8.429401</td>
<td align="right">111.33199</td>
<td align="right">0.4713748</td>
<td align="right">0.0413829</td>
</tr>
<tr class="even">
<td align="left">APBB1</td>
<td align="right">-0.0052030</td>
<td align="right">0.9948105</td>
<td align="right">1.8901033</td>
<td align="right">2.325413</td>
<td align="right">4.210314</td>
<td align="right">6.620052</td>
<td align="right">10.230909</td>
<td align="right">67.37767</td>
<td align="right">0.4502329</td>
<td align="right">0.0659406</td>
</tr>
<tr class="odd">
<td align="left">ARHGAP5</td>
<td align="right">-0.0030892</td>
<td align="right">0.9969156</td>
<td align="right">0.6316236</td>
<td align="right">3.434124</td>
<td align="right">4.062658</td>
<td align="right">1.880661</td>
<td align="right">31.004243</td>
<td align="right">58.12864</td>
<td align="right">0.1466769</td>
<td align="right">0.0840828</td>
</tr>
</tbody>
</table>
</div>
<div id="fdr-0.2-new-genes" class="section level3">
<h3><span class="header-section-number">3.5.3</span> FDR 0.2 new genes</h3>
<pre class="r"><code>knitr::kable(TADA_table_filtered[TADA_table_filtered$FDR_coding &gt;= 0.2 &amp; TADA_table_filtered$FDR_all &lt; 0.2,])</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">genename</th>
<th align="right">logBF_splicing</th>
<th align="right">BF_splicing</th>
<th align="right">logBF_noncoding</th>
<th align="right">logBF_coding</th>
<th align="right">logBF_all</th>
<th align="right">BF_noncoding</th>
<th align="right">BF_coding</th>
<th align="right">BF_all</th>
<th align="right">FDR_coding</th>
<th align="right">FDR_all</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">JUP</td>
<td align="right">-0.0064879</td>
<td align="right">0.9935331</td>
<td align="right">2.5872789</td>
<td align="right">2.131726</td>
<td align="right">4.712517</td>
<td align="right">13.2935488</td>
<td align="right">8.429401</td>
<td align="right">111.33199</td>
<td align="right">0.4713748</td>
<td align="right">0.0413829</td>
</tr>
<tr class="even">
<td align="left">APBB1</td>
<td align="right">-0.0052030</td>
<td align="right">0.9948105</td>
<td align="right">1.8901033</td>
<td align="right">2.325413</td>
<td align="right">4.210314</td>
<td align="right">6.6200523</td>
<td align="right">10.230909</td>
<td align="right">67.37767</td>
<td align="right">0.4502329</td>
<td align="right">0.0659406</td>
</tr>
<tr class="odd">
<td align="left">PNPLA7</td>
<td align="right">0.7138975</td>
<td align="right">2.0419342</td>
<td align="right">-0.0117036</td>
<td align="right">2.863029</td>
<td align="right">3.565223</td>
<td align="right">0.9883646</td>
<td align="right">17.514503</td>
<td align="right">35.34734</td>
<td align="right">0.2273822</td>
<td align="right">0.1358965</td>
</tr>
<tr class="even">
<td align="left">DOCK5</td>
<td align="right">-0.0192641</td>
<td align="right">0.9809203</td>
<td align="right">0.6180382</td>
<td align="right">2.897630</td>
<td align="right">3.496404</td>
<td align="right">1.8552847</td>
<td align="right">18.131119</td>
<td align="right">32.99658</td>
<td align="right">0.2134403</td>
<td align="right">0.1472411</td>
</tr>
<tr class="odd">
<td align="left">PLA2G16</td>
<td align="right">-0.0026059</td>
<td align="right">0.9973975</td>
<td align="right">0.5977041</td>
<td align="right">2.828890</td>
<td align="right">3.423988</td>
<td align="right">1.8179402</td>
<td align="right">16.926657</td>
<td align="right">30.69157</td>
<td align="right">0.2501432</td>
<td align="right">0.1560570</td>
</tr>
<tr class="even">
<td align="left">NME9</td>
<td align="right">-0.0063909</td>
<td align="right">0.9936295</td>
<td align="right">0.6415416</td>
<td align="right">2.787033</td>
<td align="right">3.422184</td>
<td align="right">1.8994068</td>
<td align="right">16.232789</td>
<td align="right">30.63625</td>
<td align="right">0.2762552</td>
<td align="right">0.1582017</td>
</tr>
<tr class="odd">
<td align="left">HGF</td>
<td align="right">0.7342728</td>
<td align="right">2.0839660</td>
<td align="right">-0.0049385</td>
<td align="right">2.447064</td>
<td align="right">3.176398</td>
<td align="right">0.9950737</td>
<td align="right">11.554370</td>
<td align="right">23.96029</td>
<td align="right">0.4257441</td>
<td align="right">0.1822509</td>
</tr>
<tr class="even">
<td align="left">MSL2</td>
<td align="right">0.0000000</td>
<td align="right">1.0000000</td>
<td align="right">0.6235923</td>
<td align="right">2.536128</td>
<td align="right">3.159721</td>
<td align="right">1.8656178</td>
<td align="right">12.630673</td>
<td align="right">23.56401</td>
<td align="right">0.3972018</td>
<td align="right">0.1866457</td>
</tr>
<tr class="odd">
<td align="left">PPM1D</td>
<td align="right">-0.0007843</td>
<td align="right">0.9992160</td>
<td align="right">0.6246737</td>
<td align="right">2.526200</td>
<td align="right">3.150089</td>
<td align="right">1.8676365</td>
<td align="right">12.505890</td>
<td align="right">23.33815</td>
<td align="right">0.4001637</td>
<td align="right">0.1888176</td>
</tr>
<tr class="even">
<td align="left">SLC38A3</td>
<td align="right">-0.0049565</td>
<td align="right">0.9950557</td>
<td align="right">0.5877401</td>
<td align="right">2.510331</td>
<td align="right">3.093115</td>
<td align="right">1.7999162</td>
<td align="right">12.309005</td>
<td align="right">22.04564</td>
<td align="right">0.4074689</td>
<td align="right">0.1932555</td>
</tr>
<tr class="odd">
<td align="left">ARHGAP44</td>
<td align="right">0.7308175</td>
<td align="right">2.0767777</td>
<td align="right">-0.0112135</td>
<td align="right">2.362564</td>
<td align="right">3.082168</td>
<td align="right">0.9888491</td>
<td align="right">10.618144</td>
<td align="right">21.80563</td>
<td align="right">0.4447959</td>
<td align="right">0.1976005</td>
</tr>
</tbody>
</table>
</div>
<div id="fdr-0.3-new-genes-2" class="section level3">
<h3><span class="header-section-number">3.5.4</span> FDR 0.3 new genes</h3>
<pre class="r"><code>knitr::kable(TADA_table_filtered[TADA_table_filtered$FDR_coding &gt;= 0.3 &amp; TADA_table_filtered$FDR_all &lt; 0.3,])</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">genename</th>
<th align="right">logBF_splicing</th>
<th align="right">BF_splicing</th>
<th align="right">logBF_noncoding</th>
<th align="right">logBF_coding</th>
<th align="right">logBF_all</th>
<th align="right">BF_noncoding</th>
<th align="right">BF_coding</th>
<th align="right">BF_all</th>
<th align="right">FDR_coding</th>
<th align="right">FDR_all</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">JUP</td>
<td align="right">-0.0064879</td>
<td align="right">0.9935331</td>
<td align="right">2.5872789</td>
<td align="right">2.131726</td>
<td align="right">4.712517</td>
<td align="right">13.2935488</td>
<td align="right">8.429401</td>
<td align="right">111.33199</td>
<td align="right">0.4713748</td>
<td align="right">0.0413829</td>
</tr>
<tr class="even">
<td align="left">APBB1</td>
<td align="right">-0.0052030</td>
<td align="right">0.9948105</td>
<td align="right">1.8901033</td>
<td align="right">2.325413</td>
<td align="right">4.210314</td>
<td align="right">6.6200523</td>
<td align="right">10.230909</td>
<td align="right">67.37767</td>
<td align="right">0.4502329</td>
<td align="right">0.0659406</td>
</tr>
<tr class="odd">
<td align="left">HGF</td>
<td align="right">0.7342728</td>
<td align="right">2.0839660</td>
<td align="right">-0.0049385</td>
<td align="right">2.447064</td>
<td align="right">3.176398</td>
<td align="right">0.9950737</td>
<td align="right">11.554370</td>
<td align="right">23.96029</td>
<td align="right">0.4257441</td>
<td align="right">0.1822509</td>
</tr>
<tr class="even">
<td align="left">MSL2</td>
<td align="right">0.0000000</td>
<td align="right">1.0000000</td>
<td align="right">0.6235923</td>
<td align="right">2.536128</td>
<td align="right">3.159721</td>
<td align="right">1.8656178</td>
<td align="right">12.630673</td>
<td align="right">23.56401</td>
<td align="right">0.3972018</td>
<td align="right">0.1866457</td>
</tr>
<tr class="odd">
<td align="left">PPM1D</td>
<td align="right">-0.0007843</td>
<td align="right">0.9992160</td>
<td align="right">0.6246737</td>
<td align="right">2.526200</td>
<td align="right">3.150089</td>
<td align="right">1.8676365</td>
<td align="right">12.505890</td>
<td align="right">23.33815</td>
<td align="right">0.4001637</td>
<td align="right">0.1888176</td>
</tr>
<tr class="even">
<td align="left">SLC38A3</td>
<td align="right">-0.0049565</td>
<td align="right">0.9950557</td>
<td align="right">0.5877401</td>
<td align="right">2.510331</td>
<td align="right">3.093115</td>
<td align="right">1.7999162</td>
<td align="right">12.309005</td>
<td align="right">22.04564</td>
<td align="right">0.4074689</td>
<td align="right">0.1932555</td>
</tr>
<tr class="odd">
<td align="left">ARHGAP44</td>
<td align="right">0.7308175</td>
<td align="right">2.0767777</td>
<td align="right">-0.0112135</td>
<td align="right">2.362564</td>
<td align="right">3.082168</td>
<td align="right">0.9888491</td>
<td align="right">10.618144</td>
<td align="right">21.80563</td>
<td align="right">0.4447959</td>
<td align="right">0.1976005</td>
</tr>
<tr class="even">
<td align="left">LRRC4</td>
<td align="right">0.0000000</td>
<td align="right">1.0000000</td>
<td align="right">0.6103642</td>
<td align="right">2.344952</td>
<td align="right">2.955317</td>
<td align="right">1.8411018</td>
<td align="right">10.432776</td>
<td align="right">19.20780</td>
<td align="right">0.4477285</td>
<td align="right">0.2105404</td>
</tr>
<tr class="odd">
<td align="left">PLXNB1</td>
<td align="right">-0.0173372</td>
<td align="right">0.9828122</td>
<td align="right">0.6101821</td>
<td align="right">2.352516</td>
<td align="right">2.945361</td>
<td align="right">1.8407665</td>
<td align="right">10.511988</td>
<td align="right">19.01753</td>
<td align="right">0.4468929</td>
<td align="right">0.2148734</td>
</tr>
<tr class="even">
<td align="left">CBX4</td>
<td align="right">-0.0031636</td>
<td align="right">0.9968414</td>
<td align="right">0.5351793</td>
<td align="right">2.237874</td>
<td align="right">2.769890</td>
<td align="right">1.7077544</td>
<td align="right">9.373381</td>
<td align="right">15.95687</td>
<td align="right">0.4613862</td>
<td align="right">0.2798614</td>
</tr>
<tr class="odd">
<td align="left">SIRPB2</td>
<td align="right">-0.0009817</td>
<td align="right">0.9990188</td>
<td align="right">-0.0047030</td>
<td align="right">2.736432</td>
<td align="right">2.730747</td>
<td align="right">0.9953080</td>
<td align="right">15.431826</td>
<td align="right">15.34435</td>
<td align="right">0.3009029</td>
<td align="right">0.2951882</td>
</tr>
</tbody>
</table>
</div>
<div id="gene-set-enrichment-analysis-fdr-0.1-2" class="section level3">
<h3><span class="header-section-number">3.5.5</span> Gene set enrichment analysis FDR 0.1</h3>
<pre class="r"><code>knitr::kable(foe_table(TADA_table_filtered[TADA_table_filtered$FDR_coding &gt;=0.1 &amp; TADA_table_filtered$FDR_all &lt; 0.1, ]$genename, TADA_table_filtered$genename))</code></pre>
<table>
<thead>
<tr class="header">
<th></th>
<th align="right">HI</th>
<th align="right">FMRP</th>
<th align="right">RVIS top 25%</th>
<th align="right">Brainspan top 25%</th>
<th align="right">Brain GO</th>
<th align="right">presynaptic</th>
<th align="right">postsynaptic</th>
<th align="right">ExAC top 25%</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Fold of enrichment</td>
<td align="right">9.6533090</td>
<td align="right">6.6578727</td>
<td align="right">4.007815</td>
<td align="right">3.5050705</td>
<td align="right">2.1072233</td>
<td align="right">16.5098650</td>
<td align="right">3.1211229</td>
<td align="right">2.5158636</td>
</tr>
<tr class="even">
<td>pvalue</td>
<td align="right">0.0011098</td>
<td align="right">0.1428123</td>
<td align="right">0.015525</td>
<td align="right">0.0232116</td>
<td align="right">0.4034691</td>
<td align="right">0.0593588</td>
<td align="right">0.2874136</td>
<td align="right">0.1734212</td>
</tr>
</tbody>
</table>
</div>
<div id="gene-set-enrichment-analysis-fdr-0.2-2" class="section level3">
<h3><span class="header-section-number">3.5.6</span> Gene set enrichment analysis FDR 0.2</h3>
<pre class="r"><code>knitr::kable(foe_table(TADA_table_filtered[TADA_table_filtered$FDR_coding &gt;=0.2 &amp; TADA_table_filtered$FDR_all &lt; 0.2, ]$genename, TADA_table_filtered$genename))</code></pre>
<table>
<thead>
<tr class="header">
<th></th>
<th align="right">HI</th>
<th align="right">FMRP</th>
<th align="right">RVIS top 25%</th>
<th align="right">Brainspan top 25%</th>
<th align="right">Brain GO</th>
<th align="right">presynaptic</th>
<th align="right">postsynaptic</th>
<th align="right">ExAC top 25%</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Fold of enrichment</td>
<td align="right">2.6327206</td>
<td align="right">1.8157835</td>
<td align="right">1.8217338</td>
<td align="right">1.2745711</td>
<td align="right">1.1493945</td>
<td align="right">4.5026905</td>
<td align="right">1.7024307</td>
<td align="right">1.3722892</td>
</tr>
<tr class="even">
<td>pvalue</td>
<td align="right">0.0972986</td>
<td align="right">0.4317384</td>
<td align="right">0.1137694</td>
<td align="right">0.3873558</td>
<td align="right">0.5386502</td>
<td align="right">0.2010312</td>
<td align="right">0.3316037</td>
<td align="right">0.3286614</td>
</tr>
</tbody>
</table>
</div>
<div id="gene-set-enrichment-analysis-fdr-0.3-2" class="section level3">
<h3><span class="header-section-number">3.5.7</span> Gene set enrichment analysis FDR 0.3</h3>
<pre class="r"><code>knitr::kable(foe_table(TADA_table_filtered[TADA_table_filtered$FDR_coding &gt;=0.3 &amp; TADA_table_filtered$FDR_all &lt; 0.3, ]$genename, TADA_table_filtered$genename))</code></pre>
<table>
<thead>
<tr class="header">
<th></th>
<th align="right">HI</th>
<th align="right">FMRP</th>
<th align="right">RVIS top 25%</th>
<th align="right">Brainspan top 25%</th>
<th align="right">Brain GO</th>
<th align="right">presynaptic</th>
<th align="right">postsynaptic</th>
<th align="right">ExAC top 25%</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Fold of enrichment</td>
<td align="right">2.6327206</td>
<td align="right">3.6315669</td>
<td align="right">1.8217338</td>
<td align="right">1.9118567</td>
<td align="right">1.7240918</td>
<td align="right">4.5026905</td>
<td align="right">0.8512153</td>
<td align="right">1.3722892</td>
</tr>
<tr class="even">
<td>pvalue</td>
<td align="right">0.0972986</td>
<td align="right">0.1020691</td>
<td align="right">0.1137694</td>
<td align="right">0.0626662</td>
<td align="right">0.2463658</td>
<td align="right">0.2010312</td>
<td align="right">0.7114238</td>
<td align="right">0.3286614</td>
</tr>
</tbody>
</table>
</div>
<div id="gene-set-enrichment-analysis-fdr-0.1-all-genes-background" class="section level3">
<h3><span class="header-section-number">3.5.8</span> Gene set enrichment analysis FDR 0.1, all genes background</h3>
<pre class="r"><code>knitr::kable(foe_table(TADA_table_filtered[TADA_table_filtered$FDR_coding &gt;=0.1 &amp; TADA_table_filtered$FDR_all &lt; 0.1, ]$genename, combined_BF_table$genename))</code></pre>
<table>
<thead>
<tr class="header">
<th></th>
<th align="right">HI</th>
<th align="right">FMRP</th>
<th align="right">RVIS top 25%</th>
<th align="right">Brainspan top 25%</th>
<th align="right">Brain GO</th>
<th align="right">presynaptic</th>
<th align="right">postsynaptic</th>
<th align="right">ExAC top 25%</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Fold of enrichment</td>
<td align="right">10.8140209</td>
<td align="right">7.6433251</td>
<td align="right">4.486779</td>
<td align="right">4.059374</td>
<td align="right">2.0187108</td>
<td align="right">18.6277445</td>
<td align="right">3.5901135</td>
<td align="right">2.8592218</td>
</tr>
<tr class="even">
<td>pvalue</td>
<td align="right">0.0007895</td>
<td align="right">0.1252167</td>
<td align="right">0.011065</td>
<td align="right">0.014942</td>
<td align="right">0.4180905</td>
<td align="right">0.0527312</td>
<td align="right">0.2534934</td>
<td align="right">0.1377285</td>
</tr>
</tbody>
</table>
</div>
<div id="gene-set-enrichment-analysis-fdr-0.2-all-genes-background" class="section level3">
<h3><span class="header-section-number">3.5.9</span> Gene set enrichment analysis FDR 0.2, all genes background</h3>
<pre class="r"><code>knitr::kable(foe_table(TADA_table_filtered[TADA_table_filtered$FDR_coding &gt;=0.2 &amp; TADA_table_filtered$FDR_all &lt; 0.2, ]$genename, combined_BF_table$genename))</code></pre>
<table>
<thead>
<tr class="header">
<th></th>
<th align="right">HI</th>
<th align="right">FMRP</th>
<th align="right">RVIS top 25%</th>
<th align="right">Brainspan top 25%</th>
<th align="right">Brain GO</th>
<th align="right">presynaptic</th>
<th align="right">postsynaptic</th>
<th align="right">ExAC top 25%</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Fold of enrichment</td>
<td align="right">2.9492784</td>
<td align="right">2.084543</td>
<td align="right">2.0394449</td>
<td align="right">1.4761359</td>
<td align="right">1.1011150</td>
<td align="right">5.0802940</td>
<td align="right">1.9582437</td>
<td align="right">1.5595755</td>
</tr>
<tr class="even">
<td>pvalue</td>
<td align="right">0.0741495</td>
<td align="right">0.387760</td>
<td align="right">0.0757617</td>
<td align="right">0.2766396</td>
<td align="right">0.5638718</td>
<td align="right">0.1801839</td>
<td align="right">0.2721838</td>
<td align="right">0.2414526</td>
</tr>
</tbody>
</table>
</div>
<div id="gene-set-enrichment-analysis-fdr-0.3-all-genes-background" class="section level3">
<h3><span class="header-section-number">3.5.10</span> Gene set enrichment analysis FDR 0.3, all genes background</h3>
<pre class="r"><code>knitr::kable(foe_table(TADA_table_filtered[TADA_table_filtered$FDR_coding &gt;=0.3 &amp; TADA_table_filtered$FDR_all &lt; 0.3, ]$genename, combined_BF_table$genename))</code></pre>
<table>
<thead>
<tr class="header">
<th></th>
<th align="right">HI</th>
<th align="right">FMRP</th>
<th align="right">RVIS top 25%</th>
<th align="right">Brainspan top 25%</th>
<th align="right">Brain GO</th>
<th align="right">presynaptic</th>
<th align="right">postsynaptic</th>
<th align="right">ExAC top 25%</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Fold of enrichment</td>
<td align="right">2.9492784</td>
<td align="right">4.1690864</td>
<td align="right">2.0394449</td>
<td align="right">2.2142038</td>
<td align="right">1.6516725</td>
<td align="right">5.0802940</td>
<td align="right">0.9791219</td>
<td align="right">1.5595755</td>
</tr>
<tr class="even">
<td>pvalue</td>
<td align="right">0.0741495</td>
<td align="right">0.0804903</td>
<td align="right">0.0757617</td>
<td align="right">0.0319938</td>
<td align="right">0.2682645</td>
<td align="right">0.1801839</td>
<td align="right">0.6577434</td>
<td align="right">0.2414526</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
