

```{r}
library(data.table)
#library(parallel)
source("../lib/170122_base_level_simulation_functions.R")
```

***
**NOTE**

1. First adjust for mutation rate for each 50bp window, then apply the the adjusting factor for each base so we could get the base-level adjusted mutation rate. When adjusting for mutation rates, only consider features including promoter, coding, within-10kb non-coding, and 50bp cg contents.
2. Only consider the relative risk from active promoters and enhancers (union), marked by Noonan and Roadmap brain H3k27ac. 
3. Only use bases from top 1000 TADA genes for now.
4. The model itself could be easily adapted to cases where more epigenomic features or functional annotations are included. 
5. The output will be R object which store base-level information in a compact mode.
6. 100way phylop and gerp annotation is added, brain h3k27ac, fetal DHS and adult DHS.
7. Use corrected gene assignment (See Note.html 170307).
8. have spidex_intro windows when adjusting mutation rate for other non-splicing regulatory mutations.
9. Add splicing mutations in the model for relative risk estimation.


***


## estimate relative risk from top 20 TADA genes, afeter adjusting for mutation rates

```{r, cache=TRUE, cache.extra = r, cache=TRUE, cache.extra = adjust_mutation_rate_window_to_base_compact_more_feature_splicing_hybrid}
new_control = new.env()
load("../data/debug_region_list_073116_data_matrixtransformed_for_old_code.Rdata", new_control) # this Rdata has spedix info


prefix <-system("date +%s", intern = TRUE) # prefix for temporary files that will be deleted at the end of the pipeline

write.table(new_control$data_matrix[new_control$data_matrix$Prediction == "dnv_proband" & new_control$data_matrix$Type == "SNV",c(1,2,3,7)], paste(prefix,"temp.bed",sep = ""), col.names = FALSE, row.names = FALSE, sep = "\t", quote = FALSE)
write.table(new_control$data_matrix[new_control$data_matrix$Prediction == "dnv_proband" ,c(1:5,8,7)], paste(prefix,"annovar_input_temp.bed",sep = ""), col.names = FALSE, row.names = FALSE, sep = "\t", quote = FALSE)

# SNV_ID for cases or controls. 
SNV_ID = new_control$data_matrix[new_control$data_matrix$Prediction == "dnv_proband" & new_control$data_matrix$Type == "SNV",c(1,2,3,7)]$ID
proband_SNV_mut_spidex <- new_control$mut_spedix[is.element(new_control$mut_spedix$index, SNV_ID),] # this will be paseed as an argument to the function below

# first adjuste for mutation rates at 50-bp window level and then apply the scaling factor to each base of noncoding regions of selected top percentage of TADA genes.

preprocessing_time <- system.time(adjusted_base_level_rr <- adjust_mutation_rate_window_to_base_compact_more_feature_splicing_hybrid(mut_file = paste(prefix,"temp.bed",sep = ""),
window_file = "../other_annotation/epigenomic_annotation/Whole_genome.promoter_yanyu_10kb_exons_no_utr_no_na_genes.50bp_window.bed",
cg_file = "../other_annotation/epigenomic_annotation/Whole_genome.promoter_yanyu_10kb_exons_no_utr_no_na_genes.50bp_window.bed.fasta.cg",
mutrate_file = "../other_annotation/epigenomic_annotation/Whole_genome.promoter_yanyu_10kb_exons_no_utr_no_na_genes.50bp_window.bed.mutrate",
sample_size = 314,
epigenomic_marks = c("../other_annotation/epigenomic_annotation/Noonan_brain_roadmap_union.bed","../other_annotation/epigenomic_annotation/Epigenome_E081_E082_intersection.bed","../other_annotation/epigenomic_annotation/encode_DHS_union.bed"),
sequence_annotation = c("phylop_100way", "gerp"), 
sequence_annotation_cutoff = list(phylop_100way=2, gerp =2 ), 
sequence_annotation_ref = list(phylop_100way = "../other_annotation/conservation_annotation/hg19.100way.phyloP100way.bw", gerp = "../other_annotation/conservation_annotation/hg19_gerp_score.bw"),
overlap = 0.5,
rm_nonsyn = FALSE,
annovar_folder = "/media/yuwen/Elements/ANNOVAR/annovar/",
annovar_input = "no.txt",
gene_assign_file = "../other_annotation/epigenomic_annotation/Whole_genome.promoter_yanyu_10kb_exons_no_utr_no_na_genes.50bp_window.bed_genename_corrected.bed",
gene_prior_file = "../other_annotation/gene_list/TADA_SNV_CNV_combined_Feb7_tadaname_null_PPA_p0_0.94.txt",
report_proportion = 20/18665,
mutrate_ref_file = "/media/yuwen/Elements/mutation_rate_raw/MarkDaly/Daly_mutrate.bw",
node_n = 6,
feature_start = 5,
feature_end = 9,
feature_number = 5, 
chunk_partition_num =1,
spidex_mutrate_prefix = "../other_annotation/spidex_database/mutrate_spedix_score_by_chr/lower_10pct_-1.416/spidex_public_noncommercial_v1_0_collapse_unique_base.extended_3bp.bed.fasta.trinuc.mutrate_with_spedix_gene.lower_10pct",
splicing_cutoff = -1.416,
proband_SNV_mut_spidex = proband_SNV_mut_spidex))

rm(new_control)
system(paste("rm ",prefix,"annovar_input_temp.bed",sep = ""))
system(paste("rm ",prefix, "temp.bed",sep = ""))

```


### Estimate RR in a joint model for both splicing mutations and non-splicing non-coding mutations

```{r,cache = TRUE, dependson= adjusted_base_level_rr, cache.extra=estimate_effect_size_for_simulation_data_mixture_with_categorization_for_compact_data_splicing_hybrid}
# notice now feature start is 5, and feature end is 6, two features 
simulation_estimate <- estimate_effect_size_for_simulation_data_mixture_with_categorization_for_compact_data_splicing_hybrid(data = adjusted_base_level_rr$base_info, 
                                                                                                                             splicing_data = adjusted_base_level_rr$splicing_info,
                                                                                                                             non_splicing_features = c(1,2,3,4,5), 
                                                                                                                             gene_prior_file  = "../other_annotation/gene_list/TADA_SNV_CNV_combined_Feb7_tadaname_null_PPA_p0_0.94.txt",
sample_size = 314)

fisher_info<-solve(-simulation_estimate$mle$hessian)
prop_sigma<-sqrt(diag(fisher_info))
upper<-simulation_estimate$mle$par+1.96*prop_sigma
lower<-simulation_estimate$mle$par-1.96*prop_sigma
a = data.frame(relative_risk = simulation_estimate$mle$par, lower_bound = lower, upper_bournd = upper)
rownames(a) = c("Noonan+Roadmap Brain H3K27ac enhancer + promoter","Epigenome_DHS", "ENCODE_DHS", "Phylop>2", "Gerp>2","Splicing")
knitr::kable(a)
adjusted_base_level_rr$partition_time1
adjusted_base_level_rr$partition_time2

```


### Estimate RR in two separate likelihood models for splicing mutations and non-splicing non-coding mutations

```{r,cache = TRUE, dependson= adjusted_base_level_rr, cache.extra=estimate_effect_size_for_simulation_data_mixture_with_categorization_for_compact_data_splicing_hybrid_v2}
# notice now feature start is 5, and feature end is 6, two features 
simulation_estimate <- estimate_effect_size_for_simulation_data_mixture_with_categorization_for_compact_data_splicing_hybrid_v2(data = adjusted_base_level_rr$base_info, 
                                                                                                                             splicing_data = adjusted_base_level_rr$splicing_info,
                                                                                                                             non_splicing_features = c(1,2,3,4,5), 
                                                                                                                             gene_prior_file  = "../other_annotation/gene_list/TADA_SNV_CNV_combined_Feb7_tadaname_null_PPA_p0_0.94.txt",
sample_size = 314)

fisher_info_1<-solve(-simulation_estimate$mle_1$hessian)
fisher_info_2<-solve(-simulation_estimate$mle_2$hessian)

prop_sigma<-sqrt(c(diag(fisher_info_1),diag(fisher_info_2)))
rr_estimate <- c(simulation_estimate$mle_1$par, simulation_estimate$mle_2$par)

upper<-rr_estimate+1.96*prop_sigma
lower<-rr_estimate-1.96*prop_sigma


a = data.frame(relative_risk = rr_estimate, lower_bound = lower, upper_bournd = upper)
rownames(a) = c("Noonan+Roadmap Brain H3K27ac enhancer + promoter","Epigenome_DHS", "ENCODE_DHS", "Phylop>2", "Gerp>2","Splicing")
knitr::kable(a)

```


### Choose the ones that are significant and refit the model

```{r,cache = TRUE, dependson= adjusted_base_level_rr, cache.extra=estimate_effect_size_for_simulation_data_mixture_with_categorization_for_compact_data_splicing_hybrid_v2}
# notice now feature start is 5, and feature end is 6, two features 
simulation_estimate <- estimate_effect_size_for_simulation_data_mixture_with_categorization_for_compact_data_splicing_hybrid_v2(data = adjusted_base_level_rr$base_info, 
                                                                                                                             splicing_data = adjusted_base_level_rr$splicing_info,
                                                                                                                             non_splicing_features = c(1), 
                                                                                                                             gene_prior_file  = "../other_annotation/gene_list/TADA_SNV_CNV_combined_Feb7_tadaname_null_PPA_p0_0.94.txt",
sample_size = 314)

fisher_info_1<-solve(-simulation_estimate$mle_1$hessian)
fisher_info_2<-solve(-simulation_estimate$mle_2$hessian)

prop_sigma<-sqrt(c(diag(fisher_info_1),diag(fisher_info_2)))
rr_estimate <- c(simulation_estimate$mle_1$par, simulation_estimate$mle_2$par)

upper<-rr_estimate+1.96*prop_sigma
lower<-rr_estimate-1.96*prop_sigma


a = data.frame(relative_risk = rr_estimate, lower_bound = lower, upper_bournd = upper)
rownames(a) = c("Noonan+Roadmap Brain H3K27ac enhancer + promoter","Splicing")
knitr::kable(a)

```
