<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>Burden</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/united.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-1.1/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-1.1/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">ASD Project</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="meeting.html">Meeting</a>
</li>
<li>
  <a href="Note.html">Note</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Burden</h1>

</div>


<p>Motifs haven’t been updated in this yet ref = “total_ratio” changed to “self” when using show_burden_for_selected_features Use a newly calculated set of motif change scores to define motif SNVs, but only focusing on motifs that more relevant Here motifs are defined as relevant if they are 1) in top 50% brain expressed genes, and 2) are enriched in at least one gain-enriched brain module Partition enhancers based on distance to TSS based on Yanyu’s pipeline. Basically, her pipeline takes intersections between enhancers and distance ranges, and remove bases of enhancers that are not overlap.</p>
<p>non ASD genes are the bottom 20% not plotting mutation is 5’UTR or 3’UTR added one row for showing the union of enhancers and promoters Only have Fantom, Noonan and Whole genome Look at more genesets.</p>
<p>Look at differentially regulated H3K27ac regions from Sun et al., Cell 2016</p>
<pre class="r"><code>getwd()</code></pre>
<pre><code>## [1] &quot;/media/yuwen/F/ASD/analysis&quot;</code></pre>
<pre class="r"><code>knitr::opts_chunk$set(warning=FALSE, message=FALSE)</code></pre>
<div id="read-in-coding-tada-data" class="section level3">
<h3>read in coding tada data</h3>
<pre><code>## [1] &quot;/media/yuwen/F/ASD/analysis&quot;</code></pre>
</div>
<div id="whole-genome" class="section level2">
<h2>Whole genome</h2>
</div>
<div id="asd_k27ac_cb_down_sun_et_al_cell_2016.bed" class="section level2">
<h2>ASD_K27ac_CB_down_Sun_et_al_Cell_2016.bed</h2>
</div>
<div id="asd_k27ac_cb_up_sun_et_al_cell_2016.bed" class="section level2">
<h2>ASD_K27ac_CB_up_Sun_et_al_Cell_2016.bed</h2>
</div>
<div id="asd_k27ac_pfc_down_sun_et_al_cell_2016.bed" class="section level2">
<h2>ASD_K27ac_PFC_down_Sun_et_al_Cell_2016.bed</h2>
</div>
<div id="asd_k27ac_pfc_up_sun_et_al_cell_2016.bed" class="section level2">
<h2>ASD_K27ac_PFC_up_Sun_et_al_Cell_2016.bed</h2>
</div>
<div id="asd_k27ac_tc_down_sun_et_al_cell_2016.bed" class="section level2">
<h2>ASD_K27ac_TC_down_Sun_et_al_Cell_2016.bed</h2>
</div>
<div id="asd_k27ac_tc_up_sun_et_al_cell_2016.bed" class="section level2">
<h2>ASD_K27ac_TC_up_Sun_et_al_Cell_2016.bed</h2>
<pre class="r"><code>plot_summary = plot_summary[!is.na(plot_summary$gene_list),]
plot_summary = plot_summary[!duplicated(plot_summary),]
plot_summary = plot_summary[plot_summary$gene_list != &quot;NP_genes&quot;,]
plot_summary = plot_summary[plot_summary$region_tag == &quot;Promoter&quot; | plot_summary$region_tag == &quot;Promoter+10kb_enhancer&quot; | plot_summary$region_tag == &quot;1-10Kb&quot;,]
plot_summary$gene_list = factor(plot_summary$gene_list, levels = c(&quot;stringent_ASD&quot;, &quot;TADA_top6_pct&quot;, &quot;SCZ_composite&quot;,&quot;total&quot;))</code></pre>
<pre class="r"><code>#scaling factor for plotting, which is the burden of baseline. Used for 160229 data only
scaling = 0.73</code></pre>
<p><img src="170306_ploting_burden_for_without_utr_more_genesets_160229_data_Noonan_brain_motif_top50_expressed_enhancer_partition_based_on_yanyu_pipeline_files/figure-html/unnamed-chunk-41-1.png" width="1248" /><img src="170306_ploting_burden_for_without_utr_more_genesets_160229_data_Noonan_brain_motif_top50_expressed_enhancer_partition_based_on_yanyu_pipeline_files/figure-html/unnamed-chunk-41-2.png" width="1248" /><img src="170306_ploting_burden_for_without_utr_more_genesets_160229_data_Noonan_brain_motif_top50_expressed_enhancer_partition_based_on_yanyu_pipeline_files/figure-html/unnamed-chunk-41-3.png" width="1248" /><img src="170306_ploting_burden_for_without_utr_more_genesets_160229_data_Noonan_brain_motif_top50_expressed_enhancer_partition_based_on_yanyu_pipeline_files/figure-html/unnamed-chunk-41-4.png" width="1248" /><img src="170306_ploting_burden_for_without_utr_more_genesets_160229_data_Noonan_brain_motif_top50_expressed_enhancer_partition_based_on_yanyu_pipeline_files/figure-html/unnamed-chunk-41-5.png" width="1248" /><img src="170306_ploting_burden_for_without_utr_more_genesets_160229_data_Noonan_brain_motif_top50_expressed_enhancer_partition_based_on_yanyu_pipeline_files/figure-html/unnamed-chunk-41-6.png" width="1248" /><img src="170306_ploting_burden_for_without_utr_more_genesets_160229_data_Noonan_brain_motif_top50_expressed_enhancer_partition_based_on_yanyu_pipeline_files/figure-html/unnamed-chunk-41-7.png" width="1248" /><img src="170306_ploting_burden_for_without_utr_more_genesets_160229_data_Noonan_brain_motif_top50_expressed_enhancer_partition_based_on_yanyu_pipeline_files/figure-html/unnamed-chunk-41-8.png" width="1248" /><img src="170306_ploting_burden_for_without_utr_more_genesets_160229_data_Noonan_brain_motif_top50_expressed_enhancer_partition_based_on_yanyu_pipeline_files/figure-html/unnamed-chunk-41-9.png" width="1248" /><img src="170306_ploting_burden_for_without_utr_more_genesets_160229_data_Noonan_brain_motif_top50_expressed_enhancer_partition_based_on_yanyu_pipeline_files/figure-html/unnamed-chunk-41-10.png" width="1248" /><img src="170306_ploting_burden_for_without_utr_more_genesets_160229_data_Noonan_brain_motif_top50_expressed_enhancer_partition_based_on_yanyu_pipeline_files/figure-html/unnamed-chunk-41-11.png" width="1248" /><img src="170306_ploting_burden_for_without_utr_more_genesets_160229_data_Noonan_brain_motif_top50_expressed_enhancer_partition_based_on_yanyu_pipeline_files/figure-html/unnamed-chunk-41-12.png" width="1248" /></p>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
