---
title: "Partition analysis"
output: html_document
---

remove splicing mutations that are nonsynonymous/LoF or that are in H3K27ac regions
Only consider splicing mutations that are in the 50-bp windows that are used to adjust for mutaton rates
**Here the splicing relative risk is estimated in a model-based framework using top1000 TADA genes**(170408_estimate_relative_risk_top1000_genes_from_categorization_from_each_separate_study.html)

Likelihood is calculated from each of the 5 ASD studies and then combined

```{r,echo=FALSE}
library(data.table)
library(ggplot2)
### function to generate mutation tables
# Estimate percent of variance explained by a certain class of mutations
# a little bit from Xin's code in that I directly use RR rather using burden
estimate.PVE <- function(RR, exposure, pi=0.05, t=2.18) {
  
  # exposure of the causal mutations 
  p <- exposure * pi
  
  # PVE
  beta <-t- qnorm(1- RR* (1-pnorm(t)))
  PVE <- beta^2 *p *(1-p)
  return (PVE*100)
}

draw_partition_pie <-function(PVEs, class){
  #PVEs is the estimated variance explained by a certain class of mutations
  #class is a vector of mutation class names
  percents <- round(PVEs/sum(PVEs) * 100, 0)
  labels <- class
  labels <- paste(labels, percents)
  labels <- paste(labels, "%",sep="")
  pie(PVEs,labels = labels, col=rainbow(length(labels)))
}
```

```{r}
### load in coding mutation rate and genes, fixed across all studies
pi = 0.06 # the proportion of risk genes
coding_tada = read.delim("../other_annotation/gene_list/TADA_SNV_CNV_combined_Feb7.txt", header = TRUE, sep = "\t", stringsAsFactors = FALSE)
coding_tada_denovo = data.frame(coding_tada$RefSeqName, coding_tada$TadaName, coding_tada$mut.rate, coding_tada$dn.LoF, coding_tada$dn.mis3, coding_tada$BF.SNV.dn, coding_tada$BF.combined, coding_tada$qvalue.combined)
colnames(coding_tada_denovo) = c("RefSeqName", "TadaName", "mut.rate", "dn.LoF", "dn.mis3", "BF.SNV.dn", "BF.combined", "qvalue.combined")
lof_rate = coding_tada_denovo[,3] * 0.074
mis_rate = coding_tada_denovo[,3] * 0.32
#relative risk for coding LoF and Mis mutations.
coding_gamma = c(20,4.7)
sum(lof_rate) 
sum(mis_rate)
```


### 10kb with all enhancer/promoter/
read in K27ac mutation rate files, generated by **Pipeline 22**
read in splicing mutation rate files, to start simple, I didn't subtract the mutation rate coming from Noonan brain k27ac and that may overlap with nonsynonymous mutations. The overlap lap should be very small, as coding regions are small, neitherless to say functional coding regions. And for K27ac regions, I only considered promoters and regions within 10kb of TSSs.  
```{r, cache=TRUE}

#10kb with all enhancer SNVs and indels 

pi = 0.06
gamma.enhancer.SNV =c(1.87)
gamma.splicing = c(3.22)

all_genes_10kb_de_mutation_rate_SNV = read.delim("../other_annotation/epigenomic_annotation/Noonan_brain_roadmap_union.yanyu_pipeline_enhancers.10000.bp_within_TSS.bed.mutrate", header = FALSE, sep = "\t", stringsAsFactors = FALSE)

all_genes_promoter_de_mutation_rate_SNV = read.delim("../other_annotation/epigenomic_annotation/Noonan_brain_roadmap_union.promoter_1kb_genename.bed.mutrate", header = FALSE, sep = "\t", stringsAsFactors = FALSE)

enhancer_by_gene <- read.delim("../other_annotation/epigenomic_annotation/Noonan_brain_roadmap_union.yanyu_pipeline_enhancers.10000.bp_within_TSS.bed", header = FALSE, sep = "\t", stringsAsFactors = FALSE)
enhancer_by_gene <- data.frame(enhancer_by_gene, index = seq(1,nrow(enhancer_by_gene)))

enhancer_by_gene <- merge(enhancer_by_gene, all_genes_10kb_de_mutation_rate_SNV, by.x = "index", by.y = "V1")


promoter_by_gene <- read.delim("../other_annotation/epigenomic_annotation/Noonan_brain_roadmap_union.promoter_1kb_genename.bed", header = FALSE, sep = "\t", stringsAsFactors = FALSE)
promoter_by_gene <- data.frame(promoter_by_gene, index = seq(1,nrow(promoter_by_gene)))

promoter_by_gene <- merge(promoter_by_gene, all_genes_promoter_de_mutation_rate_SNV, by.x = "index", by.y = "V1")

sum(all_genes_10kb_de_mutation_rate_SNV[,4]) + sum(all_genes_promoter_de_mutation_rate_SNV[,4])

# now read in mutation rates of lower 10% spidex score
splicing_mut_sum <- 0 
chr <- c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,"X","Y")
for(chr_i in chr){
  splicing_mut_file_name <- paste("../other_annotation/spidex_database/mutrate_spedix_score_by_chr/lower_10pct_-1.416/spidex_public_noncommercial_v1_0_collapse_unique_base.extended_3bp.bed.fasta.trinuc.mutrate_with_spedix_gene.lower_10pct.chr", chr_i, sep = "")
  splicing_mut_for_chr <- fread(splicing_mut_file_name, header = FALSE, sep = "\t", stringsAsFactors = FALSE)
  splicing_mut_sum <- splicing_mut_sum + sum(splicing_mut_for_chr[,4])
}
splicing_mut_sum
```


### gamma based on tada_q0.5, without indels

```{r,echo=FALSE,fig.width=5,fig.height=5}
mutation_class = c("Loss-of-function","Missense","Regulatory SNV","Splicing")
#postscript("piechart_pve_snvs_and_indels.eps")
#draw_partition_pie(estimate.PVE(gamma, 2*c(sum(mut_rate_table$lof_rate, na.rm = TRUE), sum(mut_rate_table$mis_rate, na.rm = TRUE),sum(mut_rate_table$enhancer_SNV_rate, na.rm = TRUE),sum(mut_rate_table$enhancer_indel_rate, na.rm = TRUE)),pi=0.06),mutation_class)
#dev.off()
gamma = c(coding_gamma, gamma.enhancer.SNV[1], gamma.splicing[1])

draw_partition_pie(estimate.PVE(gamma, 2*c(sum(lof_rate), sum(mis_rate),(sum(all_genes_10kb_de_mutation_rate_SNV[,4]) + sum(all_genes_promoter_de_mutation_rate_SNV[,4])), splicing_mut_sum),pi=0.06),mutation_class)

postscript("tmp/170418_risk_partition_combining_enhancer_promoter_with_splicing_and_without_indel.eps",width = 4, height =4)
draw_partition_pie(estimate.PVE(gamma, 2*c(sum(lof_rate), sum(mis_rate),(sum(all_genes_10kb_de_mutation_rate_SNV[,4]) + sum(all_genes_promoter_de_mutation_rate_SNV[,4])), splicing_mut_sum),pi=0.06),mutation_class)
dev.off()

pdf("tmp/170418_risk_partition_combining_enhancer_promoter_with_splicing_and_without_indel.pdf",width = 5, height =4)
draw_partition_pie(estimate.PVE(gamma, 2*c(sum(lof_rate), sum(mis_rate),(sum(all_genes_10kb_de_mutation_rate_SNV[,4]) + sum(all_genes_promoter_de_mutation_rate_SNV[,4])), splicing_mut_sum),pi=0.06),mutation_class)
dev.off()

gamma
#output PVE
estimate.PVE(gamma, 2*c(sum(lof_rate), sum(mis_rate),(sum(all_genes_10kb_de_mutation_rate_SNV[,4]) + sum(all_genes_promoter_de_mutation_rate_SNV[,4])), splicing_mut_sum),pi=0.06)

```

