---
output:
  html_document:
    toc: true
    number_sections: true
    theme: united
    highlight: textmate
---

```{r}
library(data.table)
library(parallel)
source("../lib/170122_base_level_simulation_functions.R")

```

***
**NOTE**
To address the concern that each study would have different effects from a same genomic feature (e.g., sequencing depth/GC content would have a stronger effect for a study that has shallower sequencing depth.), I will adjust mutation rate in different studies. 

Use relative risks estimated from top1000 genes to calculate BF for each gene from each study.RR esitated from [here](170408_estimate_relative_risk_top1000_genes_from_categorization_from_each_separate_study.html)


***


# Read in data from 5 studies
```{r}
new_control = new.env()
load("../data/debug_region_list_073116_data_matrixtransformed_for_old_code.Rdata", new_control) # this Rdata has spedix info
# add in the 0-based start position

new_control$mutation$start_0 <- as.numeric(as.character(new_control$mutation$start)) - 1

# mutation data in the old format, I loaded this just in order to retrive information about which study each mutation comes from
old <- new.env()
load("../data/debug_region_list_073116_data_matrix.Rdata", old)

# Use a merge function to associate mutations with study information
# I merged by matching the chromosome locations, including chr, start and end. For the case SNV data, I removed all redundant mutations, so merging in this way would ensure a 1-to-1 matching between mutation info and study info
# There is redundant SNVs and other mutations in old$mutation$mutation, and there are redundant non-SNV mutations in new_control$mutation, but I didn't use non-SNV mutations in this study. So having such type of redundant mutations should be fine.

mut_with_study_label <- merge(new_control$mutation[new_control$mutation$phenotype =="ASD" & new_control$mutation$mut_type == "SNV",], old$mutation$mutation[old$mutation$mutation$phenotype == "ASD" & old$mutation$mutation$mut_type == "SNV",], by.x = c("chr","start_0","end"), by.y = c("chr","start","end"))
# The index is consistent with new_control$mutation, but not with old$mutation$mutation
mut_with_study_label <- data.frame(index = mut_with_study_label$index.x, study = mut_with_study_label$study.y)

ASD_study_name <- as.vector(unique(mut_with_study_label$study)[1:5])

```


## Ryan K C Yuen et al. Nature Medicine 2015

```{r,cache=TRUE, cache.extra = r, cache=TRUE, cache.extra = adjust_mutation_rate_window}

prefix <-system("date +%s", intern = TRUE) # prefix for temporary files that will be deleted at the end of the pipeline

study_ID <- mut_with_study_label[mut_with_study_label$study == "Ryan K C Yuen et al. Nature Medicine 2015",]$index

# add study_ID , phenotyp_ID and mut_type ID as constraint

write.table(new_control$data_matrix[new_control$data_matrix$Prediction == "dnv_proband" & new_control$data_matrix$Type == "SNV" & is.element(new_control$data_matrix$ID, study_ID),c(1,2,3,7)], paste(prefix,"temp.bed",sep = ""), col.names = FALSE, row.names = FALSE, sep = "\t", quote = FALSE)
write.table(new_control$data_matrix[new_control$data_matrix$Prediction == "dnv_proband" & is.element(new_control$data_matrix$ID, study_ID),c(1:5,8,7)], paste(prefix,"annovar_input_temp.bed",sep = ""), col.names = FALSE, row.names = FALSE, sep = "\t", quote = FALSE)

# SNV_ID for cases or controls. 
SNV_ID = new_control$data_matrix[new_control$data_matrix$Prediction == "dnv_proband" & new_control$data_matrix$Type == "SNV" & is.element(new_control$data_matrix$ID, study_ID),c(1,2,3,7)]$ID
proband_SNV_mut_spidex <- new_control$mut_spedix[is.element(new_control$mut_spedix$index, SNV_ID) & is.element(new_control$mut_spedix$index, study_ID),] # this will be paseed as an argument to the function below

# first adjuste for mutation rates at 50-bp window level and then apply the scaling factor to each base of noncoding regions of selected top percentage of TADA genes.

adjusted_base_level_rr_1 <- adjust_mutation_rate_window_to_base_compact_more_feature_splicing_hybrid(mut_file = paste(prefix,"temp.bed",sep = ""),
window_file = "../other_annotation/epigenomic_annotation/Whole_genome.promoter_yanyu_10kb_exons_introns_with_spidex_no_na_genes.50bp_window.bed",
cg_file = "../other_annotation/epigenomic_annotation/Whole_genome.promoter_yanyu_10kb_exons_introns_with_spidex_no_na_genes.50bp_window.bed.fasta.cg",
mutrate_file = "../other_annotation/epigenomic_annotation/Whole_genome.promoter_yanyu_10kb_exons_introns_with_spidex_no_na_genes.50bp_window.bed.mutrate",
sample_size = 162,
epigenomic_marks = c("../other_annotation/epigenomic_annotation/Noonan_brain_roadmap_union.bed","../other_annotation/epigenomic_annotation/Epigenome_E081_E082_intersection.bed","../other_annotation/epigenomic_annotation/encode_DHS_union.bed"),
sequence_annotation = c("phylop_100way", "gerp"), 
sequence_annotation_cutoff = list(phylop_100way=2, gerp =2 ), 
sequence_annotation_ref = list(phylop_100way = "../other_annotation/conservation_annotation/hg19.100way.phyloP100way.bw", gerp = "../other_annotation/conservation_annotation/hg19_gerp_score.bw"),
overlap = 0.5,
rm_nonsyn = FALSE,
annovar_folder = "/media/yuwen/Elements/ANNOVAR/annovar/",
annovar_input = "no.txt",
gene_assign_file = "../other_annotation/epigenomic_annotation/Whole_genome.promoter_yanyu_10kb_exons_no_utr_no_na_genes.50bp_window.bed_genename_corrected.bed",
gene_prior_file = "../other_annotation/gene_list/TADA_SNV_CNV_combined_Feb7_tadaname_null_PPA_p0_0.94.txt",
report_proportion = 1,
mutrate_ref_file = "/media/yuwen/Elements/mutation_rate_raw/MarkDaly/Daly_mutrate.bw",
node_n = 6,
feature_start = 5,
feature_end = 9,
feature_number = 5, 
chunk_partition_num =20,
spidex_mutrate_prefix = "../other_annotation/spidex_database/mutrate_spedix_score_by_chr/lower_10pct_-1.416/spidex_public_noncommercial_v1_0_collapse_unique_base.extended_3bp.bed.fasta.trinuc.mutrate_with_spedix_gene.lower_10pct",
splicing_cutoff = -1.416,
proband_SNV_mut_spidex = proband_SNV_mut_spidex)

system(paste("rm ",prefix,"annovar_input_temp.bed",sep = ""))
system(paste("rm ",prefix, "temp.bed",sep = ""))

```

## Augustine Kong et al. Nature 2012

```{r,cache=TRUE, cache.extra = r, cache=TRUE, cache.extra = adjust_mutation_rate_window}

prefix <-system("date +%s", intern = TRUE) # prefix for temporary files that will be deleted at the end of the pipeline

study_ID <- mut_with_study_label[mut_with_study_label$study == "Augustine Kong et al. Nature 2012",]$index

# add study_ID , phenotyp_ID and mut_type ID as constraint

write.table(new_control$data_matrix[new_control$data_matrix$Prediction == "dnv_proband" & new_control$data_matrix$Type == "SNV" & is.element(new_control$data_matrix$ID, study_ID),c(1,2,3,7)], paste(prefix,"temp.bed",sep = ""), col.names = FALSE, row.names = FALSE, sep = "\t", quote = FALSE)
write.table(new_control$data_matrix[new_control$data_matrix$Prediction == "dnv_proband" & is.element(new_control$data_matrix$ID, study_ID),c(1:5,8,7)], paste(prefix,"annovar_input_temp.bed",sep = ""), col.names = FALSE, row.names = FALSE, sep = "\t", quote = FALSE)

# SNV_ID for cases or controls. 
SNV_ID = new_control$data_matrix[new_control$data_matrix$Prediction == "dnv_proband" & new_control$data_matrix$Type == "SNV" & is.element(new_control$data_matrix$ID, study_ID),c(1,2,3,7)]$ID
proband_SNV_mut_spidex <- new_control$mut_spedix[is.element(new_control$mut_spedix$index, SNV_ID) & is.element(new_control$mut_spedix$index, study_ID),] # this will be paseed as an argument to the function below

# first adjuste for mutation rates at 50-bp window level and then apply the scaling factor to each base of noncoding regions of selected top percentage of TADA genes.

adjusted_base_level_rr_2 <- adjust_mutation_rate_window_to_base_compact_more_feature_splicing_hybrid(mut_file = paste(prefix,"temp.bed",sep = ""),
window_file = "../other_annotation/epigenomic_annotation/Whole_genome.promoter_yanyu_10kb_exons_introns_with_spidex_no_na_genes.50bp_window.bed",
cg_file = "../other_annotation/epigenomic_annotation/Whole_genome.promoter_yanyu_10kb_exons_introns_with_spidex_no_na_genes.50bp_window.bed.fasta.cg",
mutrate_file = "../other_annotation/epigenomic_annotation/Whole_genome.promoter_yanyu_10kb_exons_introns_with_spidex_no_na_genes.50bp_window.bed.mutrate",
sample_size = 78,
epigenomic_marks = c("../other_annotation/epigenomic_annotation/Noonan_brain_roadmap_union.bed","../other_annotation/epigenomic_annotation/Epigenome_E081_E082_intersection.bed","../other_annotation/epigenomic_annotation/encode_DHS_union.bed"),
sequence_annotation = c("phylop_100way", "gerp"), 
sequence_annotation_cutoff = list(phylop_100way=2, gerp =2 ), 
sequence_annotation_ref = list(phylop_100way = "../other_annotation/conservation_annotation/hg19.100way.phyloP100way.bw", gerp = "../other_annotation/conservation_annotation/hg19_gerp_score.bw"),
overlap = 0.5,
rm_nonsyn = FALSE,
annovar_folder = "/media/yuwen/Elements/ANNOVAR/annovar/",
annovar_input = "no.txt",
gene_assign_file = "../other_annotation/epigenomic_annotation/Whole_genome.promoter_yanyu_10kb_exons_no_utr_no_na_genes.50bp_window.bed_genename_corrected.bed",
gene_prior_file = "../other_annotation/gene_list/TADA_SNV_CNV_combined_Feb7_tadaname_null_PPA_p0_0.94.txt",
report_proportion = 1,
mutrate_ref_file = "/media/yuwen/Elements/mutation_rate_raw/MarkDaly/Daly_mutrate.bw",
node_n = 6,
feature_start = 5,
feature_end = 9,
feature_number = 5, 
chunk_partition_num =20,
spidex_mutrate_prefix = "../other_annotation/spidex_database/mutrate_spedix_score_by_chr/lower_10pct_-1.416/spidex_public_noncommercial_v1_0_collapse_unique_base.extended_3bp.bed.fasta.trinuc.mutrate_with_spedix_gene.lower_10pct",
splicing_cutoff = -1.416,
proband_SNV_mut_spidex = proband_SNV_mut_spidex)


system(paste("rm ",prefix,"annovar_input_temp.bed",sep = ""))
system(paste("rm ",prefix, "temp.bed",sep = ""))

```


## Jinyu Wu et al.

```{r,cache=TRUE, cache.extra = r, cache=TRUE, cache.extra = adjust_mutation_rate_window}

prefix <-system("date +%s", intern = TRUE) # prefix for temporary files that will be deleted at the end of the pipeline

study_ID <- mut_with_study_label[mut_with_study_label$study == "Jinyu Wu et al.",]$index

# add study_ID , phenotyp_ID and mut_type ID as constraint

write.table(new_control$data_matrix[new_control$data_matrix$Prediction == "dnv_proband" & new_control$data_matrix$Type == "SNV" & is.element(new_control$data_matrix$ID, study_ID),c(1,2,3,7)], paste(prefix,"temp.bed",sep = ""), col.names = FALSE, row.names = FALSE, sep = "\t", quote = FALSE)
write.table(new_control$data_matrix[new_control$data_matrix$Prediction == "dnv_proband" & is.element(new_control$data_matrix$ID, study_ID),c(1:5,8,7)], paste(prefix,"annovar_input_temp.bed",sep = ""), col.names = FALSE, row.names = FALSE, sep = "\t", quote = FALSE)

# SNV_ID for cases or controls. 
SNV_ID = new_control$data_matrix[new_control$data_matrix$Prediction == "dnv_proband" & new_control$data_matrix$Type == "SNV" & is.element(new_control$data_matrix$ID, study_ID),c(1,2,3,7)]$ID
proband_SNV_mut_spidex <- new_control$mut_spedix[is.element(new_control$mut_spedix$index, SNV_ID) & is.element(new_control$mut_spedix$index, study_ID),] # this will be paseed as an argument to the function below

# first adjuste for mutation rates at 50-bp window level and then apply the scaling factor to each base of noncoding regions of selected top percentage of TADA genes.

adjusted_base_level_rr_3 <- adjust_mutation_rate_window_to_base_compact_more_feature_splicing_hybrid(mut_file = paste(prefix,"temp.bed",sep = ""),
window_file = "../other_annotation/epigenomic_annotation/Whole_genome.promoter_yanyu_10kb_exons_introns_with_spidex_no_na_genes.50bp_window.bed",
cg_file = "../other_annotation/epigenomic_annotation/Whole_genome.promoter_yanyu_10kb_exons_introns_with_spidex_no_na_genes.50bp_window.bed.fasta.cg",
mutrate_file = "../other_annotation/epigenomic_annotation/Whole_genome.promoter_yanyu_10kb_exons_introns_with_spidex_no_na_genes.50bp_window.bed.mutrate",
sample_size = 32,
epigenomic_marks = c("../other_annotation/epigenomic_annotation/Noonan_brain_roadmap_union.bed","../other_annotation/epigenomic_annotation/Epigenome_E081_E082_intersection.bed","../other_annotation/epigenomic_annotation/encode_DHS_union.bed"),
sequence_annotation = c("phylop_100way", "gerp"), 
sequence_annotation_cutoff = list(phylop_100way=2, gerp =2 ), 
sequence_annotation_ref = list(phylop_100way = "../other_annotation/conservation_annotation/hg19.100way.phyloP100way.bw", gerp = "../other_annotation/conservation_annotation/hg19_gerp_score.bw"),
overlap = 0.5,
rm_nonsyn = FALSE,
annovar_folder = "/media/yuwen/Elements/ANNOVAR/annovar/",
annovar_input = "no.txt",
gene_assign_file = "../other_annotation/epigenomic_annotation/Whole_genome.promoter_yanyu_10kb_exons_no_utr_no_na_genes.50bp_window.bed_genename_corrected.bed",
gene_prior_file = "../other_annotation/gene_list/TADA_SNV_CNV_combined_Feb7_tadaname_null_PPA_p0_0.94.txt",
report_proportion = 1,
mutrate_ref_file = "/media/yuwen/Elements/mutation_rate_raw/MarkDaly/Daly_mutrate.bw",
node_n = 6,
feature_start = 5,
feature_end = 9,
feature_number = 5, 
chunk_partition_num =20,
spidex_mutrate_prefix = "../other_annotation/spidex_database/mutrate_spedix_score_by_chr/lower_10pct_-1.416/spidex_public_noncommercial_v1_0_collapse_unique_base.extended_3bp.bed.fasta.trinuc.mutrate_with_spedix_gene.lower_10pct",
splicing_cutoff = -1.416,
proband_SNV_mut_spidex = proband_SNV_mut_spidex)


system(paste("rm ",prefix,"annovar_input_temp.bed",sep = ""))
system(paste("rm ",prefix, "temp.bed",sep = ""))

```


## Jiang YH et al. Am J Hum Genet. 2013

```{r,cache=TRUE, cache.extra = r, cache=TRUE, cache.extra = adjust_mutation_rate_window}

prefix <-system("date +%s", intern = TRUE) # prefix for temporary files that will be deleted at the end of the pipeline

study_ID <- mut_with_study_label[mut_with_study_label$study == "Jiang YH et al. Am J Hum Genet. 2013",]$index

# add study_ID , phenotyp_ID and mut_type ID as constraint

write.table(new_control$data_matrix[new_control$data_matrix$Prediction == "dnv_proband" & new_control$data_matrix$Type == "SNV" & is.element(new_control$data_matrix$ID, study_ID),c(1,2,3,7)], paste(prefix,"temp.bed",sep = ""), col.names = FALSE, row.names = FALSE, sep = "\t", quote = FALSE)
write.table(new_control$data_matrix[new_control$data_matrix$Prediction == "dnv_proband" & is.element(new_control$data_matrix$ID, study_ID),c(1:5,8,7)], paste(prefix,"annovar_input_temp.bed",sep = ""), col.names = FALSE, row.names = FALSE, sep = "\t", quote = FALSE)

# SNV_ID for cases or controls. 
SNV_ID = new_control$data_matrix[new_control$data_matrix$Prediction == "dnv_proband" & new_control$data_matrix$Type == "SNV" & is.element(new_control$data_matrix$ID, study_ID),c(1,2,3,7)]$ID
proband_SNV_mut_spidex <- new_control$mut_spedix[is.element(new_control$mut_spedix$index, SNV_ID) & is.element(new_control$mut_spedix$index, study_ID),] # this will be paseed as an argument to the function below

# first adjuste for mutation rates at 50-bp window level and then apply the scaling factor to each base of noncoding regions of selected top percentage of TADA genes.

adjusted_base_level_rr_4 <- adjust_mutation_rate_window_to_base_compact_more_feature_splicing_hybrid(mut_file = paste(prefix,"temp.bed",sep = ""),
window_file = "../other_annotation/epigenomic_annotation/Whole_genome.promoter_yanyu_10kb_exons_introns_with_spidex_no_na_genes.50bp_window.bed",
cg_file = "../other_annotation/epigenomic_annotation/Whole_genome.promoter_yanyu_10kb_exons_introns_with_spidex_no_na_genes.50bp_window.bed.fasta.cg",
mutrate_file = "../other_annotation/epigenomic_annotation/Whole_genome.promoter_yanyu_10kb_exons_introns_with_spidex_no_na_genes.50bp_window.bed.mutrate",
sample_size = 32,
epigenomic_marks = c("../other_annotation/epigenomic_annotation/Noonan_brain_roadmap_union.bed","../other_annotation/epigenomic_annotation/Epigenome_E081_E082_intersection.bed","../other_annotation/epigenomic_annotation/encode_DHS_union.bed"),
sequence_annotation = c("phylop_100way", "gerp"), 
sequence_annotation_cutoff = list(phylop_100way=2, gerp =2 ), 
sequence_annotation_ref = list(phylop_100way = "../other_annotation/conservation_annotation/hg19.100way.phyloP100way.bw", gerp = "../other_annotation/conservation_annotation/hg19_gerp_score.bw"),
overlap = 0.5,
rm_nonsyn = FALSE,
annovar_folder = "/media/yuwen/Elements/ANNOVAR/annovar/",
annovar_input = "no.txt",
gene_assign_file = "../other_annotation/epigenomic_annotation/Whole_genome.promoter_yanyu_10kb_exons_no_utr_no_na_genes.50bp_window.bed_genename_corrected.bed",
gene_prior_file = "../other_annotation/gene_list/TADA_SNV_CNV_combined_Feb7_tadaname_null_PPA_p0_0.94.txt",
report_proportion = 1,
mutrate_ref_file = "/media/yuwen/Elements/mutation_rate_raw/MarkDaly/Daly_mutrate.bw",
node_n = 6,
feature_start = 5,
feature_end = 9,
feature_number = 5, 
chunk_partition_num =20,
spidex_mutrate_prefix = "../other_annotation/spidex_database/mutrate_spedix_score_by_chr/lower_10pct_-1.416/spidex_public_noncommercial_v1_0_collapse_unique_base.extended_3bp.bed.fasta.trinuc.mutrate_with_spedix_gene.lower_10pct",
splicing_cutoff = -1.416,
proband_SNV_mut_spidex = proband_SNV_mut_spidex)


system(paste("rm ",prefix,"annovar_input_temp.bed",sep = ""))
system(paste("rm ",prefix, "temp.bed",sep = ""))

```


## Michaelson JJ et al. Cell 2012

```{r,cache=TRUE, cache.extra = r, cache=TRUE, cache.extra = adjust_mutation_rate_window}

prefix <-system("date +%s", intern = TRUE) # prefix for temporary files that will be deleted at the end of the pipeline

study_ID <- mut_with_study_label[mut_with_study_label$study == "Michaelson JJ et al. Cell 2012",]$index

# add study_ID , phenotyp_ID and mut_type ID as constraint

write.table(new_control$data_matrix[new_control$data_matrix$Prediction == "dnv_proband" & new_control$data_matrix$Type == "SNV" & is.element(new_control$data_matrix$ID, study_ID),c(1,2,3,7)], paste(prefix,"temp.bed",sep = ""), col.names = FALSE, row.names = FALSE, sep = "\t", quote = FALSE)
write.table(new_control$data_matrix[new_control$data_matrix$Prediction == "dnv_proband" & is.element(new_control$data_matrix$ID, study_ID),c(1:5,8,7)], paste(prefix,"annovar_input_temp.bed",sep = ""), col.names = FALSE, row.names = FALSE, sep = "\t", quote = FALSE)

# SNV_ID for cases or controls. 
SNV_ID = new_control$data_matrix[new_control$data_matrix$Prediction == "dnv_proband" & new_control$data_matrix$Type == "SNV" & is.element(new_control$data_matrix$ID, study_ID),c(1,2,3,7)]$ID
proband_SNV_mut_spidex <- new_control$mut_spedix[is.element(new_control$mut_spedix$index, SNV_ID) & is.element(new_control$mut_spedix$index, study_ID),] # this will be paseed as an argument to the function below

# first adjuste for mutation rates at 50-bp window level and then apply the scaling factor to each base of noncoding regions of selected top percentage of TADA genes.

adjusted_base_level_rr_5 <- adjust_mutation_rate_window_to_base_compact_more_feature_splicing_hybrid(mut_file = paste(prefix,"temp.bed",sep = ""),
window_file = "../other_annotation/epigenomic_annotation/Whole_genome.promoter_yanyu_10kb_exons_introns_with_spidex_no_na_genes.50bp_window.bed",
cg_file = "../other_annotation/epigenomic_annotation/Whole_genome.promoter_yanyu_10kb_exons_introns_with_spidex_no_na_genes.50bp_window.bed.fasta.cg",
mutrate_file = "../other_annotation/epigenomic_annotation/Whole_genome.promoter_yanyu_10kb_exons_introns_with_spidex_no_na_genes.50bp_window.bed.mutrate",
sample_size = 10,
epigenomic_marks = c("../other_annotation/epigenomic_annotation/Noonan_brain_roadmap_union.bed","../other_annotation/epigenomic_annotation/Epigenome_E081_E082_intersection.bed","../other_annotation/epigenomic_annotation/encode_DHS_union.bed"),
sequence_annotation = c("phylop_100way", "gerp"), 
sequence_annotation_cutoff = list(phylop_100way=2, gerp =2 ), 
sequence_annotation_ref = list(phylop_100way = "../other_annotation/conservation_annotation/hg19.100way.phyloP100way.bw", gerp = "../other_annotation/conservation_annotation/hg19_gerp_score.bw"),
overlap = 0.5,
rm_nonsyn = FALSE,
annovar_folder = "/media/yuwen/Elements/ANNOVAR/annovar/",
annovar_input = "no.txt",
gene_assign_file = "../other_annotation/epigenomic_annotation/Whole_genome.promoter_yanyu_10kb_exons_no_utr_no_na_genes.50bp_window.bed_genename_corrected.bed",
gene_prior_file = "../other_annotation/gene_list/TADA_SNV_CNV_combined_Feb7_tadaname_null_PPA_p0_0.94.txt",
report_proportion = 1,
mutrate_ref_file = "/media/yuwen/Elements/mutation_rate_raw/MarkDaly/Daly_mutrate.bw",
node_n = 6,
feature_start = 5,
feature_end = 9,
feature_number = 5, 
chunk_partition_num =20,
spidex_mutrate_prefix = "../other_annotation/spidex_database/mutrate_spedix_score_by_chr/lower_10pct_-1.416/spidex_public_noncommercial_v1_0_collapse_unique_base.extended_3bp.bed.fasta.trinuc.mutrate_with_spedix_gene.lower_10pct",
splicing_cutoff = -1.416,
proband_SNV_mut_spidex = proband_SNV_mut_spidex)

rm(new_control)
rm(old)
system(paste("rm ",prefix,"annovar_input_temp.bed",sep = ""))
system(paste("rm ",prefix, "temp.bed",sep = ""))

```


## Augustine Kong et al. Nature 2012
**rerun this as the first cached run shows information that the run has not been completed**
i.e., the tracking record show that `processing chromosomes` appear less than 60 times, which sould do for exactly 60 times. So the mutation information might not have been read in completedly.

```{r,cache=TRUE, cache.extra = r, cache=TRUE, cache.extra = adjust_mutation_rate_window}

prefix <-system("date +%s", intern = TRUE) # prefix for temporary files that will be deleted at the end of the pipeline

study_ID <- mut_with_study_label[mut_with_study_label$study == "Augustine Kong et al. Nature 2012",]$index

# add study_ID , phenotyp_ID and mut_type ID as constraint

write.table(new_control$data_matrix[new_control$data_matrix$Prediction == "dnv_proband" & new_control$data_matrix$Type == "SNV" & is.element(new_control$data_matrix$ID, study_ID),c(1,2,3,7)], paste(prefix,"temp.bed",sep = ""), col.names = FALSE, row.names = FALSE, sep = "\t", quote = FALSE)
write.table(new_control$data_matrix[new_control$data_matrix$Prediction == "dnv_proband" & is.element(new_control$data_matrix$ID, study_ID),c(1:5,8,7)], paste(prefix,"annovar_input_temp.bed",sep = ""), col.names = FALSE, row.names = FALSE, sep = "\t", quote = FALSE)

# SNV_ID for cases or controls. 
SNV_ID = new_control$data_matrix[new_control$data_matrix$Prediction == "dnv_proband" & new_control$data_matrix$Type == "SNV" & is.element(new_control$data_matrix$ID, study_ID),c(1,2,3,7)]$ID
proband_SNV_mut_spidex <- new_control$mut_spedix[is.element(new_control$mut_spedix$index, SNV_ID) & is.element(new_control$mut_spedix$index, study_ID),] # this will be paseed as an argument to the function below

# first adjuste for mutation rates at 50-bp window level and then apply the scaling factor to each base of noncoding regions of selected top percentage of TADA genes.

adjusted_base_level_rr_2 <- adjust_mutation_rate_window_to_base_compact_more_feature_splicing_hybrid(mut_file = paste(prefix,"temp.bed",sep = ""),
window_file = "../other_annotation/epigenomic_annotation/Whole_genome.promoter_yanyu_10kb_exons_introns_with_spidex_no_na_genes.50bp_window.bed",
cg_file = "../other_annotation/epigenomic_annotation/Whole_genome.promoter_yanyu_10kb_exons_introns_with_spidex_no_na_genes.50bp_window.bed.fasta.cg",
mutrate_file = "../other_annotation/epigenomic_annotation/Whole_genome.promoter_yanyu_10kb_exons_introns_with_spidex_no_na_genes.50bp_window.bed.mutrate",
sample_size = 78,
epigenomic_marks = c("../other_annotation/epigenomic_annotation/Noonan_brain_roadmap_union.bed","../other_annotation/epigenomic_annotation/Epigenome_E081_E082_intersection.bed","../other_annotation/epigenomic_annotation/encode_DHS_union.bed"),
sequence_annotation = c("phylop_100way", "gerp"), 
sequence_annotation_cutoff = list(phylop_100way=2, gerp =2 ), 
sequence_annotation_ref = list(phylop_100way = "../other_annotation/conservation_annotation/hg19.100way.phyloP100way.bw", gerp = "../other_annotation/conservation_annotation/hg19_gerp_score.bw"),
overlap = 0.5,
rm_nonsyn = FALSE,
annovar_folder = "/media/yuwen/Elements/ANNOVAR/annovar/",
annovar_input = "no.txt",
gene_assign_file = "../other_annotation/epigenomic_annotation/Whole_genome.promoter_yanyu_10kb_exons_no_utr_no_na_genes.50bp_window.bed_genename_corrected.bed",
gene_prior_file = "../other_annotation/gene_list/TADA_SNV_CNV_combined_Feb7_tadaname_null_PPA_p0_0.94.txt",
report_proportion = 1,
mutrate_ref_file = "/media/yuwen/Elements/mutation_rate_raw/MarkDaly/Daly_mutrate.bw",
node_n = 6,
feature_start = 5,
feature_end = 9,
feature_number = 5, 
chunk_partition_num =20,
spidex_mutrate_prefix = "../other_annotation/spidex_database/mutrate_spedix_score_by_chr/lower_10pct_-1.416/spidex_public_noncommercial_v1_0_collapse_unique_base.extended_3bp.bed.fasta.trinuc.mutrate_with_spedix_gene.lower_10pct",
splicing_cutoff = -1.416,
proband_SNV_mut_spidex = proband_SNV_mut_spidex)


system(paste("rm ",prefix,"annovar_input_temp.bed",sep = ""))
system(paste("rm ",prefix, "temp.bed",sep = ""))

```


# Calculate Bayes Factors from non-coding features
(not including allele-specific mutations, i.e., splicing mutations here)
RR esitated from [here](170408_estimate_relative_risk_top1000_genes_from_categorization_from_each_separate_study.html)

```{r}
source("../lib/TADA_denovo.R")
```
```{r,cache=TRUE}
source("../lib/170117_gene_validation_functions.R")
```

```{r}
mut_collapsed_data_list <- list(adjusted_base_level_rr_1$base_info,
                                adjusted_base_level_rr_2$base_info,
                                adjusted_base_level_rr_3$base_info,
                                adjusted_base_level_rr_4$base_info,
                                adjusted_base_level_rr_5$base_info)


TADA_table<-bayes_factor_for_each_gene_base_level_from_collapsed_data_multiple_studies(mut_collapsed_data_list = mut_collapsed_data_list, 
                                                                      features = c(1), 
                                                                      rr = 0.6258986,
                                                                      coding_bayes_table_file = "../other_annotation/gene_list/TADA_SNV_CNV_combined_Feb7_tadaname_BF_for_de_novo_mutations.txt",
                                                                      TADA_p0 = 0.94)

```

### test
```{r}
length(adjusted_base_level_rr_1$splicing_info)
length(adjusted_base_level_rr_2$splicing_info)
length(adjusted_base_level_rr_3$splicing_info)
length(adjusted_base_level_rr_4$splicing_info)
length(adjusted_base_level_rr_5$splicing_info)

length(adjusted_base_level_rr_1$base_info)
length(adjusted_base_level_rr_2$base_info)
length(adjusted_base_level_rr_3$base_info)
length(adjusted_base_level_rr_4$base_info)
length(adjusted_base_level_rr_5$base_info)

```


** Use [adjust_mutation_rate_window_to_base_compact_more_feature_splicing_hybrid_patch] to correct for the splicing info incorrectly returned from [adjust_mutation_rate_window_to_base_compact_more_feature_splicing_hybrid]**  


# Read in data from 5 studies
```{r}
new_control = new.env()
load("../data/debug_region_list_073116_data_matrixtransformed_for_old_code.Rdata", new_control) # this Rdata has spedix info
# add in the 0-based start position

new_control$mutation$start_0 <- as.numeric(as.character(new_control$mutation$start)) - 1

# mutation data in the old format, I loaded this just in order to retrive information about which study each mutation comes from
old <- new.env()
load("../data/debug_region_list_073116_data_matrix.Rdata", old)

# Use a merge function to associate mutations with study information
# I merged by matching the chromosome locations, including chr, start and end. For the case SNV data, I removed all redundant mutations, so merging in this way would ensure a 1-to-1 matching between mutation info and study info
# There is redundant SNVs and other mutations in old$mutation$mutation, and there are redundant non-SNV mutations in new_control$mutation, but I didn't use non-SNV mutations in this study. So having such type of redundant mutations should be fine.

mut_with_study_label <- merge(new_control$mutation[new_control$mutation$phenotype =="ASD" & new_control$mutation$mut_type == "SNV",], old$mutation$mutation[old$mutation$mutation$phenotype == "ASD" & old$mutation$mutation$mut_type == "SNV",], by.x = c("chr","start_0","end"), by.y = c("chr","start","end"))
# The index is consistent with new_control$mutation, but not with old$mutation$mutation
mut_with_study_label <- data.frame(index = mut_with_study_label$index.x, study = mut_with_study_label$study.y)

ASD_study_name <- as.vector(unique(mut_with_study_label$study)[1:5])

```


## Ryan K C Yuen et al. Nature Medicine 2015

```{r,cache=TRUE, cache.extra = r, cache=TRUE, cache.extra = adjust_mutation_rate_window_to_base_compact_more_feature_splicing_hybrid_patch}

prefix <-system("date +%s", intern = TRUE) # prefix for temporary files that will be deleted at the end of the pipeline

study_ID <- mut_with_study_label[mut_with_study_label$study == "Ryan K C Yuen et al. Nature Medicine 2015",]$index

# add study_ID , phenotyp_ID and mut_type ID as constraint

write.table(new_control$data_matrix[new_control$data_matrix$Prediction == "dnv_proband" & new_control$data_matrix$Type == "SNV" & is.element(new_control$data_matrix$ID, study_ID),c(1,2,3,7)], paste(prefix,"temp.bed",sep = ""), col.names = FALSE, row.names = FALSE, sep = "\t", quote = FALSE)
write.table(new_control$data_matrix[new_control$data_matrix$Prediction == "dnv_proband" & is.element(new_control$data_matrix$ID, study_ID),c(1:5,8,7)], paste(prefix,"annovar_input_temp.bed",sep = ""), col.names = FALSE, row.names = FALSE, sep = "\t", quote = FALSE)

# SNV_ID for cases or controls. 
SNV_ID = new_control$data_matrix[new_control$data_matrix$Prediction == "dnv_proband" & new_control$data_matrix$Type == "SNV" & is.element(new_control$data_matrix$ID, study_ID),c(1,2,3,7)]$ID
proband_SNV_mut_spidex <- new_control$mut_spedix[is.element(new_control$mut_spedix$index, SNV_ID) & is.element(new_control$mut_spedix$index, study_ID),] # this will be paseed as an argument to the function below

# first adjuste for mutation rates at 50-bp window level and then apply the scaling factor to each base of noncoding regions of selected top percentage of TADA genes.

adjusted_base_level_rr_1 <- adjust_mutation_rate_window_to_base_compact_more_feature_splicing_hybrid_patch(mut_file = paste(prefix,"temp.bed",sep = ""),
window_file = "../other_annotation/epigenomic_annotation/Whole_genome.promoter_yanyu_10kb_exons_introns_with_spidex_no_na_genes.50bp_window.bed",
cg_file = "../other_annotation/epigenomic_annotation/Whole_genome.promoter_yanyu_10kb_exons_introns_with_spidex_no_na_genes.50bp_window.bed.fasta.cg",
mutrate_file = "../other_annotation/epigenomic_annotation/Whole_genome.promoter_yanyu_10kb_exons_introns_with_spidex_no_na_genes.50bp_window.bed.mutrate",
sample_size = 162,
overlap = 0.5,
rm_nonsyn = FALSE,
annovar_folder = "/media/yuwen/Elements/ANNOVAR/annovar/",
annovar_input = "no.txt",
gene_prior_file = "../other_annotation/gene_list/TADA_SNV_CNV_combined_Feb7_tadaname_null_PPA_p0_0.94.txt",
report_proportion = 1,
spidex_mutrate_prefix = "../other_annotation/spidex_database/mutrate_spedix_score_by_chr/lower_10pct_-1.416/spidex_public_noncommercial_v1_0_collapse_unique_base.extended_3bp.bed.fasta.trinuc.mutrate_with_spedix_gene.lower_10pct",
splicing_cutoff = -1.416,
proband_SNV_mut_spidex = proband_SNV_mut_spidex)

system(paste("rm ",prefix,"annovar_input_temp.bed",sep = ""))
system(paste("rm ",prefix, "temp.bed",sep = ""))

```

## Augustine Kong et al. Nature 2012

```{r,cache=TRUE, cache.extra = r, cache=TRUE, cache.extra = adjust_mutation_rate_window_to_base_compact_more_feature_splicing_hybrid_patch}

prefix <-system("date +%s", intern = TRUE) # prefix for temporary files that will be deleted at the end of the pipeline

study_ID <- mut_with_study_label[mut_with_study_label$study == "Augustine Kong et al. Nature 2012",]$index

# add study_ID , phenotyp_ID and mut_type ID as constraint

write.table(new_control$data_matrix[new_control$data_matrix$Prediction == "dnv_proband" & new_control$data_matrix$Type == "SNV" & is.element(new_control$data_matrix$ID, study_ID),c(1,2,3,7)], paste(prefix,"temp.bed",sep = ""), col.names = FALSE, row.names = FALSE, sep = "\t", quote = FALSE)
write.table(new_control$data_matrix[new_control$data_matrix$Prediction == "dnv_proband" & is.element(new_control$data_matrix$ID, study_ID),c(1:5,8,7)], paste(prefix,"annovar_input_temp.bed",sep = ""), col.names = FALSE, row.names = FALSE, sep = "\t", quote = FALSE)

# SNV_ID for cases or controls. 
SNV_ID = new_control$data_matrix[new_control$data_matrix$Prediction == "dnv_proband" & new_control$data_matrix$Type == "SNV" & is.element(new_control$data_matrix$ID, study_ID),c(1,2,3,7)]$ID
proband_SNV_mut_spidex <- new_control$mut_spedix[is.element(new_control$mut_spedix$index, SNV_ID) & is.element(new_control$mut_spedix$index, study_ID),] # this will be paseed as an argument to the function below

# first adjuste for mutation rates at 50-bp window level and then apply the scaling factor to each base of noncoding regions of selected top percentage of TADA genes.

adjusted_base_level_rr_2 <- adjust_mutation_rate_window_to_base_compact_more_feature_splicing_hybrid_patch(mut_file = paste(prefix,"temp.bed",sep = ""),
window_file = "../other_annotation/epigenomic_annotation/Whole_genome.promoter_yanyu_10kb_exons_introns_with_spidex_no_na_genes.50bp_window.bed",
cg_file = "../other_annotation/epigenomic_annotation/Whole_genome.promoter_yanyu_10kb_exons_introns_with_spidex_no_na_genes.50bp_window.bed.fasta.cg",
mutrate_file = "../other_annotation/epigenomic_annotation/Whole_genome.promoter_yanyu_10kb_exons_introns_with_spidex_no_na_genes.50bp_window.bed.mutrate",
sample_size = 78,
overlap = 0.5,
rm_nonsyn = FALSE,
annovar_folder = "/media/yuwen/Elements/ANNOVAR/annovar/",
annovar_input = "no.txt",
gene_prior_file = "../other_annotation/gene_list/TADA_SNV_CNV_combined_Feb7_tadaname_null_PPA_p0_0.94.txt",
report_proportion = 1,
spidex_mutrate_prefix = "../other_annotation/spidex_database/mutrate_spedix_score_by_chr/lower_10pct_-1.416/spidex_public_noncommercial_v1_0_collapse_unique_base.extended_3bp.bed.fasta.trinuc.mutrate_with_spedix_gene.lower_10pct",
splicing_cutoff = -1.416,
proband_SNV_mut_spidex = proband_SNV_mut_spidex)


system(paste("rm ",prefix,"annovar_input_temp.bed",sep = ""))
system(paste("rm ",prefix, "temp.bed",sep = ""))

```


## Jinyu Wu et al.

```{r,cache=TRUE, cache.extra = r, cache=TRUE, cache.extra = adjust_mutation_rate_window_to_base_compact_more_feature_splicing_hybrid_patch}

prefix <-system("date +%s", intern = TRUE) # prefix for temporary files that will be deleted at the end of the pipeline

study_ID <- mut_with_study_label[mut_with_study_label$study == "Jinyu Wu et al.",]$index

# add study_ID , phenotyp_ID and mut_type ID as constraint

write.table(new_control$data_matrix[new_control$data_matrix$Prediction == "dnv_proband" & new_control$data_matrix$Type == "SNV" & is.element(new_control$data_matrix$ID, study_ID),c(1,2,3,7)], paste(prefix,"temp.bed",sep = ""), col.names = FALSE, row.names = FALSE, sep = "\t", quote = FALSE)
write.table(new_control$data_matrix[new_control$data_matrix$Prediction == "dnv_proband" & is.element(new_control$data_matrix$ID, study_ID),c(1:5,8,7)], paste(prefix,"annovar_input_temp.bed",sep = ""), col.names = FALSE, row.names = FALSE, sep = "\t", quote = FALSE)

# SNV_ID for cases or controls. 
SNV_ID = new_control$data_matrix[new_control$data_matrix$Prediction == "dnv_proband" & new_control$data_matrix$Type == "SNV" & is.element(new_control$data_matrix$ID, study_ID),c(1,2,3,7)]$ID
proband_SNV_mut_spidex <- new_control$mut_spedix[is.element(new_control$mut_spedix$index, SNV_ID) & is.element(new_control$mut_spedix$index, study_ID),] # this will be paseed as an argument to the function below

# first adjuste for mutation rates at 50-bp window level and then apply the scaling factor to each base of noncoding regions of selected top percentage of TADA genes.

adjusted_base_level_rr_3 <- adjust_mutation_rate_window_to_base_compact_more_feature_splicing_hybrid_patch(mut_file = paste(prefix,"temp.bed",sep = ""),
window_file = "../other_annotation/epigenomic_annotation/Whole_genome.promoter_yanyu_10kb_exons_introns_with_spidex_no_na_genes.50bp_window.bed",
cg_file = "../other_annotation/epigenomic_annotation/Whole_genome.promoter_yanyu_10kb_exons_introns_with_spidex_no_na_genes.50bp_window.bed.fasta.cg",
mutrate_file = "../other_annotation/epigenomic_annotation/Whole_genome.promoter_yanyu_10kb_exons_introns_with_spidex_no_na_genes.50bp_window.bed.mutrate",
sample_size = 32,
overlap = 0.5,
rm_nonsyn = FALSE,
annovar_folder = "/media/yuwen/Elements/ANNOVAR/annovar/",
annovar_input = "no.txt",
gene_prior_file = "../other_annotation/gene_list/TADA_SNV_CNV_combined_Feb7_tadaname_null_PPA_p0_0.94.txt",
report_proportion = 1,
spidex_mutrate_prefix = "../other_annotation/spidex_database/mutrate_spedix_score_by_chr/lower_10pct_-1.416/spidex_public_noncommercial_v1_0_collapse_unique_base.extended_3bp.bed.fasta.trinuc.mutrate_with_spedix_gene.lower_10pct",
splicing_cutoff = -1.416,
proband_SNV_mut_spidex = proband_SNV_mut_spidex)


system(paste("rm ",prefix,"annovar_input_temp.bed",sep = ""))
system(paste("rm ",prefix, "temp.bed",sep = ""))

```


## Jiang YH et al. Am J Hum Genet. 2013

```{r,cache=TRUE, cache.extra = r, cache=TRUE, cache.extra = adjust_mutation_rate_window_to_base_compact_more_feature_splicing_hybrid_patch}

prefix <-system("date +%s", intern = TRUE) # prefix for temporary files that will be deleted at the end of the pipeline

study_ID <- mut_with_study_label[mut_with_study_label$study == "Jiang YH et al. Am J Hum Genet. 2013",]$index

# add study_ID , phenotyp_ID and mut_type ID as constraint

write.table(new_control$data_matrix[new_control$data_matrix$Prediction == "dnv_proband" & new_control$data_matrix$Type == "SNV" & is.element(new_control$data_matrix$ID, study_ID),c(1,2,3,7)], paste(prefix,"temp.bed",sep = ""), col.names = FALSE, row.names = FALSE, sep = "\t", quote = FALSE)
write.table(new_control$data_matrix[new_control$data_matrix$Prediction == "dnv_proband" & is.element(new_control$data_matrix$ID, study_ID),c(1:5,8,7)], paste(prefix,"annovar_input_temp.bed",sep = ""), col.names = FALSE, row.names = FALSE, sep = "\t", quote = FALSE)

# SNV_ID for cases or controls. 
SNV_ID = new_control$data_matrix[new_control$data_matrix$Prediction == "dnv_proband" & new_control$data_matrix$Type == "SNV" & is.element(new_control$data_matrix$ID, study_ID),c(1,2,3,7)]$ID
proband_SNV_mut_spidex <- new_control$mut_spedix[is.element(new_control$mut_spedix$index, SNV_ID) & is.element(new_control$mut_spedix$index, study_ID),] # this will be paseed as an argument to the function below

# first adjuste for mutation rates at 50-bp window level and then apply the scaling factor to each base of noncoding regions of selected top percentage of TADA genes.

adjusted_base_level_rr_4 <- adjust_mutation_rate_window_to_base_compact_more_feature_splicing_hybrid_patch(mut_file = paste(prefix,"temp.bed",sep = ""),
window_file = "../other_annotation/epigenomic_annotation/Whole_genome.promoter_yanyu_10kb_exons_introns_with_spidex_no_na_genes.50bp_window.bed",
cg_file = "../other_annotation/epigenomic_annotation/Whole_genome.promoter_yanyu_10kb_exons_introns_with_spidex_no_na_genes.50bp_window.bed.fasta.cg",
mutrate_file = "../other_annotation/epigenomic_annotation/Whole_genome.promoter_yanyu_10kb_exons_introns_with_spidex_no_na_genes.50bp_window.bed.mutrate",
sample_size = 32,
overlap = 0.5,
rm_nonsyn = FALSE,
annovar_folder = "/media/yuwen/Elements/ANNOVAR/annovar/",
annovar_input = "no.txt",
gene_prior_file = "../other_annotation/gene_list/TADA_SNV_CNV_combined_Feb7_tadaname_null_PPA_p0_0.94.txt",
report_proportion = 1,
spidex_mutrate_prefix = "../other_annotation/spidex_database/mutrate_spedix_score_by_chr/lower_10pct_-1.416/spidex_public_noncommercial_v1_0_collapse_unique_base.extended_3bp.bed.fasta.trinuc.mutrate_with_spedix_gene.lower_10pct",
splicing_cutoff = -1.416,
proband_SNV_mut_spidex = proband_SNV_mut_spidex)


system(paste("rm ",prefix,"annovar_input_temp.bed",sep = ""))
system(paste("rm ",prefix, "temp.bed",sep = ""))

```


## Michaelson JJ et al. Cell 2012

```{r,cache=TRUE, cache.extra = r, cache=TRUE, cache.extra = adjust_mutation_rate_window_to_base_compact_more_feature_splicing_hybrid_patch}

prefix <-system("date +%s", intern = TRUE) # prefix for temporary files that will be deleted at the end of the pipeline

study_ID <- mut_with_study_label[mut_with_study_label$study == "Michaelson JJ et al. Cell 2012",]$index

# add study_ID , phenotyp_ID and mut_type ID as constraint

write.table(new_control$data_matrix[new_control$data_matrix$Prediction == "dnv_proband" & new_control$data_matrix$Type == "SNV" & is.element(new_control$data_matrix$ID, study_ID),c(1,2,3,7)], paste(prefix,"temp.bed",sep = ""), col.names = FALSE, row.names = FALSE, sep = "\t", quote = FALSE)
write.table(new_control$data_matrix[new_control$data_matrix$Prediction == "dnv_proband" & is.element(new_control$data_matrix$ID, study_ID),c(1:5,8,7)], paste(prefix,"annovar_input_temp.bed",sep = ""), col.names = FALSE, row.names = FALSE, sep = "\t", quote = FALSE)

# SNV_ID for cases or controls. 
SNV_ID = new_control$data_matrix[new_control$data_matrix$Prediction == "dnv_proband" & new_control$data_matrix$Type == "SNV" & is.element(new_control$data_matrix$ID, study_ID),c(1,2,3,7)]$ID
proband_SNV_mut_spidex <- new_control$mut_spedix[is.element(new_control$mut_spedix$index, SNV_ID) & is.element(new_control$mut_spedix$index, study_ID),] # this will be paseed as an argument to the function below

# first adjuste for mutation rates at 50-bp window level and then apply the scaling factor to each base of noncoding regions of selected top percentage of TADA genes.

adjusted_base_level_rr_5 <- adjust_mutation_rate_window_to_base_compact_more_feature_splicing_hybrid_patch(mut_file = paste(prefix,"temp.bed",sep = ""),
window_file = "../other_annotation/epigenomic_annotation/Whole_genome.promoter_yanyu_10kb_exons_introns_with_spidex_no_na_genes.50bp_window.bed",
cg_file = "../other_annotation/epigenomic_annotation/Whole_genome.promoter_yanyu_10kb_exons_introns_with_spidex_no_na_genes.50bp_window.bed.fasta.cg",
mutrate_file = "../other_annotation/epigenomic_annotation/Whole_genome.promoter_yanyu_10kb_exons_introns_with_spidex_no_na_genes.50bp_window.bed.mutrate",
sample_size = 10,
overlap = 0.5,
rm_nonsyn = FALSE,
annovar_folder = "/media/yuwen/Elements/ANNOVAR/annovar/",
annovar_input = "no.txt",
gene_prior_file = "../other_annotation/gene_list/TADA_SNV_CNV_combined_Feb7_tadaname_null_PPA_p0_0.94.txt",
report_proportion = 1,
spidex_mutrate_prefix = "../other_annotation/spidex_database/mutrate_spedix_score_by_chr/lower_10pct_-1.416/spidex_public_noncommercial_v1_0_collapse_unique_base.extended_3bp.bed.fasta.trinuc.mutrate_with_spedix_gene.lower_10pct",
splicing_cutoff = -1.416,
proband_SNV_mut_spidex = proband_SNV_mut_spidex)

rm(new_control)
rm(old)
system(paste("rm ",prefix,"annovar_input_temp.bed",sep = ""))
system(paste("rm ",prefix, "temp.bed",sep = ""))

```

## corrected splicing info for genes

```{r}
splicing_data_list = list(adjusted_base_level_rr_1$splicing_info, adjusted_base_level_rr_2$splicing_info, adjusted_base_level_rr_3$splicing_info, adjusted_base_level_rr_4$splicing_info, adjusted_base_level_rr_5$splicing_info)

```

```{r}
length(adjusted_base_level_rr_1$splicing_info)
length(adjusted_base_level_rr_2$splicing_info)
length(adjusted_base_level_rr_3$splicing_info)
length(adjusted_base_level_rr_4$splicing_info)
length(adjusted_base_level_rr_5$splicing_info)
```

## generate corrected splicing BF for genes
```{r}
splicing_BF_table <- generate_BF_for_splicing_mutations_multiple_studies(splicing_data_list, 1.1692002)
```

## *cacluate the bayes factor table for splicing mutations, summing mutation rate from ChrX and ChrY for a gene if it is on both ChrX and ChrY* 


```{r}
### currently the table below has duplicated entries, because there are a few genes that have appeared both on X chromosome and Y chromosome. 
### Now I sum up the matation rates from both chrx and chrY for such genes. 

head(splicing_BF_table$splicing_BF_table)
splicing_BF_table_collapsed <- by(splicing_BF_table$splicing_BF_table, splicing_BF_table$splicing_BF_table$genename, function(x) sum(x$logBF_splicing)) 
#splicing_BF_table_collapsed 
splicing_BF_table_collapsed <- data.table(genename = names(splicing_BF_table_collapsed), logBF_splicing = as.vector(splicing_BF_table_collapsed), BF_splicing = exp(as.vector(splicing_BF_table_collapsed)))

```


## *now combine information from splicing (collapsed version) to other non_coding mutations*
```{r}
combined_BF_table <- splicing_BF_table_collapsed[TADA_table$gene_BF_table, on = "genename"]
combined_BF_table

# get a list of genes that don't have splicing info.
genes_with_no_info_from_splicing_BF_table <- combined_BF_table[is.na(combined_BF_table$logBF_splicing),]$genename

# change the logBF_splicing of these genes to 0 and BF of these genes to 1
combined_BF_table[is.na(combined_BF_table$logBF_splicing),][,c("logBF_splicing")] = 0
combined_BF_table[is.na(combined_BF_table$BF_splicing),][,c("BF_splicing")] = 1

combined_BF_table[is.element(combined_BF_table$genename, genes_with_no_info_from_splicing_BF_table),]
# add splicing info to BF_all
combined_BF_table$logBF_all <- combined_BF_table$logBF_splicing + combined_BF_table$logBF_noncoding + combined_BF_table$logBF_coding
combined_BF_table$BF_all <- exp(combined_BF_table$logBF_all)
combined_BF_table <- combined_BF_table[order(combined_BF_table$BF_all, decreasing = TRUE),]
combined_BF_table$FDR_all = Bayesian.FDR(combined_BF_table$BF_all, 0.94)$FDR # here 0.94 was chosen to be consisten with the prior used for getting BF and FDR using other types of mutations. 

head(combined_BF_table)
```

### Number of gained and lost genes after having non-coding mutations and splicing mutations
```{r}
TADA_table_filtered =combined_BF_table[!is.element(combined_BF_table$genename, TADA_table$genes_with_no_epi),]
knitr::kable(performance_table(TADA_table_filtered))
```

### FDR 0.1 new genes
```{r}
knitr::kable(TADA_table_filtered[TADA_table_filtered$FDR_coding >= 0.1 & TADA_table_filtered$FDR_all < 0.1,])
```

### FDR 0.2 new genes
```{r}
knitr::kable(TADA_table_filtered[TADA_table_filtered$FDR_coding >= 0.2 & TADA_table_filtered$FDR_all < 0.2,])
```

### FDR 0.3 new genes
```{r}
knitr::kable(TADA_table_filtered[TADA_table_filtered$FDR_coding >= 0.3 & TADA_table_filtered$FDR_all < 0.3,])
```

### Gene set enrichment analysis FDR 0.1
```{r}
knitr::kable(foe_table(TADA_table_filtered[TADA_table_filtered$FDR_coding >=0.1 & TADA_table_filtered$FDR_all < 0.1, ]$genename, TADA_table_filtered$genename))
```

### Gene set enrichment analysis FDR 0.2
```{r}
knitr::kable(foe_table(TADA_table_filtered[TADA_table_filtered$FDR_coding >=0.2 & TADA_table_filtered$FDR_all < 0.2, ]$genename, TADA_table_filtered$genename))
```

### Gene set enrichment analysis FDR 0.3
```{r}
knitr::kable(foe_table(TADA_table_filtered[TADA_table_filtered$FDR_coding >=0.3 & TADA_table_filtered$FDR_all < 0.3, ]$genename, TADA_table_filtered$genename))
```

### Gene set enrichment analysis FDR 0.1, all genes background
```{r}
knitr::kable(foe_table(TADA_table_filtered[TADA_table_filtered$FDR_coding >=0.1 & TADA_table_filtered$FDR_all < 0.1, ]$genename, combined_BF_table$genename))
```

### Gene set enrichment analysis FDR 0.2, all genes background
```{r}
knitr::kable(foe_table(TADA_table_filtered[TADA_table_filtered$FDR_coding >=0.2 & TADA_table_filtered$FDR_all < 0.2, ]$genename, combined_BF_table$genename))
```

### Gene set enrichment analysis FDR 0.3, all genes background
```{r}
knitr::kable(foe_table(TADA_table_filtered[TADA_table_filtered$FDR_coding >=0.3 & TADA_table_filtered$FDR_all < 0.3, ]$genename, combined_BF_table$genename))
```

### Gene set enrichment analysis FDR 0.3, all genes background, remove SIRPB1 which doesn't have splicing mutations or H3k27ac mutations
```{r}
knitr::kable(foe_table(TADA_table_filtered[TADA_table_filtered$FDR_coding >=0.3 & TADA_table_filtered$FDR_all < 0.3 & TADA_table_filtered$genename != "SIRPB2", ]$genename, combined_BF_table$genename))
```


## Number of gained and lost genes, remove genes that don't have have splicing mutations or noncoding mutations, and don't have K27ac, splicing RR 2.4
The qvalue of these genes just change from very little greater than 0.3 to very little bigger than 0.3
##
```{r}
TADA_table_filtered =combined_BF_table[!is.element(combined_BF_table$genename, TADA_table$genes_with_no_epi),]
TADA_table_filtered = TADA_table_filtered[genename != "SIRPB2" & genename != "NUDT17"]
knitr::kable(performance_table(TADA_table_filtered))
```

### FDR 0.1 new genes
```{r}
knitr::kable(TADA_table_filtered[TADA_table_filtered$FDR_coding >= 0.1 & TADA_table_filtered$FDR_all < 0.1,])
```

### FDR 0.2 new genes
```{r}
knitr::kable(TADA_table_filtered[TADA_table_filtered$FDR_coding >= 0.2 & TADA_table_filtered$FDR_all < 0.2,])
```

### FDR 0.3 new genes
```{r}
knitr::kable(TADA_table_filtered[TADA_table_filtered$FDR_coding >= 0.3 & TADA_table_filtered$FDR_all < 0.3,])
```

### Gene set enrichment analysis FDR 0.1
```{r}
knitr::kable(foe_table(TADA_table_filtered[TADA_table_filtered$FDR_coding >=0.1 & TADA_table_filtered$FDR_all < 0.1, ]$genename, TADA_table_filtered$genename))
```

### Gene set enrichment analysis FDR 0.2
```{r}
knitr::kable(foe_table(TADA_table_filtered[TADA_table_filtered$FDR_coding >=0.2 & TADA_table_filtered$FDR_all < 0.2, ]$genename, TADA_table_filtered$genename))
```

### Gene set enrichment analysis FDR 0.3
```{r}
knitr::kable(foe_table(TADA_table_filtered[TADA_table_filtered$FDR_coding >=0.3 & TADA_table_filtered$FDR_all < 0.3, ]$genename, TADA_table_filtered$genename))
```

### Gene set enrichment analysis FDR 0.1, all genes background
```{r}
knitr::kable(foe_table(TADA_table_filtered[TADA_table_filtered$FDR_coding >=0.1 & TADA_table_filtered$FDR_all < 0.1, ]$genename, combined_BF_table$genename))
```

### Gene set enrichment analysis FDR 0.2, all genes background
```{r}
knitr::kable(foe_table(TADA_table_filtered[TADA_table_filtered$FDR_coding >=0.2 & TADA_table_filtered$FDR_all < 0.2, ]$genename, combined_BF_table$genename))
```

### Gene set enrichment analysis FDR 0.3, all genes background
```{r}
knitr::kable(foe_table(TADA_table_filtered[TADA_table_filtered$FDR_coding >=0.3 & TADA_table_filtered$FDR_all < 0.3, ]$genename, combined_BF_table$genename))
```

