<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title></title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/yeti.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>




<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 45px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 50px;
  margin-top: -50px;
}

.section h2 {
  padding-top: 50px;
  margin-top: -50px;
}
.section h3 {
  padding-top: 50px;
  margin-top: -50px;
}
.section h4 {
  padding-top: 50px;
  margin-top: -50px;
}
.section h5 {
  padding-top: 50px;
  margin-top: -50px;
}
.section h6 {
  padding-top: 50px;
  margin-top: -50px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Collaboration J Project</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="meeting.html">Meeting</a>
</li>
<li>
  <a href="Note.html">Note</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">




</div>


<h1 style="color:grey;font-family:arial; position:relative; top:15px;text-align:center;">
ASD project<img src="website_materials/research_website_logo.png"  style="float:right;width:200px;height:150px;">
</h1>
<p><br></p>
<hr />
<div id="adjust-mutation-rate-using-glm-50bp-window" class="section level3">
<h3>1. Adjust mutation rate using GLM: 50bp window</h3>
<p><br></p>
<ul>
<li><p><a href="./161120_mutation_rate_calibration_removing_utr_and_na_genes.html">Study the effect of promoter, coding, and cg content on mutation rate.For promoters and coding windows, remove those that have “n/a” as gene names, for coding also remove utrs, all types of coding mutations were used for cases</a></p></li>
<li><p><a href="./161120_mutation_rate_calibration_removing_utr_and_na_genes_with_epigenomic_mark.html">Study the effect of promoter, coding, and cg content and Noonan+Rodamap brain H3K27ac on mutation rate.For promoters and coding windows, remove those that have “n/a” as gene names, for coding also remove utrs, all types of coding mutations were used for cases</a></p></li>
<li><p><a href="./161120_mutation_rate_calibration_removing_utr_and_na_genes_with_epigenomic_mark_remove_nonsyn_for_cases.html">Study the effect of promoter, coding, and cg content and Noonan+Rodamap brain H3K27ac on mutation rate.For promoters and coding windows, remove those that have “n/a” as gene names, for coding also remove utrs, only synonymouse mutations were used for training the model for cases</a></p></li>
<li><p><a href="./161205_mutation_rate_10fold_likelihood_with_Noonan_brain_roadmap_union_1bp_overlap_using_synony_only_for_ASD_coding.html">10-fold loglikelihood-ratio comparing trinucleotide model and our adjusted model (features include promoter, coding, cg content and Noonan+Brain H3K27ac, using synonymous mutations for training cases)</a></p></li>
</ul>
</div>
<div id="compare-number-of-observed-mutations-of-different-types-with-expected-based-on-adjusted-mutation-rate" class="section level3">
<h3>2. Compare number of observed mutations of different types with expected based on adjusted mutation rate</h3>
<ul>
<li><p><a href="./161121_compare_observed_mutation_number_with_calibrated_mutation_rate_for_all_genes_all_categories.html">Compare all types of mutations. Use only 1 epi data for mutation rate adjustment, use all coding mutations for training in cases</a></p></li>
<li><p><a href="./161201_compare_observed_coding_mutation_number_with_calibrated_mutation_rate_for_Noonan_Roadmap_brain_syn_trained_for_cases.html">Compare only coding mutations. Use only 1 epi data for mutation rate adjustment, only use synonymous mutations for coding mutations in cases for training</a></p></li>
</ul>
</div>
<div id="effect-size-estimation-from-simulation-data-50bp-window" class="section level3">
<h3>3. Effect size estimation from simulation data: 50bp-window</h3>
<ul>
<li><p>[Use all TADA genes]</p></li>
<li><p><a href="./161214_161214_simulate_mutation_and_estimate_effect_top6_pct_with_Noonan_brain_roadmap_union_1bp_overlap_using_synony_only_for_ASD_coding.html">Use only top 6% TADA genes, simulate dataset and do optimization multiple times and compare the distribution of paramter estimates with true values</a></p></li>
</ul>
</div>
<div id="effect-size-estimation-from-real-data-50bp-window" class="section level3">
<h3>4. Effect size estimation from real data: 50bp-window</h3>
<ul>
<li><p><a href="./161211_estimate_effect_with_Noonan_brain_roadmap_union_1bp_overlap_using_synony_only_for_ASD_coding.html">ASD_manuscript data, 1 epigenome data sets first adjust mutation rate, then esitamte relative risks using all TADA genes, relative risk of promoter and enhancer estimated separately</a></p></li>
<li><p><a href="./161229_fitting_model_50bp_more_epi_features_and_phastcons_features.html">ASD_manuscript data, 3 epigenome data sets and 1 phastcons100way, first adjust mutation rate (Not including nonsynonymous mutations), then esitamte relative risks using only top6% TADA genes or all genes, estimate the relative risk of active promoters and enhancers jointly.</a></p></li>
<li><p><a href="./170110_fitting_model_50bp_more_epi_features_separate_Noonan_and_Roadmap_and_phastcons_features_not_using_epi_features_for_mutation_rates_temp.html">ASD_manuscript data, 4 epigenome data sets (separate out Noonan brain H3K27ac and Roadmap brain H3K27ac, when estimating relative risks)and 1 phastcons100way, first adjust mutation rate, then esitamte relative risks using only top6% TADA genes, estimate the relative risk of active promoters and enhancers jointly. Notice that when adjusting mutation rates, didn’t use epigenomic features and base-level annotation features (e.g., conservation scores)</a></p></li>
<li><p><a href="./170111_estimate_RR_of_splicing_mutations_along_with_other_epigenomic_and_annotation_features.html">ASD manuscript data, 1 epignome dataset (Noonan_brain_Roadmap_union), 1 phastcons conservation score, and 1 for splicing mutations. Didn’t exclude splicing mutations that are also epigenome mutations for now. Adjust for mutation rates using promoter, coding and cg contents. Use the Top 6% TADA genes for estimating relative risks. Relative risk estimtion for splicing mutations is done at the gene level not at the window level. Splicing mutation rates scaled relative to the adjusted synonymous mutation rates.</a></p></li>
<li><p><a href="./170115_fitting_model_50bp_more_epi_features_and_phastcons_features_not_using_epi_features_for_mutation_rates_all_coding_mutations_no_splicing_estimate.html">ASD_manuscript data, 3 epigenome data sets and 1 phastcons100way, first adjust mutation rate (including both synonymous mutations and nonsynoymous mutations), then esitamte relative risks using only top6% TADA genes or all genes, estimate the relative risk of active promoters and enhancers jointly. Splicing mutations were not incoporated in relative risk estimation here.</a></p></li>
<li><p><a href="./170117_fitting_model_50bp_rr_using_all_genes_more_epi_features_and_phastcons_features_not_using_epi_features_for_mutation_rates_all_coding_mutations_no_splicing_estimate.html">ASD_manuscript data, 3 epigenome data sets and 1 phastcons100way, first adjust mutation rate (including both synonymous mutations and nonsynoymous mutations), then esitamte relative risks using all all genes, estimate the relative risk of active promoters and enhancers jointly. Splicing mutations were not incoporated in relative risk estimation here (running time 30h).</a></p></li>
<li><p><a href="./170117_fitting_model_50bp_rr_using_all_genes_more_epi_features_and_phastcons_features_not_using_epi_features_for_mutation_rates_all_coding_mutations_no_splicing_estimate.html">ASD_manuscript data, 1 epigenome data sets and 1 phastcons100way, first adjust mutation rate (including both synonymous mutations and nonsynoymous mutations), then esitamte relative risks using all all genes, estimate the relative risk of active promoters and enhancers jointly. Splicing mutations were not incoporated in relative risk estimation here (running time 30h).</a></p></li>
</ul>
</div>
<div id="adjust-mutation-rate-using-optim-1bp-base-level." class="section level3">
<h3>5. Adjust mutation rate using optim(): 1bp base level.</h3>
<ul>
<li><a href="./170105_base_level_simulation_and_mutation_rate_adjustion_parameter_estimation_with_categorization.html">With categorization: test method using simulated dataset. 1e6 bases, 5 binary features and 2 continous features</a></li>
</ul>
</div>
<div id="window-level-bayes-factor-for-each-gene-and-novel-gene-identification." class="section level3">
<h3>6. Window-level Bayes Factor for each gene and novel gene identification.</h3>
<ul>
<li><p><a href="./170115_generate_logBF_and_new_TADA_genes_all_splicing_mutations.html">Used mutation rates ajusted by promoter, within 10kb non-coding regions, coding regions and CG contents (All coding mutations included in model training). Relative risks were estimated using Top 6% TADA genes when estimating relative risks of Noonan+Roadmap Brain H3K27ac, ENCODE brain DHS, Roadmap brain DHS and phastcons100way conservation scores. Only Noonan+Roadmap H3K27ac have a significant effect and thus is used in calculating Bayese factor for each gene. The splicing mutations (I haven’t excluded splicing mutations that overlap with other functional mutations here, but the overlap should be small) were added at the gene level in the form of splicing Bayes factor</a></p></li>
<li><p><a href="./170115_generate_logBF_and_new_TADA_genes_all_splicing_mutations_using_1.9_rr_for_noncoding.html">Used mutation rates ajusted by promoter, within 10kb non-coding regions, coding regions and CG contents (All coding mutations included in model training). Relative risks of Noonan+Roadmap H3K27ac mutations and splicing mutations were based on the first version of the manuscript.</a></p></li>
</ul>
</div>
<div id="model-at-the-base-level" class="section level3">
<h3>7. Model at the base level</h3>
<ul>
<li><p><a href="./170131_base_level_model_workflow.html">General workflow of base-level modeling</a></p></li>
<li><p><a href="./170125_base_level_two_feature_mixture_poisson_rr_estimate.html">Compare the cost of time between using categorization and non-categorization optimization. Two binary features were included for this analysis. The catagorization approach works as follows: 1, Going through the dataset once to first partition by genename, then by feature pattern. 2. pre-computed all parts of the optimization function that don’t need relative risk parameters. 3. optimization. Step 2 is very important to accelerate speed.</a></p></li>
<li><p><a href="./170127_base_level_RR_estimate_with_only_1_epi_feature_and_top_6pct_TADA_genes.html">Adjust for mutation rate using features including promoter, coding, within-10kb noncoding, and cg contents for each 50-bp window. Apply the adjusting factor to each single base. Then use top6% TADA genes to estimate relative risk for Noonan+Roadmap brain H3k27ac mutations by a mixture Poisson model. Categorization optimization was applied.</a></p></li>
<li><p><a href="./170131_base_level_RR_estimate_toy_model_using_optimization_with_categorization_for_big_data.html">Show the performance of [adjust_mutation_rate_window_to_base]+[estimate_effect_size_for_simulation_data_mixture_with_categorization_v2] is consistent with [adjust_mutation_rate_window_to_base_RCC]+[estimate_effect_size_for_simulation_data_mixture_with_categorization_for_big_data]. Demonstrate the optimization function for big data works well.</a></p></li>
</ul>
<hr />
<p><strong>Last updated:</strong> 2017-03-01 16:56:58</p>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
